
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Bioops</title>
  <meta name="author" content="Bioops">

  
  <meta name="description" content="5月6日的Nature同时发布了两篇研究方向非常相似的论文，揭示了染色体镶嵌性会随着年龄的增长而增加，并与癌症发生有一定关联。 一篇由National Cancer Institute (NCI)的Stephen Chanock领导完成。将 31,717个癌症患者和26, &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bioops.info/page4">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="Bioops" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="/javascripts/libs/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Fjalla+One" rel="stylesheet" type="text/css">
<!--- MathJax Configuration -->
<script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-21431888-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   class="collapse-sidebar sidebar-footer" >
  <header role="banner"><hgroup>
  <h1><a href="/">Bioops</a></h1>
  
    <h2>Bioinformatics=(ACGAAG->AK)+(#!/bin/sh)+(P(A|B)=P(B|A)*P(A)/P(B))</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscribe" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 13.310204,73.332654 C 5.967347,73.332654 0,79.322448 0,86.621428 c 0,7.338776 5.967347,13.262246 13.310204,13.262246 7.370408,0 13.328572,-5.92245 13.328572,-13.262246 0,-7.29898 -5.958164,-13.288774 -13.328572,-13.288774 z M 0.01530612,33.978572 V 53.143878 C 12.493878,53.143878 24.229592,58.02347 33.068368,66.865306 41.894898,75.685714 46.767346,87.47449 46.767346,100 h 19.25 C 66.017346,63.592858 36.4,33.979592 0.01530612,33.978572 l 0,0 z M 0.03877552,0 V 19.17449 C 44.54796,19.17551 80.77551,55.437756 80.77551,100 H 100 C 100,44.87653 55.15102,0 0.03877552,0 z"></path></svg></a></li>
  
</ul>
  
  
  
  
  
<ul class="subscribe">
  <li><a href="https://github.com/bioops" rel="subscribe-github" title="@bioops on GitHub"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 50,0 C 22.385714,0 0,22.385714 0,50 0,77.614286 22.385714,100 50,100 77.614286,100 100,77.614286 100,50 100,22.385714 77.614286,0 50,0 z m 29.692858,79.692858 c -3.859184,3.859182 -8.351022,6.887754 -13.35,9.00306 -1.27041,0.536736 -2.560204,1.009184 -3.867348,1.415306 v -7.493878 c 0,-3.938774 -1.35102,-6.835714 -4.053062,-8.690816 1.692858,-0.163264 3.24694,-0.390816 4.663266,-0.683672 1.416326,-0.292858 2.913266,-0.716328 4.491838,-1.27041 1.57857,-0.55408 2.994896,-1.213264 4.247958,-1.97755 1.253062,-0.765306 2.458164,-1.758164 3.613266,-2.978572 1.155102,-1.220408 2.12449,-2.604082 2.905102,-4.15 0.780612,-1.545918 1.4,-3.40204 1.855102,-5.566326 0.455102,-2.164286 0.683674,-4.54898 0.683674,-7.153062 0,-5.045918 -1.643878,-9.341836 -4.931634,-12.890816 C 77.44796,33.35 77.285714,29.10204 75.463266,24.512244 l -1.22143,-0.145918 c -0.845918,-0.09796 -2.368366,0.260204 -4.565306,1.07449 -2.196938,0.814286 -4.663264,2.14796 -7.396938,4.004082 -3.87449,-1.07449 -7.893878,-1.611224 -12.061224,-1.611224 -4.19898,0 -8.203062,0.536734 -12.012246,1.611224 -1.72449,-1.17245 -3.361224,-2.139796 -4.907142,-2.905102 C 31.753062,25.77449 30.516326,25.254082 29.587756,24.97653 28.660204,24.7 27.79796,24.528572 27,24.463266 c -0.79796,-0.0653 -1.310204,-0.08062 -1.537756,-0.04898 -0.22755,0.03164 -0.390816,0.0653 -0.487754,0.09796 -1.82347,4.62245 -1.985714,8.87143 -0.487756,12.743878 -3.287754,3.54796 -4.931632,7.844898 -4.931632,12.890816 0,2.604082 0.227552,4.988776 0.683674,7.153062 0.456122,2.164286 1.07449,4.020408 1.855102,5.566326 0.780612,1.545918 1.75,2.929592 2.905102,4.15 1.155102,1.220408 2.360204,2.213266 3.613264,2.978572 1.253062,0.766326 2.669388,1.42449 4.24796,1.97755 1.578572,0.554082 3.07551,0.976532 4.491836,1.27041 1.416328,0.292856 2.970408,0.521428 4.663266,0.683672 -2.669388,1.82347 -4.004082,4.720408 -4.004082,8.690816 v 7.639796 C 36.536734,89.818368 35.083674,89.3 33.656122,88.695918 c -4.99898,-2.115306 -9.490816,-5.143878 -13.35,-9.00306 -3.859184,-3.859184 -6.887754,-8.351022 -9.00306,-13.35 C 9.1163263,61.171428 8.0071428,55.67347 8.0071428,50 c 0,-5.67347 1.1091835,-11.171428 3.2969392,-16.342858 2.115306,-4.998978 5.143878,-9.490816 9.00306,-13.35 3.859184,-3.859182 8.351022,-6.887754 13.35,-9.00306 C 38.828572,9.1163266 44.32653,8.0071428 50,8.0071428 c 5.67347,0 11.171428,1.1091838 16.342858,3.2969392 5,2.115306 9.490816,5.143878 13.35,9.00306 3.859182,3.859184 6.887754,8.351022 9.00306,13.35 2.186736,5.17245 3.295918,10.67041 3.295918,16.342858 0,5.672448 -1.109182,11.171428 -3.296938,16.342858 -2.115306,4.998978 -5.143878,9.490816 -9.00204,13.35 l 0,0 z"></path></svg></a></li>
</ul>
  
  
  
  
  
  
  
    
      <form action="http://google.com/search" method="get">
        <fieldset role="search">
          <input type="hidden" name="sitesearch" value="bioops.info" />
    
          <input class="search" type="text" name="q" results="0" placeholder="Search"/>
        </fieldset>
      </form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div class="blog-index">
  
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/05/relationship-of-clonal-mosaicism-aging-cancer/">Nature: 染色体镶嵌性、年龄和癌症之间的关系</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-05-08T00:00:00+00:00'><span class='date'><span class='date-month'>May</span> <span class='date-day'>8</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/05/relationship-of-clonal-mosaicism-aging-cancer/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>5月6日的Nature同时发布了两篇研究方向非常相似的论文，揭示了染色体镶嵌性会随着年龄的增长而增加，并与癌症发生有一定关联。</p>

<p><a href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.2270.html" target="_blank">一篇</a>由<a href="http://www.cancer.gov/" target="_blank">National Cancer Institute (NCI)</a>的<a href="http://dceg.cancer.gov/about/staff-bios/chanock-stephen" target="_blank">Stephen Chanock</a>领导完成。将 31,717个癌症患者和26,136健康人血液或口腔拭子的SNP芯片数据分析比较。健康人中染色体复制异常的比率随着年龄而增长，50岁以下0.23%，到了70多岁就飙到了1.91%。平均下来，癌症患者中染色体复制异常的比率比健康者高一些（0.97% VS 0.74%）。有些白血病患者被诊断出来前一年的血液样本中的染色体复制异常也比健康人普遍。</p>

<p>从<a href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.2271.html" target="_blank">另一篇</a>来自<a href="https://www.genevastudy.org/" target="_blank">Gene Environment Association Studies Consortium (GENEVA)</a>。他们分析了超5万人血液样本的SNP芯片数据，发现50岁以下人的基因组样本中，复制异常低于0.5%，而在50岁之后的人中迅速增长至2–3%。有很多复制异常都在以前发现的和血液癌症有关的位点上。虽然这些有“污点记录”的位点只占所发现的异常位点的3%，统计学分析表明，有复制异常的人得血液癌症的概率是正常人的十倍。</p>

<p>笔记：上面涉及到一些概率，我没仔细看文章，可能理解有错。人体细胞复制不可靠。年老的人更容易得癌症。50岁是个坎儿。生孩子要趁早。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/04/rna-studies-under-fire/">上了火的RNA-seq</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-04-25T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>25</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/04/rna-studies-under-fire/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>Nature News今天的一篇<a href="http://www.nature.com/news/rna-studies-under-fire-1.10502" target="_blank">评论</a>，说是现在越来越多人使用高通量的RNA数据，但忽略了统计分析的严密性和重要性，而得到了一些值得怀疑的结论。</p>

<p>举了两个例子。</p>

<p>第一个，2010年，Harvard的<a href="http://www.hms.harvard.edu/dms/neuroscience/fac/dulac.php">Catherine Dulac</a>和<a href="http://www.neuro.utah.edu/people/faculty/gregg.html">Christopher Gregg</a>在Science上发了一篇<a href="http://www.sciencemag.org/content/329/5992/643.long" target="_blank">文章</a>，号称1300多个小鼠的基因都属于烙印基因（<a href="http://en.wikipedia.org/wiki/Genomic_imprinting" target="_blank">imprinted gene</a>：两个等位基因，一个表达一个沉默）。但是Standford的Tomas Babak用同样的数据得到了不同的结果，并于2012年3月将其<a href="http://www.plosgenetics.org/article/info%3Adoi%2F10.1371%2Fjournal.pgen.1002600" target="_blank">分析结果</a>发表在PLoS  Genetics上。Babak认为原文章所用的统计方法会有很高的假阳性。</p>

<p>另一个例子就是前段时间引起巨大争议的关于RNA编辑广泛存在的那篇<a href="http://www.sciencemag.org/content/333/6038/53" target="_blank">文章</a>。后来又有多篇评论发表在Science上，强烈置疑该文章分析方法的可靠性。</p>

<p>造成这种争议的原因是什么呢？</p>

<p>一是RNA-Seq还没有像DNA测序一样建立起一个标准化的、错误偏差少的分析方法。</p>

<p>二是这些文章一般号称在生物学上某一点上取得了突破，所以一般是搞生物的来审核，缺少搞数学和统计的人把关。</p>

<p>个人觉得：很少有懂统计的人把生物理解的很清楚，更少有搞生物的把统计学好的，现在生物大数据时代，都需要补补课了。这两篇文章因为是有了“突破性的发现”，所以很多双眼睛在盯着，引起了怀疑。至于其他灌水文章，也不知道能有多少篇把统计方法用对的。这算是方法论的问题，只要提出来了，多探索一般能有个定论。更可怕的是搞科研的为了发文章，不惜更改数据或忽略负面数据，前几天不是有人<a href="http://online.wsj.com/article/SB10001424052970204571404577257513760102538.html" target="_blank">抱怨</a>癌症治疗的研究大部分都是在骗人么？！</p>

<p>希望以后有所改观，至少做到自重吧。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/04/guide-to-eukaryoticgenome-annotation/">真核基因组注释</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-04-23T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>23</span><span class='date-suffix'>rd</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/04/guide-to-eukaryoticgenome-annotation/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>测一个未知基因组（<em>de nove</em> sequence），要进行测序、拼接及注释。关于测序仪和拼接软件已经讲的很多了，很少有关于基因组注释的文章。一篇最近在Nature Review Genetics上的<a href="http://www.nature.com/nrg/journal/v13/n5/full/nrg3174.html" target="_blank">文章</a>，A beginner&#8217;s guide to eukaryotic genome annotation，非常详细地讲解了如何做基因组注释，是一篇非常好的入门文章。</p>

<p>基因组拼接好后，一般要先进行重复序列的检测和注释，然后mask掉这些重复序列，再进行编码基因的预测（有时候也预测非编码RNA），最后一步是整合。因为要通过不同的方法和参考来源来预测，会得到不同的结果，整合时综合考虑预测错误和可变剪接，得到可靠的注释，这一步要一个个手工检测。</p>

<p>有很多软件可以做注释（可见文章内的列表），主要分为ab initio和evidence-driven两种预测方法。</p>

<p>现在RNA-seq技术也很成熟了，一般都是在测基因组时也要做RNA-seq，这些RNA-seq既可用于分析基因的表达，也是非常好的基因注释的参考资源。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/04/desktop-sequencers-comparison/">三种小型测序仪的比较</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-04-23T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>23</span><span class='date-suffix'>rd</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/04/desktop-sequencers-comparison/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://pathogenomics.bham.ac.uk/staff/nloman.html" target="_blank">Nicholas Loman</a>在Nature Biotechnology发了一篇<a href="http://www.nature.com/nbt/journal/vaop/ncurrent/full/nbt.2198.html" target="_blank">文章</a>，综合比较了三个小型测序仪，<a href="http://www.gsjunior.com/" target="_blank">454 GS Junior</a> (Roche), <a href="http://www.illumina.com/systems/miseq.ilmn" target="_blank">MiSeq</a> (Illumina) and <a href="http://www.iontorrent.com/ion-personal-genome-machine-sequencer/" target="_blank">Ion Torrent PGM </a>(Life Technologies)，并且接受了Nature的<a href="http://www.nature.com/news/next-generation-genome-sequencers-compared-1.10497" target="_blank">采访</a>。</p>

<p>小型测序仪一般只有台式电脑大小，价格便宜、操作简单，但通量相对较小，一般用于重测序和小基因组或基因组片段的<em>de novo</em>测序。（BTW, 我们实验室有一台MiSeq）</p>

<p>正如Nicholas Loman所说，这三个测序仪都有各自的优缺点。PGM单位时间内通量高；MiSeq每个run通量高；MiSeq错误率低，454可读取最长的序列片段但通量低；PGM和454在测重复序列时容易出错，且多次重复测序仍然无法消除；MiSeq整体稳定性和质量较好。价格方面没有透露。我所了解到的是MiSeq比PGM高一点，454最贵，但PGM更新换代花费少，只需要更新一个芯片。</p>

<p>但是话又说回来了，现在测序技术发展非常快，远远<a href="http://medgadget.com/2011/05/dna-sequencing-beating-moores-law.html" target="_blank">超过</a>了电脑行业的所谓摩尔定律。别看现在Illumina笑傲群雄，Ion Torrent正快速追赶，加上Nanopore技术的异军突起，估计会各领风骚几年。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/04/wellcome-trust-academic-spring/">Wellcome Trust即将推出开放期刊eLife</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-04-10T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>10</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/04/wellcome-trust-academic-spring/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://www.wellcome.ac.uk/" target="_blank">Wellcome Trust</a>基金会去年<a href="http://wellcometrust.wordpress.com/2011/11/07/elife-a-journal-by-scientists-for-scientists/">宣布</a>计划出版一个新的可开放获取的顶尖的学术期刊系列，eLife。基金会的主管Mark Walport<a href="http://www.guardian.co.uk/science/2012/apr/09/wellcome-trust-academic-spring">现在宣称</a>eLife的发布工作已经到了最后的阶段，并希望eLife成为能和Nature和Science竞争但可免费获取的期刊系列。</p>

<p>eLife的发布将给“学术之春”（<a href="http://americanlibrariesmagazine.org/features/04042012/academic-spring" target="_blank">academic spring</a>）活动注入一股强心剂。本来该活动是由于Elsevier高昂的订阅费而导致数学家Timothy Gowers<a href="https://gowers.wordpress.com/2012/01/21/elsevier-my-part-in-its-downfall/" target="_blank">号召</a>学术界联合抵制Elsevier，不发表，不审核，不编辑。现在已经有<a href="http://thecostofknowledge.com/" target="_blank">9000</a>多人签名支持，并引起了关于学术期刊免费开放的广泛讨论。</p>

<p>Wellcome Trust是仅次于Bill &amp; Melinda Gates Foundation（盖茨夫妇）对科研投资最大的非政府组织，每年投入超过600万欧。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/04/heng-li-2012-benjamin-franklin-award/">李恒获得2012年Benjamin Franklin奖</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-04-10T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>10</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/04/heng-li-2012-benjamin-franklin-award/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>旧闻了，但觉得有必要拿出来分享一下。</p>

<p><a href="http://lh3lh3.users.sourceforge.net/" target="_blank">李恒</a>因在下一代测序技术的软件开发与应用上的突出贡献，<a href="http://www.bio-itworld.com/2012/03/14/broads-heng-li-wins-2012-benjamin-franklin-award.html" target="_blank">获得</a>了<a href="http://www.bioinformatics.org/" target="_blank">Bioinformatics.Org</a>颁发的2012年<a href="http://www.bioinformatics.org/franklin/" target="_blank">Benjamin Franklin奖</a>。
<p style="text-align: left;">李恒算是生物信息学界的一个牛人了。南大生物物理本科毕业，2006年在中科院理论物理研究所获得博士学位。读博期间师从<a href="http://power.itp.ac.cn/~zheng/" target="_blank">郑伟谋</a>，但貌似是在Sanger的<a href="http://www.sanger.ac.uk/research/faculty/rdurbin/" target="_blank">Richard Durbin</a>的指导下与BGI的<a href="http://www.weibo.com/n/%E7%8E%8B%E4%BF%8A%E5%8D%8E%E5%A4%A7%E5%9F%BA%E5%9B%A0" target="_blank">王俊</a>等人合作搞<a href="http://www.treefam.org/" target="_blank">TreeFam</a>数据库。博士毕业后理所应当地到了Richard Durbin实验室做博士后，并在此期间开发了后来名震天下的<a href="http://samtools.sourceforge.net/" target="_blank">SAMtools</a>、<a href="http://bio-bwa.sourceforge.net/" target="_blank">BWA</a>和<a href="http://maq.sourceforge.net/" target="_blank">MAQ</a>。而Richard Durbin也培养了另外一个Benjamin Franklin奖获得者，2005年获该奖的<a href="http://www.ebi.ac.uk/~birney/" target="_blank">Ewan Birney</a>。他工作于EBI，领导了<a href="http://www.bioperl.org/">Bioperl</a>和<a href="http://www.ensembl.org/">Ensembl</a>的开发，以及<a href="http://encodeproject.org/ENCODE/" target="_blank">ENCODE</a>工程。李恒后来又转到<a href="http://www.broadinstitute.org/">Broad Institute</a>与<a href="http://genepath.med.harvard.edu/~reich/" target="_blank">David Reich</a>和<a href="http://www.broadinstitute.org/about/bios/bio-altshuler.html">David Altshuler</a>共事，进行人类群体遗传学方面的研究，并经常在Nature和Science灌水。他在<a href="http://seqanswers.com/" target="_blank">SEQanswers</a>和<a href="http://www.biostars.org/" target="_blank">BioStars</a>上也很活跃（两个论坛上的id应该都是lh3）。</p></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/04/next-generation-sequencing-ngs-technologies-market/">2016年下一代测序技术的全球市场总额将达23.43亿刀</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-04-09T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>9</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/04/next-generation-sequencing-ngs-technologies-market/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://www.marketsandmarkets.com/" target="_blank">MarketsandMarkets</a>新发布了一个比较系统的关于下一代测序技术的<a href="http://www.marketsandmarkets.com/Market-Reports/next-generation-sequencing-ngs-technologies-market-546.html" target="_blank">报告</a>，总结现状，更主要的是为了预测未来。</p>

<p>2011年，下一代测序技术的全球市场总额是8.425亿刀，预计会按照每年22.7%的速度递增，2016年将达到23.43亿刀。北美地区现占51.8%的市场份额，欧洲31.1%。</p>

<p>市场增长的主要驱动力来自于越来越便宜的测序技术以及测序被应用于越来越多的领域，如癌症、生物能源、海洋生物学，畜牧业、农业以及兽医药学。即将来临的个人基因组测序的普及，将更大地刺激对测序设备的需求。</p>

<p>现阶段这个领域主要的公司有 <a href="http://www.illumina.com/" target="_blank">Illumina</a> (美帝)、<a href="http://www.lifetechnologies.com/" target="_blank">Life Technologies</a> (美帝)、<a href="http://www.454.com/" target="_blank">454 Roche </a>(瑞士)、<a href="http://www.nanoporetech.com/" target="_blank">Oxford Nanopore Technologies</a> (英帝)、 and <a href="http://www.pacificbiosciences.com/" target="_blank">Pacific Biosciences</a> (美帝)。在目前以大规模短片段测序占主流的情况下，Illumina和Life Technologies在这一技术上<a href="http://bioops.info/2012/01/life-technologies-illumina-new-machine-competitio/" target="_blank">竞争非常激烈</a>。以后大片段的测序是趋势，而454 Roche和Pacific Biosciences又占有优势。当然更少不了新兴的Oxford Nanopore Technologies，利用纳米测序技术能够长片段、单细胞测序且设备非常小，有很大的潜力。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/03/bioperl-howto-seqio/">利用Bioperl进行序列输入和输出（Bioperl HOWTO翻译13）</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-03-19T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>19</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/03/bioperl-howto-seqio/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><h3>利用Bioperl进行序列输入和输出</h3>
<a href="http://www.bioperl.org/wiki/HOWTO:SeqIO" target="_blank">英文原文</a></p>

<div id="toctitle">
<h2>目录</h2>
</div>
<ul>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Authors">1 作者 </a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Copyright">2 版权</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#The_basics">3 基础</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#10_second_overview">4 十秒速览</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Background_Information">5 背景</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Formats">6 格式</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Working_Examples">7 举例</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#To_and_From_a_String">8 字符串相关</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#And_more_examples...">9 更多示例</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Caveats">10 警告</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Error_Handling">11 纠错</a></li>
	<li><a href="http://bioops.info/2012/03/bioperl-howto-seqio/#Speed.2C_Bio::Seq::SeqBuilder">12 使用Bio::Seq::SeqBuilder提速</a></li>
</ul>
<a id="Authors" name="Authors"></a>
<h2>作者</h2>
<ul>
	<li><a title="Ewan Birney" href="http://www.bioperl.org/wiki/Ewan_Birney">Ewan Birney</a> &lt;<a title="mailto:birney-at-ebi.ac.uk" href="mailto:birney-at-ebi.ac.uk" rel="nofollow">birney at ebi.ac.uk</a>&gt;</li>
	<li><a title="Brian Osborne" href="http://www.bioperl.org/wiki/Brian_Osborne">Brian Osborne</a> &lt;<a title="mailto:briano-at-bioteam.net" href="mailto:briano-at-bioteam.net" rel="nofollow">briano at bioteam.net</a>&gt;</li>
	<li><a title="Darin London" href="http://www.bioperl.org/wiki/Darin_London">Darin London</a> &lt;<a title="mailto:darin.london-at-duke.edu" href="mailto:darin.london-at-duke.edu" rel="nofollow">darin.london at duke.edu</a>&gt;</li>
</ul>
<a id="Copyright" name="Copyright"></a>
<h2>版权</h2>
<a title="Ewan Birney" href="http://www.bioperl.org/wiki/Ewan_Birney">Ewan Birney</a>拥有此文档版权。 基于 <a href="http://www.perl.com/pub/language/misc/Artistic.html">Perl Artistic License</a>协议有限共享。</p>

<p><a id="The_basics" name="The_basics"></a>
<h2>基础</h2>
作者假定阅读此文章的读者没有用过<a title="BioPerl" href="http://www.bioperl.org/wiki/BioPerl">BioPerl</a>，也许是一个想要获取一些序列信息的生物学者；也许是一个想学习一下目前流行的所谓“<a href="http://www.bioperl.org/wiki/Bioinformatics" target="_blank">生物新信息学</a>”的IT人士。我们讲解的第一个脚本就是如何从包含一个或多个序列的序列文件中提取相关信息。</p>

<p>一个小建议：坚持使用<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>模块来处理序列。先来看一下完成任务所需要的前几行脚本：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1">#!/bin/perl</span>
</span><span class='line'>
</span><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$file</span>         <span class="o">=</span> <span class="nb">shift</span><span class="p">;</span> <span class="c1"># get the file name, somehow</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqio_object</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="nv">$file</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq_object</span>   <span class="o">=</span> <span class="nv">$seqio_object</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">;</span>
</span></code></pre></td></tr></table></div></figure>

<p>为什么要用<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>？部分原因是SeqIO能识别各种不同的序列格式并从中提取然后创建相应的BioPerl对象。有些序列格式很简单，例如<a title="FASTA sequence format" href="http://www.bioperl.org/wiki/FASTA_sequence_format">FASTA</a>。fasta格式仅包含序列和一个简单的序列标识符，而不包含序列内部的一些详细的特征或信息（<a title="HOWTO:Feature-Annotation" href="http://www.bioperl.org/wiki/HOWTO:Feature-Annotation">Feature-Annotation HOWTO</a>里具体讲了序列的特征）。一般来说，SeqIO能识别不同的序列格式，并创建相应的对象，例如输入fasta，SeqIO会创建一个<a title="Module:Bio::Seq" href="http://www.bioperl.org/wiki/Module:Bio::Seq">Bio::Seq</a>对象，而输入Genbank或EMBL这种包含注释信息的序列文件，SeqIO则会创建一个<a title="Module:Bio::Seq::RichSeq" href="http://www.bioperl.org/wiki/Module:Bio::Seq::RichSeq">Bio::Seq::RichSeq</a>对象。</p>

<p>至于SeqIO针对不同格式的序列文件，到底创建了哪种对象，我们并不需要关心。SeqIO会把自动这些细节问题处理好。</p>

<p><a id="10_second_overview" name="10_second_overview"></a>
<h2>十秒速览</h2>
生物信息学的工作会接触到各种格式的序列。而又有几乎同样多的程序来处理这些序列。SeqIO的优势是可以通过创建和输出序列对象的方式来处理很多不同格式的序列。可以将SeqIO想象成一个智能的文件控制系统。</p>

<p><a id="Background_Information" name="Background_Information"></a>
<h2>背景</h2>
SeqIO系统可以处理很多复杂的序列格式。提供给Bioperl输入序列（序列可以是普通文件、标准输入输出（<tt>STDIN<span style="font-family: Georgia, 'Times New Roman', 'Bitstream Charter', Times, serif;">和</span></tt><tt>STDOUT）、变量等等</tt>）以及序列格式，Bioperl即可从序列中提取所有可以提取的信息。有时候并不需要说明序列格式。SeqIO可通过文件扩展名或文件内容来猜测出序列格式。如果文件没有扩展名或者文件内容不完整甚至压根输入的就不是一个文件，Bioperl直接默认是fasta格式（译者注：而这样就很可能出错）。除非明确知道所用的序列就是fasta格式，最好还是明确告知Bioperl你所输入序列的格式。</p>

<p>SeqIO可以接受多种形式的输入。唯一的要求是序列必须包含在Perl所规定的“句柄”中（详见<a title="http://search.cpan.org/perldoc?IO::Handle" href="http://search.cpan.org/perldoc?IO::Handle" rel="nofollow">IO::Handle</a>）。大多数人使用文件句柄或者标准输入/输出（STDIN/STDOUT）。Perl还提供了如何将一段字符串转为句柄（见下文）。这样一来，几乎所有东西都可以作为SeqIO的输入，只要其中含有序列信息。SeqIO的主要工作流程就是获取一个句柄，然后依次将一条条序列信息转化为SeqI对象，若有需要则创建一个句柄以输出序列文件。SeqIO可以识别多个序列之间的分隔符，如genbank格式中的“// ”，fasta格式中的“&gt;”，也能识别用关键字或符号标识的其他序列信息。</p>

<p><a id="Formats" name="Formats"></a>
<h2>格式</h2>
BioPerl的SeqIO模块能识别很多序列格式并且互相转换. 表1是最新版本的SeqIO所支持的格式。（译者注：表格见<a href="http://www.bioperl.org/wiki/HOWTO:SeqIO#Formats" rel="nofollow">原文</a>。如前所述，SeqIO将输入的不同格式序列用相应的模块转换成相应的序列对象）</p>

<p><strong>注:</strong> <a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>需要<a title="Bioperl-ext" href="http://www.bioperl.org/wiki/Bioperl-ext">bioperl-ext</a>包来和<a title="http://staden.sourceforge.net/" href="http://staden.sourceforge.net/" rel="nofollow">Staden</a>包中的<tt>io_lib库<span style="font-family: Georgia, 'Times New Roman', 'Bitstream Charter', Times, serif;">才能读取</span></tt><tt>scf, abi, alf, pln, exp, ctf,和ztr格式的序列</tt>.</p>

<p>也许有些人会对上面提到的不同序列格式对应不同序列对象产生疑问，如“为了转换序列格式，如何才能把一个<em>PrimarySeq</em>对象转换为<em>RichSeq</em>对象？”答案是：这种问题压根就不用担心，SeqIO会自动转换。之所以不同的格式对应不同的对象，因为有些序列有很少信息，而有些格式又包含很多信息。（译者注：可能是避免大炮打蚊子似的浪费） 你并不需要关心SeqIO是如何转换的。列出表2，展示一下常见的序列格式和对应的序列对象，也可满足一些人的好奇心。（表格见<a href="http://www.bioperl.org/wiki/HOWTO:SeqIO#Formats" rel="nofollow">原文</a>。）</p>

<p><a id="Working_Examples" name="Working_Examples"></a>
<h2>举例</h2>
先举个最简单的例子，如何从序列文件中提取<a title="Accession number" href="http://www.bioperl.org/wiki/Accession_number">accession number</a>。</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
<span class='line-number'>36</span>
<span class='line-number'>37</span>
<span class='line-number'>38</span>
<span class='line-number'>39</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1"># first, bring in the SeqIO module</span>
</span><span class='line'>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># Notice that you do not have to use any Bio:SeqI</span>
</span><span class='line'><span class="c1"># objects, because SeqIO does this for you. In fact, it</span>
</span><span class='line'><span class="c1"># even knows which SeqI object to use for the provided</span>
</span><span class='line'><span class="c1"># format.</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># Bring in the file and format, or die with a nice</span>
</span><span class='line'><span class="c1"># usage statement if one or both arguments are missing.</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>  <span class="o">=</span> <span class="s">&quot;getaccs.pl file format\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$file</span>   <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$format</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># Now create a new SeqIO object to bring in the input</span>
</span><span class='line'><span class="c1"># file. The new method takes arguments in the format</span>
</span><span class='line'><span class="c1"># key =&gt; value, key =&gt; value. The basic keys that it</span>
</span><span class='line'><span class="c1"># can accept values for are &#39;-file&#39; which expects some</span>
</span><span class='line'><span class="c1"># information on how to access your data, and &#39;-format&#39;</span>
</span><span class='line'><span class="c1"># which expects one of the Bioperl-format-labels mentioned</span>
</span><span class='line'><span class="c1"># above. Although it is optional, it is good</span>
</span><span class='line'><span class="c1"># programming practice to provide &gt; and &lt; in front of any</span>
</span><span class='line'><span class="c1"># filenames provided in the -file parameter. This makes the</span>
</span><span class='line'><span class="c1"># resulting filehandle created by SeqIO explicitly read (&lt;)</span>
</span><span class='line'><span class="c1"># or write(&gt;).  It will definitely help others reading your</span>
</span><span class='line'><span class="c1"># code understand the function of the SeqIO object.</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                            <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&lt;$file&quot;</span><span class="p">,</span>
</span><span class='line'>                            <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$format</span><span class="p">,</span>
</span><span class='line'>                            <span class="p">);</span>
</span><span class='line'><span class="c1"># Now that we have a seq stream,</span>
</span><span class='line'><span class="c1"># we need to tell it to give us a $seq.</span>
</span><span class='line'><span class="c1"># We do this using the &#39;next_seq&#39; method of SeqIO.</span>
</span><span class='line'>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$inseq</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="k">print</span> <span class="nv">$seq</span><span class="o">-&gt;</span><span class="n">accession_number</span><span class="p">,</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>运行此脚本需要两个参数，输入序列的文件名和文件格式。上例显示了最基本的处理genbank文件的方式。其他格式，如<tt>fasta</tt>,<tt>swissprot</tt>, <tt>ace</tt>都是类似的，需要给bioperl提供其所支持的格式名称。（译者注：一般情况下bioperl可以自己猜测出来序列的格式，而不需提供format参数，为保险期间最好说明格式类型。）</p>

<p>需要注意的是，SeqIO默认是处理多个序列的。每一次调用next_seq时，返回的是下一个序列，直到返回一个undef，即文件末尾。 <span style="font-family: monospace;">脚本是一个一个序列依次读取，相比一次性把所有序列读取后再处理更节省内存。undef的重要性在于，当文件达到末尾的时候，undef可终止while循环。</span></p>

<p>下例是把<a title="EMBL sequence format" href="http://www.bioperl.org/wiki/EMBL_sequence_format">EMBL</a>文件中的所有序列信息都放先在一个数组里，然后从数组里提取所需信息。</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$input_file</span> <span class="o">=</span> <span class="nb">shift</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq_in</span>  <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                              <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;embl&#39;</span><span class="p">,</span>
</span><span class='line'>                              <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="nv">$input_file</span><span class="p">,</span>
</span><span class='line'>                              <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># loads the whole file into memory - be careful</span>
</span><span class='line'><span class="c1"># if this is a big file, then this script will</span>
</span><span class='line'><span class="c1"># use a lot of memory</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">@seq_array</span><span class="p">;</span>
</span><span class='line'><span class="k">while</span><span class="p">(</span> <span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$seq_in</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">()</span> <span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nb">push</span><span class="p">(</span><span class="nv">@seq_array</span><span class="p">,</span><span class="nv">$seq</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># now do something with these. First sort by length,</span>
</span><span class='line'><span class="c1"># find the average and median lengths and print them out</span>
</span><span class='line'>
</span><span class='line'><span class="nv">@seq_array</span> <span class="o">=</span> <span class="nb">sort</span> <span class="p">{</span> <span class="nv">$a</span><span class="o">-&gt;</span><span class="nb">length</span> <span class="sr">&lt;=&gt;</span> <span class="nv">$b</span><span class="o">-&gt;</span><span class="nb">length</span> <span class="p">}</span> <span class="nv">@seq_array</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$total</span> <span class="o">=</span> <span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$count</span> <span class="o">=</span> <span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="k">foreach</span> <span class="k">my</span> <span class="nv">$seq</span> <span class="p">(</span> <span class="nv">@seq_array</span> <span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$total</span> <span class="o">+=</span> <span class="nv">$seq</span><span class="o">-&gt;</span><span class="nb">length</span><span class="p">;</span>
</span><span class='line'>    <span class="nv">$count</span><span class="o">++</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;Mean length &quot;</span><span class="p">,</span><span class="nv">$total</span><span class="sr">/$count,&quot; Median &quot;,$seq_array[$count/</span><span class="mi">2</span><span class="p">]</span><span class="o">-&gt;</span><span class="nb">length</span><span class="p">,</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span></code></pre></td></tr></table></div></figure>

<p>现在来看如何转换序列格式。当你从一个序列文件创建一个<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>对象后，每次调用next_seq，Bioperl都会通过运行很多相关的脚本，来提取下一条序列的信息，并记录到一个SeqI对象中。神奇的是，next_seq一次性提取整条序列的所有信息，因为Bioperl能够理解序列的起始和终止位置，当提取完一条序列信息后自动停止，等待下一次next_seq的调用。同时，Bioperl也能识别出序列的注释信息，例如在genbank序列 <tt><a href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord#LocusB" target="_blank">LOCUS</a>行中的</tt><a href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord#GenBankDivisionB" target="_blank">DIVISION</a>信息。 要转换格式，就是把存储序列信息的一个<a title="Module:Bio::SeqI" href="http://www.bioperl.org/wiki/Module:Bio::SeqI">Bio::SeqI</a>对象以另一种格式输出到新文件中。 这时就要用<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>中的另一个方法，write_seq。write_seq的工作原理与刚才所说的读取序列的过程正好相反，并通过next_seq将输入和输出的过程连接起来。Bioperl可以将一个SeqI对象按照所需格式输出到新的序列文件中。看下面的示例会更清楚一些：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'><span class="c1"># get command-line arguments, or die with a usage statement</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>         <span class="o">=</span> <span class="s">&quot;x2y.pl infile infileformat outfile outfileformat\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$infile</span>        <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$infileformat</span>  <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outfile</span>       <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outfileformat</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># create one SeqIO object to read in,and another to write out</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq_in</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                             <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&lt;$infile&quot;</span><span class="p">,</span>
</span><span class='line'>                             <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$infileformat</span><span class="p">,</span>
</span><span class='line'>                             <span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq_out</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                              <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&gt;$outfile&quot;</span><span class="p">,</span>
</span><span class='line'>                              <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$outfileformat</span><span class="p">,</span>
</span><span class='line'>                              <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># write each entry in the input file to the output file</span>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nv">$seq_in</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$seq_out</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$inseq</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>可以将$seq_in和$seq_out想象成两个特殊的文件句柄，并且这个文件句柄“知道”序列及其格式。用文件句柄时一般用类似&lt;F&gt;的操作符，而$seq_in和$seq_out则使用next_seq()方法来读取或输出序列对象，如用“$seqio-&gt;write_seq($seq_object)”相对于“print F $line”。</p>

<p>注：<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>允许使用一种办法来模仿文件句柄的输入输出（也不知道用这种方法是傻还是聪明），让<code>&lt;F&gt;存储序列，“</code><code>print F”则是输出序列。但这种方式对很多人来说是一种非主流想法，看起来很别扭，有时候也会引起一些不必要的困惑。</code></p>

<p>这个可通用与序列格式转换的脚本只比前面例子里提取accession号和计算序列平均长度的脚本多了几行代码。（主要增加的部分是为了从命令行获取参数）这就是Bioperl的优势所在，可以用很少的代码完成很复杂的任务。</p>

<p>接着，我们来看通过简单修改上面例子中的代码来展示一下SeqIO的灵活性。例如，用其他程序输出一段内容，然后用Bioperl来接受其输出的内容，处理后输出到新文件中。这里需要明确亮点，一个和Perl有关，另一个是Bioperl特有的。Perl可以通过在一个句柄的名字前加“*”号将其转换（GLOB）成一个文件句柄，以便于后面使用。 另外，<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>可以将前面转换后的文件句柄作为<code>-fh的参数，以代替</code><code>-file参数。下面展示的例子中，all2y.pl可以从其他程序的输出端获取内容，使用时类似于：</code>
<pre>    &gt;cat myseqs.fa | all2y.pl fasta newseqs.gb genbank</pre>
其代码如下：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'><span class="c1"># get command-line arguments, or die with a usage statement</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>     <span class="o">=</span> <span class="s">&quot;all2y.pl informat outfile outfileformat\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$informat</span>  <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outfile</span>   <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outformat</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># create one SeqIO object to read in, and another to write out</span>
</span><span class='line'><span class="c1"># *STDIN is a &#39;globbed&#39; filehandle with the contents of Standard In</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqin</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                            <span class="o">-</span><span class="n">fh</span>     <span class="o">=&gt;</span> <span class="o">\*</span><span class="bp">STDIN</span><span class="p">,</span>
</span><span class='line'>                            <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$informat</span><span class="p">,</span>
</span><span class='line'>                            <span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqout</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                             <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&gt;$outfile&quot;</span><span class="p">,</span>
</span><span class='line'>                             <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$outformat</span><span class="p">,</span>
</span><span class='line'>                             <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># write each entry in the input file to the output file</span>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nv">$seqin</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$seqout</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$inseq</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>有时候可能根本就用不着文件。可以直接使用 <tt>STDIN<span style="font-family: Georgia, 'Times New Roman', 'Bitstream Charter', Times, serif;">和</span></tt><tt>STDOUT，将输出序列做为另一程序的输入，比如：</tt>
<pre>    cat *.seq | in2out.pl EMBL Genbank | someother program</pre>
代码如下：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'><span class="c1"># get command-line arguments, or die with a usage statement</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>     <span class="o">=</span> <span class="s">&quot;in2out.pl informat outformat\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$informat</span>  <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outformat</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># create one SeqIO object to read in, and another to write out</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqin</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                            <span class="o">-</span><span class="n">fh</span>     <span class="o">=&gt;</span> <span class="o">\*</span><span class="bp">STDIN</span><span class="p">,</span>
</span><span class='line'>                            <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$informat</span><span class="p">,</span>
</span><span class='line'>                            <span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outseq</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                             <span class="o">-</span><span class="n">fh</span>     <span class="o">=&gt;</span> <span class="o">\*</span><span class="bp">STDOUT</span><span class="p">,</span>
</span><span class='line'>                             <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$outformat</span><span class="p">,</span>
</span><span class='line'>                             <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># write each entry in the input to the output</span>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nv">$seqin</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$outseq</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$inseq</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p><a id="To_and_From_a_String" name="To_and_From_a_String"></a>
<h2>字符串相关</h2>
一个很常见的问题是：“如果我有一个字符串，其中包含了一系列的某格式的序列，我该如何才能把这些序列转换成Bio::Seq对象？”比如有些时候，需要用户在网页表格内输入一段序列，然后对这段序列做一些处理。可以通过把用户输入的一段字符串转化成符合Perl标准的句柄(Perl 5.8.0以后可以直接用open来完成），然后-fh参数就可设置成这个句柄，而不用设置成文件名。</p>

<p>下面的两个例子都不是完整的程序，只是给个基本的示例。假设通过<a title="wp:CGI" href="http://en.wikipedia.org/wiki/CGI">CGI</a>脚本获取到用户输入的序列和序列格式，并分别放到两个字符串变量中。刚才提到Bio::seqIO可通过文件后缀名识别序列格式，这里输入的序列不是文件，所以必须告知Bioperl文件格式。</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">IO::</span><span class="n">String</span><span class="p">;</span>   <span class="c1"># only needed for Perl versions previous to 5.8.0</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1">## get a string into $string somehow, with its format in $format, say from a web form.</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$string</span>   <span class="o">=</span> <span class="s">&quot;&gt;SEQ1\nacgt\n&gt;revseq1\ntgca\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$format</span>   <span class="o">=</span> <span class="s">&quot;fasta&quot;</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$stringfh</span> <span class="o">=</span> <span class="nn">IO::</span><span class="n">String</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="nv">$string</span><span class="p">);</span>                                         <span class="c1"># Use this for Perl BEFORE 5.8.0</span>
</span><span class='line'><span class="nb">open</span><span class="p">(</span><span class="nv">$stringfh</span><span class="p">,</span> <span class="s">&quot;&lt;&quot;</span><span class="p">,</span> <span class="o">\</span><span class="nv">$string</span><span class="p">)</span> <span class="ow">or</span> <span class="nb">die</span> <span class="s">&quot;Could not open string for reading: $!&quot;</span><span class="p">;</span>   <span class="c1"># Use this for Perl AFTER 5.8.0 (inclusive)</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqio</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span> <span class="k">new</span><span class="p">(</span>
</span><span class='line'>                             <span class="o">-</span><span class="n">fh</span>     <span class="o">=&gt;</span> <span class="nv">$stringfh</span><span class="p">,</span>
</span><span class='line'>                             <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$format</span><span class="p">,</span>
</span><span class='line'>                             <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">while</span><span class="p">(</span> <span class="k">my</span> <span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$seqio</span><span class="o">-&gt;</span><span class="n">next_seq</span> <span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="c1"># process each seq</span>
</span><span class='line'>    <span class="k">print</span> <span class="nv">$seq</span><span class="o">-&gt;</span><span class="n">id</span> <span class="o">.</span> <span class="s">&#39; = &#39;</span><span class="o">.</span><span class="nv">$seq</span><span class="o">-&gt;</span><span class="n">seq</span><span class="p">()</span><span class="o">.</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>同样道理，也可以将一个序列对象以一字符串的形式输出。示例（注意open函数内“&lt;”和“&gt;”的区别，“&lt;”用于输入，“&gt;”用于输出）：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">IO::</span><span class="n">String</span><span class="p">;</span>   <span class="c1"># only needed for Perl versions BEFORE 5.8.0</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$string</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$stringfh</span> <span class="o">=</span> <span class="nn">IO::</span><span class="n">String</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="o">\</span><span class="nv">$string</span><span class="p">);</span>                                        <span class="c1"># Use this for Perl BEFORE 5.8.0</span>
</span><span class='line'><span class="nb">open</span><span class="p">(</span><span class="nv">$stringfh</span><span class="p">,</span> <span class="s">&quot;&gt;&quot;</span><span class="p">,</span> <span class="o">\</span><span class="nv">$string</span><span class="p">)</span> <span class="ow">or</span> <span class="nb">die</span> <span class="s">&quot;Could not open string for writing: $!&quot;</span><span class="p">;</span>   <span class="c1"># Use this for Perl AFTER 5.8.0 (inclusive)</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqOut</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                             <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;swiss&#39;</span><span class="p">,</span>
</span><span class='line'>                             <span class="o">-</span><span class="n">fh</span>     <span class="o">=&gt;</span> <span class="nv">$io</span><span class="p">,</span>
</span><span class='line'>                             <span class="p">);</span>
</span><span class='line'><span class="nv">$seqOut</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seq_obj</span><span class="p">);</span>
</span><span class='line'><span class="k">print</span> <span class="nv">$string</span><span class="p">;</span>
</span></code></pre></td></tr></table></div></figure>

<p><a id="And_more_examples..." name="And_more_examples..."></a>
<h2>更多示例</h2>
<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>中的<code>-file</code>参数可以是多个文件。也可以是一段字符串，用于表明从<a title="wp:Pipe (Unix)" href="http://en.wikipedia.org/wiki/Pipe_(Unix)">管道</a>输入的内容。格式是<code>'-file' =&gt; '命令 |'。注意后面的“|”符号。当需要处理很大的压缩文件而又没有足够空间解压缩的时候，这种办法很管用。这里给出一个从gzip压缩的genbank文件中提取序列并转换成新的fasta格式的文件，使用方法如下：</code>
<pre>     gzip2fasta.pl gbpri1.seq.gz Genbank gbpri1.fa</pre>
下面是gzip2fasta.pl的代码：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'><span class="c1"># get command-line arguments, or die with a usage statement</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>    <span class="o">=</span> <span class="s">&quot;gzip2fasta.pl infile informat outfile\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$infile</span>   <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$informat</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outfile</span>  <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># create one SeqIO object to read in, and another to write out</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqin</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                            <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;/usr/local/bin/gunzip -c $infile |&quot;</span><span class="p">,</span>
</span><span class='line'>                            <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$informat</span><span class="p">,</span>
</span><span class='line'>                            <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqout</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                             <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&gt;$outfile&quot;</span><span class="p">,</span>
</span><span class='line'>                             <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Fasta&#39;</span><span class="p">,</span>
</span><span class='line'>                             <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># write each entry in the input to the output file</span>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nv">$seqin</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$seqout</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$inseq</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>当然Bioperl也可一“管道输出”。使用格式是：<code>'-file' =&gt; "| 命令"。这个时候“|”变到了命令的前面。下面举个例子，用Bioperl将一个genbank序列文件转为fasta格式的序列，但不输出文件，直接传递给WashU Blast，让其建立序列数据库。用法：</code>
<pre>    any2wublastable.pl myfile.gb Genbank mywublastable p</pre>
<tt>any2wublastable.pl的代码：</tt></p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># get command-line arguments, or die with a usage statement</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>     <span class="o">=</span> <span class="s">&quot;any2wublastable.pl infile informat outdbname outdbtype\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$infile</span>    <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$informat</span>  <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outdbname</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$outdbtype</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># create one SeqIO object to read in, and another to write out</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqin</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                            <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&lt;$infile&quot;</span><span class="p">,</span>
</span><span class='line'>                            <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="nv">$informat</span><span class="p">,</span>
</span><span class='line'>                            <span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqout</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                             <span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="s">&quot;| /usr/local/bin/xdformat -o $outdbname -${outdbtype} -- -&quot;</span><span class="p">,</span>
</span><span class='line'>                             <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Fasta&#39;</span><span class="p">,</span>
</span><span class='line'>                             <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># write each entry in the input to the output</span>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nv">$seqin</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$seqout</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$inseq</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>可能一些有经验的读者已经意识到new方法返回的是一个引用变量（reference），引用的可以是任意格式的数据。现在来看一个用引用变量存储序列信息的例子。任务是要从一个genbank序列文件中提取出人类的序列并输出到human.gb文件，剩下的序列输出到other.gb文件。这里需要用到两个哈希数组（hash），并分别以“human”和“other”作为下标（key）。
<pre>     splitgb.pl inseq.gb</pre>
<tt>splitgb.pl</tt>的代码:</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># get command-line argument, or die with a usage statement</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>  <span class="o">=</span> <span class="s">&quot;splitgb.pl infile\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$infile</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                            <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&lt;$infile&quot;</span><span class="p">,</span>
</span><span class='line'>                            <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Genbank&#39;</span><span class="p">,</span>
</span><span class='line'>                            <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">%outfiles</span> <span class="o">=</span> <span class="p">(</span>
</span><span class='line'>                <span class="s">&#39;human&#39;</span> <span class="o">=&gt;</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                           <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&#39;&gt;human.gb&#39;</span><span class="p">,</span>
</span><span class='line'>                                           <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Genbank&#39;</span><span class="p">,</span>
</span><span class='line'>                                           <span class="p">),</span>
</span><span class='line'>                <span class="s">&#39;other&#39;</span> <span class="o">=&gt;</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                           <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&#39;&gt;other.gb&#39;</span><span class="p">,</span>
</span><span class='line'>                                           <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Genbank&#39;</span><span class="p">,</span>
</span><span class='line'>                                           <span class="p">),</span>
</span><span class='line'>                <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$seqin</span> <span class="o">=</span> <span class="nv">$inseq</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="c1"># here we make use of the species attribute, which returns a</span>
</span><span class='line'>    <span class="c1"># species object, which has a binomial attribute that</span>
</span><span class='line'>    <span class="c1"># holds the binomial species name of the source of the sequence</span>
</span><span class='line'>    <span class="k">if</span> <span class="p">(</span><span class="nv">$seqin</span><span class="o">-&gt;</span><span class="n">species</span><span class="o">-&gt;</span><span class="n">binomial</span> <span class="o">=~</span> <span class="sr">m/Homo sapiens/</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>        <span class="nv">$outfiles</span><span class="p">{</span><span class="s">&#39;human&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seqin</span><span class="p">);</span>
</span><span class='line'>    <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
</span><span class='line'>        <span class="nv">$outfiles</span><span class="p">{</span><span class="s">&#39;other&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seqin</span><span class="p">);</span>
</span><span class='line'>    <span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>现在来看用更复杂的多维hash来存储Genbank和fasta的序列信息。</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
<span class='line-number'>36</span>
<span class='line-number'>37</span>
<span class='line-number'>38</span>
<span class='line-number'>39</span>
<span class='line-number'>40</span>
<span class='line-number'>41</span>
<span class='line-number'>42</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'><span class="c1"># get command-line argument, or die with a usage statement</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$usage</span>  <span class="o">=</span> <span class="s">&quot;splitgb.pl infile\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$infile</span> <span class="o">=</span> <span class="nb">shift</span> <span class="ow">or</span> <span class="nb">die</span> <span class="nv">$usage</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$inseq</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                            <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&lt;$infile&quot;</span><span class="p">,</span>
</span><span class='line'>                            <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Genbank&#39;</span><span class="p">,</span>
</span><span class='line'>                            <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">%outfiles</span> <span class="o">=</span> <span class="p">(</span>
</span><span class='line'>                <span class="n">human</span> <span class="o">=&gt;</span> <span class="p">{</span>
</span><span class='line'>                          <span class="n">Genbank</span> <span class="o">=&gt;</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&#39;&gt;human.gb&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Genbank&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="p">),</span>
</span><span class='line'>                          <span class="n">Fasta</span>   <span class="o">=&gt;</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&#39;&gt;human.fa&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Fasta&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="p">),</span>
</span><span class='line'>                          <span class="p">},</span>
</span><span class='line'>                <span class="n">other</span> <span class="o">=&gt;</span> <span class="p">{</span>
</span><span class='line'>                          <span class="n">Genbank</span> <span class="o">=&gt;</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&#39;&gt;other.gb&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Genbank&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="p">),</span>
</span><span class='line'>                          <span class="n">Fasta</span>   <span class="o">=&gt;</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="s">&#39;&gt;other.fa&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;Fasta&#39;</span><span class="p">,</span>
</span><span class='line'>                                                     <span class="p">),</span>
</span><span class='line'>                          <span class="p">}</span>
</span><span class='line'>                <span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="k">my</span> <span class="nv">$seqin</span> <span class="o">=</span> <span class="nv">$inseq</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="k">if</span> <span class="p">(</span><span class="nv">$seqin</span><span class="o">-&gt;</span><span class="n">species</span><span class="o">-&gt;</span><span class="n">binomial</span> <span class="o">=~</span> <span class="sr">m/Homo sapiens/</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>        <span class="nv">$outfiles</span><span class="p">{</span><span class="s">&#39;human&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="p">{</span><span class="s">&#39;Genbank&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seqin</span><span class="p">);</span>
</span><span class='line'>        <span class="nv">$outfiles</span><span class="p">{</span><span class="s">&#39;human&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="p">{</span><span class="s">&#39;Fasta&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seqin</span><span class="p">);</span>
</span><span class='line'>    <span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
</span><span class='line'>        <span class="nv">$outfiles</span><span class="p">{</span><span class="s">&#39;other&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="p">{</span><span class="s">&#39;Genbank&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seqin</span><span class="p">);</span>
</span><span class='line'>        <span class="nv">$outfiles</span><span class="p">{</span><span class="s">&#39;other&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="p">{</span><span class="s">&#39;Fasta&#39;</span><span class="p">}</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seqin</span><span class="p">);</span>
</span><span class='line'>    <span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>最后要说的是<a href="http://en.wikipedia.org/wiki/One-liner_program" target="_blank">单行</a>程序。Perl允许在命令行里用单行程序的方式编写脚本，而不需要放在脚本文件中。如下所示，-e后面用单引号括住脚本代码，<tt>-M后面跟所调用的模块。（译者注：-M后面没有空格，-e后面可有可无，多个模块则写多个-M。）当脚本内部也需要使用单引号时必须用q()来代替，防止混淆。下面举的例子是要找出在一个压缩的序列文件中有多少GSS序列。注：为了便于阅读，下面的代码有分行，但在命令行中一般没有写完代码不分行。（译者注：有单引号存在的时候，也可以分行，直到单引号结束再回车才会运行命令。）</tt>
<pre>perl -MBio::SeqIO -e 'my $gss=0;
my $in=Bio::SeqIO-&gt;new(q(-file)=&gt;q(/usr/local/bin/gunzip -c gbpri1.seq.gz |), q(-format)=&gt;q(Genbank));
while (my $seq = $in-&gt;next_seq) { $gss++ if ($seq-&gt;keywords =~ m/GSS/);}
 print "There are $gss GSS sequences in gbpri1.seq.gz\n";'</pre>
<a id="Caveats" name="Caveats"></a>
<h2>警告</h2>
因为<a title="BioPerl" href="http://www.bioperl.org/wiki/BioPerl">BioPerl</a>用一个单一而通用的数据接口来存取所有支持格式的文件，有时候Bioperl就会或多或少改变原本文件的数据结构。例如，从<a title="GenBank" href="http://www.bioperl.org/wiki/GenBank">GenBank</a>数据库中下载到一个genbank格式的序列，然后用bioperl转换成一个新的genbank格式的序列。用&#8221;<tt>diff origfile newfile</tt>&#8220;后会惊奇地发现两者有所不同，这时你就应该明白我第一句说的是什么了。需要记住Bioperl并不提供“往返行程”，它只是通过一个通用的数据接口来获取多种多样序列的信息，并用于其他脚本或程序从中提取所需信息。</p>

<p><a id="Error_Handling" name="Error_Handling"></a>
<h2>纠错</h2>
如果所给定的文件不存在，脚本会运行出错并给出相应的错误信息。这用编程术语叫做“<a href="http://en.wikipedia.org/wiki/Exception_handling" target="_blank">抛出异常</a>”。例如：
<pre>      user@localhost ~/src/bioperl-live&gt; perl t.pl bollocks silly
      ------------- EXCEPTION  -------------
      MSG: Could not open bollocks for reading: No such file or directory
      STACK Bio::Root::IO::_initialize_io Bio/Root/IO.pm:259
      STACK Bio::SeqIO::_initialize Bio/SeqIO.pm:441
      STACK Bio::SeqIO::genbank::_initialize Bio/SeqIO/genbank.pm:122
      STACK Bio::SeqIO::new Bio/SeqIO.pm:359
      STACK Bio::SeqIO::new Bio/SeqIO.pm:372
      STACK toplevel t.pl:9
      --------------------------------------
</pre>

这种错误信息非常有用。因为可以从中可以找出出错的地方，也叫“<a href="http://en.wikipedia.org/wiki/Stack_trace" target="_blank">堆栈追踪</a>”。如上例中给出了出错的文件和与错误相关的行号。（译者注：上例中t.pl是用户自己的脚本，出错是在t.pl文件的第9行。其他都是bioperl包含的一些相关模块文件，所列行号只是和错误相关，但一般不会是bioperl本身出错。）</p>

<p>Bioperl会自动识别出类似的错误并终止程序运行。但最好是自己就在脚本中设置一些纠错机制。 可以通过类似下例来“捕获异常”。</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$input_file</span>  <span class="o">=</span> <span class="nb">shift</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$output_file</span> <span class="o">=</span> <span class="nb">shift</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># we have to declare $seq_in and $seq_out before</span>
</span><span class='line'><span class="c1"># the eval block as we want to use them afterwards</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq_in</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq_out</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="nb">eval</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$seq_in</span>   <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;genbank&#39;</span><span class="p">,</span>
</span><span class='line'>                                <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="nv">$input_file</span><span class="p">,</span>
</span><span class='line'>                                <span class="p">);</span>
</span><span class='line'>
</span><span class='line'>    <span class="nv">$seq_out</span>  <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span>
</span><span class='line'>                                <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;fasta&#39;</span><span class="p">,</span>
</span><span class='line'>                                <span class="o">-</span><span class="n">file</span>   <span class="o">=&gt;</span> <span class="s">&quot;&gt;$output_file&quot;</span><span class="p">,</span>
</span><span class='line'>                                <span class="p">);</span>
</span><span class='line'><span class="p">};</span>
</span><span class='line'><span class="k">if</span><span class="p">(</span> <span class="vg">$@</span> <span class="p">)</span> <span class="p">{</span> <span class="c1"># an error occurred</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;Was not able to open files, sorry!\n&quot;</span><span class="p">;</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;Full error is\n\n$@\n&quot;</span><span class="p">;</span>
</span><span class='line'>    <span class="nb">exit</span><span class="p">(</span><span class="o">-</span><span class="mi">1</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq</span><span class="p">;</span>
</span><span class='line'><span class="k">while</span><span class="p">(</span> <span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$seq_in</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">()</span> <span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$seq_out</span><span class="o">-&gt;</span><span class="n">write_seq</span><span class="p">(</span><span class="nv">$seq</span><span class="p">);</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>eval是perl的一种纠错机制，不止是<a title="BioPerl" href="http://www.bioperl.org/wiki/BioPerl">BioPerl</a>才有。如果eval {BLOCK}中的一段程序出错后，回将错误信息返回到$@变量。需要注意的是：必须提前声明变量 <code>$seq_in<span style="font-family: Georgia, 'Times New Roman', 'Bitstream Charter', Times, serif;">和</span></code><code>$seq_out。如果在eval {}内用my声明变量，此变量只被设定在eval{}中，当在eval{}之后调用此变量是就会出错（用use strict时会提示错误并终止运行）。</code></p>

<p><a id="Speed.2C_Bio::Seq::SeqBuilder" name="Speed.2C_Bio::Seq::SeqBuilder"></a>
<h2>提速, <a title="Module:Bio::Seq::SeqBuilder" href="http://www.bioperl.org/wiki/Module:Bio::Seq::SeqBuilder">Bio::Seq::SeqBuilder</a></h2>
如果需要处理大批量的序列，且只需要从中提取很少部分信息（如只提取genbank文件中的序列），可以使用<a title="Module:Bio::Seq::SeqBuilder" href="http://www.bioperl.org/wiki/Module:Bio::Seq::SeqBuilder">Bio::Seq::SeqBuilder</a>来设定<a title="Module:Bio::SeqIO" href="http://www.bioperl.org/wiki/Module:Bio::SeqIO">Bio::SeqIO</a>只提取需要的部分信息。这样能够节省很多时间从而提升脚本的运行速度。</p>

<p>例如，只需要genbank文件中三类注释信息，可提升大约6倍的速度。</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1">#!/usr/bin/perl</span>
</span><span class='line'>
</span><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="n">Benchmark</span> <span class="sx">qw(:all)</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$file</span> <span class="o">=</span> <span class="s">&quot;gbbct10.seq&quot;</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="n">timethis</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="k">sub </span><span class="p">{</span>
</span><span class='line'>   <span class="k">my</span> <span class="nv">$in</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="nv">$file</span><span class="p">,</span> <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&quot;genbank&quot;</span><span class="p">);</span>
</span><span class='line'>   <span class="k">for</span> <span class="p">(</span><span class="mi">1</span><span class="o">..</span><span class="mi">1000</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>      <span class="k">my</span> <span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$in</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">;</span>
</span><span class='line'>   <span class="p">}</span>
</span><span class='line'><span class="p">});</span>
</span><span class='line'>
</span><span class='line'><span class="n">timethis</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="k">sub </span><span class="p">{</span>
</span><span class='line'>   <span class="k">my</span> <span class="nv">$in</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="nv">$file</span><span class="p">,</span> <span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&quot;genbank&quot;</span><span class="p">);</span>
</span><span class='line'>   <span class="k">my</span> <span class="nv">$builder</span> <span class="o">=</span> <span class="nv">$in</span><span class="o">-&gt;</span><span class="n">sequence_builder</span><span class="p">();</span>
</span><span class='line'>   <span class="nv">$builder</span><span class="o">-&gt;</span><span class="n">want_none</span><span class="p">();</span>
</span><span class='line'>   <span class="nv">$builder</span><span class="o">-&gt;</span><span class="n">add_wanted_slot</span><span class="p">(</span><span class="s">&#39;display_id&#39;</span><span class="p">,</span><span class="s">&#39;desc&#39;</span><span class="p">,</span><span class="s">&#39;seq&#39;</span><span class="p">);</span>
</span><span class='line'>   <span class="k">for</span> <span class="p">(</span><span class="mi">1</span><span class="o">..</span><span class="mi">1000</span><span class="p">)</span> <span class="p">{</span>
</span><span class='line'>      <span class="k">my</span> <span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$in</span><span class="o">-&gt;</span><span class="n">next_seq</span><span class="p">;</span>
</span><span class='line'>   <span class="p">}</span>
</span><span class='line'><span class="p">});</span>
</span></code></pre></td></tr></table></div></figure>

<p>结果：
<pre>timethis 1: 10 wallclock secs ( 9.64 usr +  0.02 sys =  9.66 CPU) @  0.10/s (n=1)
            (warning: too few iterations for a reliable count)
timethis 1:  1 wallclock secs ( 1.63 usr +  0.00 sys =  1.63 CPU) @  0.61/s (n=1)
            (warning: too few iterations for a reliable count)</pre>
更多相关内容见<a title="HOWTO:Feature-Annotation" href="http://www.bioperl.org/wiki/HOWTO:Feature-Annotation#Customizing_Sequence_Object_Construction">HOWTO:Feature-Annotation</a>
<div id="catlinks"></div></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/03/bioperl-howto-beginners-else/">Bioperl入门其他部分（Bioperl HOWTO翻译12）</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-03-19T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>19</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/03/bioperl-howto-beginners-else/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>(译者注：<a href="http://www.bioperl.org/wiki/HOWTO:Beginners" target="_blank">Beginners HOWTO</a>翻译到此结束，从<a href="http://www.bioperl.org/wiki/HOWTO:Beginners#Searching_for_genes_in_genomic_DNA" target="_blank">Searching for genes in genomic DNA</a>部分往后都是一些有关bioperl和其他程序接口：调用其他程序并分析结果。专一性比较强，且需要对各个软件有很好的了解并正确安装和设定。个人觉得不怎么值得翻译，对此有兴趣的童鞋应该可以看懂英文然后自己研究清楚的。</p>

<p>后面的HOWTO都是一个个专题，会将同一个HOWTO放在单篇blog里，大部分HOWTO内容都比较长，翻译的时间可能也比较长，先发布全英文的，然后再慢慢翻译成中文，请见谅。)</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/03/bioperl-howto-index-for-fast-etrieval/">Bioperl快速检索序列（Bioperl HOWTO翻译11）</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-03-03T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>3</span><span class='date-suffix'>rd</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/03/bioperl-howto-index-for-fast-etrieval/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><h3>索引序列文件以用于快速检索</h3>
<a href="http://www.bioperl.org/wiki/HOWTO:Beginners#Indexing_for_Fast_Retrieval" target="_blank">英文原文</a></p>

<p>索引序列是Bioperl一个非常出色的功能。你可以将需要处理的本地序列文件建立一个索引然后利用这个索引提取你所需要的序列文件。这样做最重要的原因是——速度！从有索引的大量本地序列中提取所需序列要比用<a title="http://search.cpan.org/perldoc?Bio::DB::GenBank" href="http://search.cpan.org/perldoc?Bio::DB::GenBank" rel="nofollow" target="_blank">Bio::DB::GenBank</a>从远程数据库中提取快得多。也比用SeqIO快，因为SeqIO要一个一个通读每一个序列，然后才能找到所需的。灵活性是另一个原因。自己的有些序列毕竟不同于公共数据库上的序列，将自己收集的序列建立索引可用于日后快速的提取。</p>

<p>需要说明，在提取序列时的唯一要求是，所提供的要提取序列的id必须具有唯一性。（译者注：如果要提取所有相同有某特征序列，比如想要refseq，即id中包含“_”的序列，可以灵活变通：先输出所有id，然后选取那些有“_”的id，然后再用这些id提取序列信息。）</p>

<p>Bioperl有多个用于索引序列的模块，如 Bio::Index::*类和<a title="http://search.cpan.org/perldoc?Bio::DB::Fasta" href="http://search.cpan.org/perldoc?Bio::DB::Fasta" rel="nofollow" target="_blank">Bio::DB::Fasta</a>。 这些模块的原理都是一样的，先索引一个或多个序列文件，然后再根据id提取序列。先以<a href="http://www.bioperl.org/wiki/Module:Bio::Index::Fasta" target="_blank">Bio::Index::Fasta</a>为例。使用此模块时要先设定一个终端环境变量，如下：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::Index::</span><span class="n">Fasta</span><span class="p">;</span>
</span><span class='line'><span class="nv">$ENV</span><span class="p">{</span><span class="n">BIOPERL_INDEX_TYPE</span><span class="p">}</span> <span class="o">=</span> <span class="s">&quot;SDBM_File&quot;</span><span class="p">;</span>
</span></code></pre></td></tr></table></div></figure>

<p>可见在Perl脚本中也可以设定终端环境变量，此变量存贮在叫%ENV的特殊hash中。此设置当然也可以在终端中设定，在tcsh或csh中如下:
<pre>&gt;setenv BIOPERL_INDEX_TYPE SDBM_File</pre>

在bash shell中设定如下：

<pre>&gt;export BIOPERL_INDEX_TYPE=SDBM_File</pre>

BIOPERL_INDEX_TYPE是环境变量名， SDBM_File是变量的值。（译者注：SDBM是perl使用的一种文件格式，用于将hash保存在一个SDBM文件中。见<a href="http://search.cpan.org/dist/perl-5.12.4/ext/SDBM_File/SDBM_File.pm" target="_blank">这里</a>）. BIOPERL_INDEX指定索引文件所在的目录，可以通过设定此变量建立多个索引文件。（译者注：可以设定“setenv BIOPERL_INDEX /nfs/datadisk/bioperlindex/”，<a href="http://www.bioperl.org/wiki/Module:Bio::Index::Fasta" target="_blank">详见</a>）</p>

<p>看例子：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="nv">$ENV</span><span class="p">{</span><span class="n">BIOPERL_INDEX_TYPE</span><span class="p">}</span> <span class="o">=</span> <span class="s">&quot;SDBM_File&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::Index::</span><span class="n">Fasta</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="nv">$file_name</span> <span class="o">=</span> <span class="s">&quot;sequence.fa&quot;</span><span class="p">;</span>
</span><span class='line'><span class="nv">$id</span> <span class="o">=</span> <span class="s">&quot;48882&quot;</span><span class="p">;</span>
</span><span class='line'><span class="nv">$inx</span> <span class="o">=</span> <span class="nn">Bio::Index::</span><span class="n">Fasta</span><span class="o">-&gt;</span><span class="k">new</span> <span class="p">(</span><span class="o">-</span><span class="n">filename</span> <span class="o">=&gt;</span> <span class="nv">$file_name</span> <span class="o">.</span> <span class="s">&quot;.idx&quot;</span><span class="p">,</span> <span class="o">-</span><span class="n">write_flag</span> <span class="o">=&gt;</span> <span class="mi">1</span><span class="p">);</span>
</span><span class='line'><span class="nv">$inx</span><span class="o">-&gt;</span><span class="n">make_index</span><span class="p">(</span><span class="nv">$file_name</span><span class="p">);</span>
</span></code></pre></td></tr></table></div></figure>

<p>在sequence.fa所在的目录下运行此脚本，将会在该目录下自动生成一个equence.fa.idx索引文件，然后可以通过下一语句提取所需序列：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="nv">$seq_obj</span> <span class="o">=</span> <span class="nv">$inx</span><span class="o">-&gt;</span><span class="n">fetch</span><span class="p">(</span><span class="nv">$id</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure>

<p>默认以 &gt;符号来识别fasta格式的序列，如果是要提取“48882”为id的序列，该序列头信息应该是类似下面的格式：
<pre>&gt;48882 pdb|1CRA</pre>
但是，如果想按照其他字符串提取序列，比如上例中“1CRA”，可以通过id_parser改变索引id的格式，（将自定义的index格式通过一个子函数传递给id_parser），详见<a title="http://search.cpan.org/perldoc?Bio::Index::Fasta" href="http://search.cpan.org/perldoc?Bio::Index::Fasta" rel="nofollow" target="_blank">Bio::Index::Fasta</a>。例如：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="nv">$inx</span><span class="o">-&gt;</span><span class="n">id_parser</span><span class="p">(</span><span class="o">\&amp;</span><span class="n">get_id</span><span class="p">);</span>
</span><span class='line'><span class="nv">$inx</span><span class="o">-&gt;</span><span class="n">make_index</span><span class="p">(</span><span class="nv">$file_name</span><span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">sub </span><span class="nf">get_id</span> <span class="p">{</span>
</span><span class='line'>    <span class="nv">$header</span> <span class="o">=</span> <span class="nb">shift</span><span class="p">;</span>
</span><span class='line'>    <span class="nv">$header</span> <span class="o">=~</span><span class="sr"> /pdb\|(\S+)/</span><span class="p">;</span>
</span><span class='line'>    <span class="nv">$1</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>准确地说，id_parser接受的是一个子函数的引用变量（a reference to a function）。</p>

<p>再来看<a title="http://search.cpan.org/perldoc?Bio::DB::Fasta" href="http://search.cpan.org/perldoc?Bio::DB::Fasta" rel="nofollow" target="_blank">Bio::DB::Fasta</a>，相对<a title="http://search.cpan.org/perldoc?Bio::Index::Fasta" href="http://search.cpan.org/perldoc?Bio::Index::Fasta" rel="nofollow" target="_blank">Bio::Index::Fasta</a>有些更好的功能。主要是：可以处理大批量序列 ；可以很方便地提取序列片段。示例：</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="nn">Bio::DB::</span><span class="n">Fasta</span><span class="p">;</span>
</span><span class='line'><span class="p">(</span><span class="nv">$file</span><span class="p">,</span><span class="nv">$id</span><span class="p">,</span><span class="nv">$start</span><span class="p">,</span><span class="nv">$end</span><span class="p">)</span> <span class="o">=</span> <span class="p">(</span><span class="s">&quot;genome.fa&quot;</span><span class="p">,</span><span class="s">&quot;CHROMOSOME_I&quot;</span><span class="p">,</span><span class="mi">11250</span><span class="p">,</span><span class="mi">11333</span><span class="p">);</span>
</span><span class='line'><span class="nv">$db</span> <span class="o">=</span> <span class="nn">Bio::DB::</span><span class="n">Fasta</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="nv">$file</span><span class="p">);</span>
</span><span class='line'><span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$db</span><span class="o">-&gt;</span><span class="n">seq</span><span class="p">(</span><span class="nv">$id</span><span class="p">,</span><span class="nv">$start</span><span class="p">,</span><span class="nv">$end</span><span class="p">);</span>
</span><span class='line'><span class="k">print</span> <span class="nv">$seq</span><span class="p">,</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span></code></pre></td></tr></table></div></figure>

<p>此脚本将genome.fa建立索引，然后提取CHROMOSOME_I上从11250到11333的序列片段。也可以如<a title="http://search.cpan.org/perldoc?Bio::Index::Fasta" href="http://search.cpan.org/perldoc?Bio::Index::Fasta" rel="nofollow" target="_blank">Bio::Index::Fasta</a>一样自定义要索引的id。（译者注：在new()中设定-makeid参数，此参数使用方法和上面的id_parser一样。详见<a title="http://search.cpan.org/perldoc?Bio::DB::Fasta" href="http://search.cpan.org/perldoc?Bio::DB::Fasta" rel="nofollow" target="_blank">Bio::DB::Fasta</a>）</p>

<p>更多的信息可见<a title="HOWTO:Local Databases" href="http://www.bioperl.org/wiki/HOWTO:Local_Databases" target="_blank">HOWTO:Local_Databases</a></p>
</div>
  
  


    </article>
  
  <div class="pagination">
    
      <a class="prev" href="/page5">&larr; Older</a>
    
    <a href="/blog/archives">Blog Archives</a>
    
    <a class="next" href="/page3">Newer &rarr;</a>
    
  </div>
</div>
<aside class="sidebar">
  
    
  
</aside>

    </div>
  </div>
  <footer role="contentinfo"><!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>

<p>
  Copyright &copy; 2016 - Bioops -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a> | Themed with <a href="https://github.com/lucaslew/whitespace">Whitespace</a></span>
</p>


</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'bioopsgithub';
      
        
        var disqus_script = 'count.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>






<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
