
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Bioops</title>
  <meta name="author" content="Bioops">

  
  <meta name="description" content="R code demo of Linear discriminant analysis (LDA) Quadratic discriminant analysis (QDA) k-nearest neighbor (KNN) 1
2
3
4
5
6
7
8
9
10
11
12
13
14
15 &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bioops.info">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="Bioops" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="/javascripts/libs/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Fjalla+One" rel="stylesheet" type="text/css">
<!--- MathJax Configuration -->
<script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-21431888-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   class="collapse-sidebar sidebar-footer" >
  <header role="banner"><hgroup>
  <h1><a href="/">Bioops</a></h1>
  
    <h2>Bioinformatics=(ACGAAG->AK)+(#!/bin/sh)+(P(A|B)=P(B|A)*P(A)/P(B))</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscribe" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 13.310204,73.332654 C 5.967347,73.332654 0,79.322448 0,86.621428 c 0,7.338776 5.967347,13.262246 13.310204,13.262246 7.370408,0 13.328572,-5.92245 13.328572,-13.262246 0,-7.29898 -5.958164,-13.288774 -13.328572,-13.288774 z M 0.01530612,33.978572 V 53.143878 C 12.493878,53.143878 24.229592,58.02347 33.068368,66.865306 41.894898,75.685714 46.767346,87.47449 46.767346,100 h 19.25 C 66.017346,63.592858 36.4,33.979592 0.01530612,33.978572 l 0,0 z M 0.03877552,0 V 19.17449 C 44.54796,19.17551 80.77551,55.437756 80.77551,100 H 100 C 100,44.87653 55.15102,0 0.03877552,0 z"></path></svg></a></li>
  
</ul>
  
  
  
  
  
<ul class="subscribe">
  <li><a href="https://github.com/bioops" rel="subscribe-github" title="@bioops on GitHub"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 50,0 C 22.385714,0 0,22.385714 0,50 0,77.614286 22.385714,100 50,100 77.614286,100 100,77.614286 100,50 100,22.385714 77.614286,0 50,0 z m 29.692858,79.692858 c -3.859184,3.859182 -8.351022,6.887754 -13.35,9.00306 -1.27041,0.536736 -2.560204,1.009184 -3.867348,1.415306 v -7.493878 c 0,-3.938774 -1.35102,-6.835714 -4.053062,-8.690816 1.692858,-0.163264 3.24694,-0.390816 4.663266,-0.683672 1.416326,-0.292858 2.913266,-0.716328 4.491838,-1.27041 1.57857,-0.55408 2.994896,-1.213264 4.247958,-1.97755 1.253062,-0.765306 2.458164,-1.758164 3.613266,-2.978572 1.155102,-1.220408 2.12449,-2.604082 2.905102,-4.15 0.780612,-1.545918 1.4,-3.40204 1.855102,-5.566326 0.455102,-2.164286 0.683674,-4.54898 0.683674,-7.153062 0,-5.045918 -1.643878,-9.341836 -4.931634,-12.890816 C 77.44796,33.35 77.285714,29.10204 75.463266,24.512244 l -1.22143,-0.145918 c -0.845918,-0.09796 -2.368366,0.260204 -4.565306,1.07449 -2.196938,0.814286 -4.663264,2.14796 -7.396938,4.004082 -3.87449,-1.07449 -7.893878,-1.611224 -12.061224,-1.611224 -4.19898,0 -8.203062,0.536734 -12.012246,1.611224 -1.72449,-1.17245 -3.361224,-2.139796 -4.907142,-2.905102 C 31.753062,25.77449 30.516326,25.254082 29.587756,24.97653 28.660204,24.7 27.79796,24.528572 27,24.463266 c -0.79796,-0.0653 -1.310204,-0.08062 -1.537756,-0.04898 -0.22755,0.03164 -0.390816,0.0653 -0.487754,0.09796 -1.82347,4.62245 -1.985714,8.87143 -0.487756,12.743878 -3.287754,3.54796 -4.931632,7.844898 -4.931632,12.890816 0,2.604082 0.227552,4.988776 0.683674,7.153062 0.456122,2.164286 1.07449,4.020408 1.855102,5.566326 0.780612,1.545918 1.75,2.929592 2.905102,4.15 1.155102,1.220408 2.360204,2.213266 3.613264,2.978572 1.253062,0.766326 2.669388,1.42449 4.24796,1.97755 1.578572,0.554082 3.07551,0.976532 4.491836,1.27041 1.416328,0.292856 2.970408,0.521428 4.663266,0.683672 -2.669388,1.82347 -4.004082,4.720408 -4.004082,8.690816 v 7.639796 C 36.536734,89.818368 35.083674,89.3 33.656122,88.695918 c -4.99898,-2.115306 -9.490816,-5.143878 -13.35,-9.00306 -3.859184,-3.859184 -6.887754,-8.351022 -9.00306,-13.35 C 9.1163263,61.171428 8.0071428,55.67347 8.0071428,50 c 0,-5.67347 1.1091835,-11.171428 3.2969392,-16.342858 2.115306,-4.998978 5.143878,-9.490816 9.00306,-13.35 3.859184,-3.859182 8.351022,-6.887754 13.35,-9.00306 C 38.828572,9.1163266 44.32653,8.0071428 50,8.0071428 c 5.67347,0 11.171428,1.1091838 16.342858,3.2969392 5,2.115306 9.490816,5.143878 13.35,9.00306 3.859182,3.859184 6.887754,8.351022 9.00306,13.35 2.186736,5.17245 3.295918,10.67041 3.295918,16.342858 0,5.672448 -1.109182,11.171428 -3.296938,16.342858 -2.115306,4.998978 -5.143878,9.490816 -9.00204,13.35 l 0,0 z"></path></svg></a></li>
</ul>
  
  
  
  
  
  
  
    
      <form action="http://google.com/search" method="get">
        <fieldset role="search">
          <input type="hidden" name="sitesearch" value="bioops.info" />
    
          <input class="search" type="text" name="q" results="0" placeholder="Search"/>
        </fieldset>
      </form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div class="blog-index">
  
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/02/demo-of-classification/">Demo of Classification</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-02-24T17:35:23+00:00'><span class='date'><span class='date-month'>Feb</span> <span class='date-day'>24</span><span class='date-suffix'>th</span>, <span class='date-year'>2015</span></span> <span class='time'>5:35 pm</span></time>
        
         | <a href="/2015/02/demo-of-classification/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>R code demo of</p>

<ol>
  <li>Linear discriminant analysis (LDA)</li>
  <li>Quadratic discriminant analysis (QDA)</li>
  <li>k-nearest neighbor (KNN)</li>
</ol>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># read data</span>
</span><span class="line"><span class="kn">library</span><span class="p">(</span>RCurl<span class="p">)</span>
</span><span class="line">data <span class="o">&lt;-</span> getURL<span class="p">(</span><span class="s">&quot;https://raw.githubusercontent.com/bioops/mis_scripts/master/statistics/data/admission.txt&quot;</span><span class="p">)</span>
</span><span class="line">admission<span class="o">&lt;-</span>read.table<span class="p">(</span>text<span class="o">=</span>data<span class="p">,</span>header<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</span><span class="line">admission<span class="o">$</span>CLASS<span class="o">&lt;-</span><span class="kp">as.factor</span><span class="p">(</span>admission<span class="o">$</span>CLASS<span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># scale the data</span>
</span><span class="line">admission<span class="o">$</span>GMAT<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>admission<span class="o">$</span>GMAT<span class="p">)</span>
</span><span class="line">admission<span class="o">$</span>GPA<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>admission<span class="o">$</span>GPA<span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># (1) LDA</span>
</span><span class="line"><span class="kn">library</span><span class="p">(</span>MASS<span class="p">)</span>
</span><span class="line">lda.fit<span class="o">&lt;-</span>lda<span class="p">(</span>CLASS<span class="o">~</span>GMAT<span class="o">+</span>GPA<span class="p">,</span>data<span class="o">=</span>admission<span class="p">)</span>
</span><span class="line">lda.pred<span class="o">&lt;-</span>predict<span class="p">(</span>lda.fit<span class="p">)</span>
</span><span class="line"><span class="c1"># plot</span>
</span><span class="line"><span class="c1"># different symbols represents true classifications</span>
</span><span class="line"><span class="c1"># different colors are predicted classifications</span>
</span><span class="line">plot<span class="p">(</span>admission<span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span>col<span class="o">=</span>lda.pred<span class="o">$</span><span class="kp">class</span><span class="p">,</span> pch<span class="o">=</span><span class="kp">as.numeric</span><span class="p">(</span>admission<span class="o">$</span>CLASS<span class="p">),</span>
</span><span class="line">     xlab<span class="o">=</span><span class="s">&quot;GPA&quot;</span><span class="p">,</span>ylab<span class="o">=</span><span class="s">&quot;GMAT&quot;</span><span class="p">,</span>main<span class="o">=</span><span class="s">&quot;LDA&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><img src="http://bioops.info/images/uploads/2015/class1.png" alt="LDA" /></p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># (2) QDA</span>
</span><span class="line">qda.fit<span class="o">&lt;-</span>qda<span class="p">(</span>CLASS<span class="o">~</span>GMAT<span class="o">+</span>GPA<span class="p">,</span>data<span class="o">=</span>admission<span class="p">)</span>
</span><span class="line">qda.pred<span class="o">&lt;-</span>predict<span class="p">(</span>qda.fit<span class="p">)</span>
</span><span class="line"><span class="c1"># plot</span>
</span><span class="line"><span class="c1"># different symbols represents true classifications</span>
</span><span class="line"><span class="c1"># different colors are predicted classifications</span>
</span><span class="line">plot<span class="p">(</span>admission<span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span>col<span class="o">=</span>qda.pred<span class="o">$</span><span class="kp">class</span><span class="p">,</span> pch<span class="o">=</span><span class="kp">as.numeric</span><span class="p">(</span>admission<span class="o">$</span>CLASS<span class="p">),</span>
</span><span class="line">     xlab<span class="o">=</span><span class="s">&quot;GPA&quot;</span><span class="p">,</span>ylab<span class="o">=</span><span class="s">&quot;GMAT&quot;</span><span class="p">,</span>main<span class="o">=</span><span class="s">&quot;QDA&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><img src="http://bioops.info/images/uploads/2015/class2.png" alt="QDA" /></p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="kn">library</span><span class="p">(</span><span class="kp">class</span><span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># tune parameter using cross-validation (CV)</span>
</span><span class="line">k<span class="o">&lt;-</span><span class="kp">seq</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">15</span><span class="p">,</span><span class="m">1</span><span class="p">)</span> <span class="c1"># different k</span>
</span><span class="line">cv.err<span class="o">&lt;-</span><span class="kc">NULL</span> <span class="c1"># cv error</span>
</span><span class="line"><span class="kr">for</span> <span class="p">(</span>ki <span class="kr">in</span> k<span class="p">){</span>
</span><span class="line">  knn.pred.cv<span class="o">&lt;-</span>knn.cv<span class="p">(</span>admission<span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span>admission<span class="p">[,</span><span class="m">3</span><span class="p">],</span>k<span class="o">=</span>ki<span class="p">)</span>
</span><span class="line">  cv.err<span class="o">&lt;-</span><span class="kt">c</span><span class="p">(</span>cv.err<span class="p">,</span> <span class="kp">mean</span><span class="p">(</span>knn.pred.cv<span class="o">!=</span>admission<span class="p">[,</span><span class="m">3</span><span class="p">]))</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">plot<span class="p">(</span>k<span class="p">,</span>cv.err<span class="p">,</span> main<span class="o">=</span><span class="s">&quot;CV error vs k&quot;</span><span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># using the optimal parameter</span>
</span><span class="line">knn.pred<span class="o">&lt;-</span>knn<span class="p">(</span>admission<span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span>admission<span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span> admission<span class="p">[,</span><span class="m">3</span><span class="p">],</span>k<span class="o">=</span>k<span class="p">[</span><span class="kp">which.min</span><span class="p">(</span>cv.err<span class="p">)])</span>
</span><span class="line">
</span><span class="line"><span class="c1"># plot</span>
</span><span class="line"><span class="c1"># different symbols represents true classifications</span>
</span><span class="line"><span class="c1"># different colors are predicted classifications</span>
</span><span class="line">plot<span class="p">(</span>admission<span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span>col<span class="o">=</span>knn.pred<span class="p">,</span> pch<span class="o">=</span><span class="kp">as.numeric</span><span class="p">(</span>admission<span class="o">$</span>CLASS<span class="p">),</span>
</span><span class="line">     xlab<span class="o">=</span><span class="s">&quot;GPA&quot;</span><span class="p">,</span>ylab<span class="o">=</span><span class="s">&quot;GMAT&quot;</span><span class="p">,</span>main<span class="o">=</span><span class="s">&quot;KNN&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><img src="http://bioops.info/images/uploads/2015/class3.png" alt="KNN" /></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/02/solving-bridge-regression-using-local-quadratic-approximation-lqa/">Solving Bridge Regression Using Local Quadratic Approximation (LQA)</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-02-24T16:27:50+00:00'><span class='date'><span class='date-month'>Feb</span> <span class='date-day'>24</span><span class='date-suffix'>th</span>, <span class='date-year'>2015</span></span> <span class='time'>4:27 pm</span></time>
        
         | <a href="/2015/02/solving-bridge-regression-using-local-quadratic-approximation-lqa/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>Bridge regression is a broad class of penalized regression, and can be used in high-dimensional regression problems.</p>

<p>It includes the ridge (q=2) and lasso (q  =1) as special cases.</p>

<p>More technical details can be found <a href="http://www.sciencedirect.com/science/article/pii/S0378375811001960">here</a>. Below R code demonstrates:</p>

<ol>
  <li>sovling bridge regression using local quadratic approximation (LQA) and Newton–Raphson algorithm.</li>
  <li>simulation of tuning parameters using 50/50/200 observations (training/validation/testing).</li>
</ol>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
<span class="line-number">27</span>
<span class="line-number">28</span>
<span class="line-number">29</span>
<span class="line-number">30</span>
<span class="line-number">31</span>
<span class="line-number">32</span>
<span class="line-number">33</span>
<span class="line-number">34</span>
<span class="line-number">35</span>
<span class="line-number">36</span>
<span class="line-number">37</span>
<span class="line-number">38</span>
<span class="line-number">39</span>
<span class="line-number">40</span>
<span class="line-number">41</span>
<span class="line-number">42</span>
<span class="line-number">43</span>
<span class="line-number">44</span>
<span class="line-number">45</span>
<span class="line-number">46</span>
<span class="line-number">47</span>
<span class="line-number">48</span>
<span class="line-number">49</span>
<span class="line-number">50</span>
<span class="line-number">51</span>
<span class="line-number">52</span>
<span class="line-number">53</span>
<span class="line-number">54</span>
<span class="line-number">55</span>
<span class="line-number">56</span>
<span class="line-number">57</span>
<span class="line-number">58</span>
<span class="line-number">59</span>
<span class="line-number">60</span>
<span class="line-number">61</span>
<span class="line-number">62</span>
<span class="line-number">63</span>
<span class="line-number">64</span>
<span class="line-number">65</span>
<span class="line-number">66</span>
<span class="line-number">67</span>
<span class="line-number">68</span>
<span class="line-number">69</span>
<span class="line-number">70</span>
<span class="line-number">71</span>
<span class="line-number">72</span>
<span class="line-number">73</span>
<span class="line-number">74</span>
<span class="line-number">75</span>
<span class="line-number">76</span>
<span class="line-number">77</span>
<span class="line-number">78</span>
<span class="line-number">79</span>
<span class="line-number">80</span>
<span class="line-number">81</span>
<span class="line-number">82</span>
<span class="line-number">83</span>
<span class="line-number">84</span>
<span class="line-number">85</span>
<span class="line-number">86</span>
<span class="line-number">87</span>
<span class="line-number">88</span>
<span class="line-number">89</span>
<span class="line-number">90</span>
<span class="line-number">91</span>
<span class="line-number">92</span>
<span class="line-number">93</span>
<span class="line-number">94</span>
<span class="line-number">95</span>
<span class="line-number">96</span>
<span class="line-number">97</span>
<span class="line-number">98</span>
<span class="line-number">99</span>
<span class="line-number">100</span>
<span class="line-number">101</span>
<span class="line-number">102</span>
<span class="line-number">103</span>
<span class="line-number">104</span>
<span class="line-number">105</span>
<span class="line-number">106</span>
<span class="line-number">107</span>
<span class="line-number">108</span>
<span class="line-number">109</span>
<span class="line-number">110</span>
<span class="line-number">111</span>
<span class="line-number">112</span>
<span class="line-number">113</span>
<span class="line-number">114</span>
<span class="line-number">115</span>
<span class="line-number">116</span>
<span class="line-number">117</span>
<span class="line-number">118</span>
<span class="line-number">119</span>
<span class="line-number">120</span>
<span class="line-number">121</span>
<span class="line-number">122</span>
<span class="line-number">123</span>
<span class="line-number">124</span>
<span class="line-number">125</span>
<span class="line-number">126</span>
<span class="line-number">127</span>
<span class="line-number">128</span>
<span class="line-number">129</span>
<span class="line-number">130</span>
<span class="line-number">131</span>
<span class="line-number">132</span>
<span class="line-number">133</span>
<span class="line-number">134</span>
<span class="line-number">135</span>
<span class="line-number">136</span>
<span class="line-number">137</span>
<span class="line-number">138</span>
<span class="line-number">139</span>
<span class="line-number">140</span>
<span class="line-number">141</span>
<span class="line-number">142</span>
<span class="line-number">143</span>
<span class="line-number">144</span>
<span class="line-number">145</span>
<span class="line-number">146</span>
<span class="line-number">147</span>
<span class="line-number">148</span>
<span class="line-number">149</span>
<span class="line-number">150</span>
<span class="line-number">151</span>
<span class="line-number">152</span>
<span class="line-number">153</span>
<span class="line-number">154</span>
<span class="line-number">155</span>
<span class="line-number">156</span>
<span class="line-number">157</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># The function to solve bridge regression</span>
</span><span class="line">bridge<span class="o">&lt;-</span><span class="kr">function</span><span class="p">(</span>x<span class="p">,</span> y<span class="p">,</span> lambda<span class="p">,</span> q<span class="o">=</span><span class="m">1</span><span class="p">,</span> eta<span class="o">=</span><span class="m">0.001</span><span class="p">){</span>
</span><span class="line">  <span class="kn">library</span><span class="p">(</span>glmnet<span class="p">)</span>
</span><span class="line">  <span class="c1"># use ridge coefficients as a starting value</span>
</span><span class="line">  beta.start<span class="o">&lt;-</span>coef<span class="p">(</span>glmnet<span class="p">(</span>x<span class="p">,</span> y<span class="p">,</span> alpha<span class="o">=</span><span class="m">0</span><span class="p">,</span> lambda<span class="o">=</span>lambda<span class="p">,</span> intercept<span class="o">=</span><span class="bp">F</span><span class="p">))</span>
</span><span class="line">  beta.mat<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span>ncol<span class="o">=</span><span class="kp">ncol</span><span class="p">(</span>beta.start<span class="p">),</span>nrow<span class="o">=</span><span class="kp">nrow</span><span class="p">(</span>beta.start<span class="p">)</span><span class="m">-1</span><span class="p">)</span>
</span><span class="line">
</span><span class="line">  <span class="c1"># take each lambda value in the grid</span>
</span><span class="line">  <span class="kr">for</span><span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="kp">length</span><span class="p">(</span>grid<span class="p">)){</span>
</span><span class="line">    <span class="c1"># initial beta without intercept</span>
</span><span class="line">    beta_prev<span class="o">&lt;-</span><span class="kp">as.vector</span><span class="p">(</span>beta.start<span class="p">[</span><span class="m">-1</span><span class="p">,</span>i<span class="p">])</span>
</span><span class="line">    <span class="c1"># initial converge</span>
</span><span class="line">    converge<span class="o">&lt;-</span><span class="m">10</span><span class="o">^</span><span class="m">10</span>
</span><span class="line">
</span><span class="line">    <span class="c1"># iteration until converge or two many iterations</span>
</span><span class="line">    iteration<span class="o">&lt;-</span><span class="m">0</span>
</span><span class="line">    <span class="kr">while</span><span class="p">(</span>converge<span class="o">&gt;</span>eta <span class="o">&amp;&amp;</span> iteration<span class="o">&lt;=</span><span class="m">100</span><span class="p">){</span>
</span><span class="line">      iteration<span class="o">&lt;-</span>iteration<span class="m">+1</span>
</span><span class="line">      <span class="c1"># judge whether some beta is small enough</span>
</span><span class="line">      del<span class="o">&lt;-</span><span class="kp">which</span><span class="p">(</span><span class="kp">abs</span><span class="p">(</span>beta_prev<span class="p">)</span><span class="o">&lt;</span>eta<span class="p">)</span>
</span><span class="line">
</span><span class="line">      <span class="c1"># if all coefficient are small enough then stop the iteration</span>
</span><span class="line">      <span class="kr">if</span><span class="p">(</span><span class="kp">length</span><span class="p">(</span>del<span class="p">)</span><span class="o">==</span><span class="kp">length</span><span class="p">(</span>beta_prev<span class="p">)){</span>
</span><span class="line">        beta_prev<span class="p">[</span>del<span class="p">]</span><span class="o">&lt;-</span><span class="m">0</span>
</span><span class="line">        converge<span class="o">&lt;-</span><span class="m">0</span>
</span><span class="line">
</span><span class="line">        <span class="c1"># else if we need to remove some but not all of the coefficients  </span>
</span><span class="line">      <span class="p">}</span><span class="kr">else</span> <span class="kr">if</span><span class="p">(</span><span class="kp">length</span><span class="p">(</span>del<span class="p">)</span><span class="o">&gt;</span><span class="m">0</span><span class="p">){</span>
</span><span class="line">        <span class="c1"># set these beta to 0</span>
</span><span class="line">        beta_prev<span class="p">[</span>del<span class="p">]</span><span class="o">&lt;-</span><span class="m">0</span>
</span><span class="line">
</span><span class="line">        <span class="c1"># update design matrix x</span>
</span><span class="line">        x.new<span class="o">&lt;-</span>x<span class="p">;</span>x.new<span class="o">&lt;-</span>x.new<span class="p">[,</span><span class="o">-</span>del<span class="p">]</span>
</span><span class="line">
</span><span class="line">        <span class="c1"># calculate the diagonal matrix involving penalty terms</span>
</span><span class="line">        <span class="c1"># and the next beta</span>
</span><span class="line">        <span class="kr">if</span><span class="p">(</span><span class="kp">length</span><span class="p">(</span>beta_prev<span class="p">)</span><span class="o">-</span><span class="kp">length</span><span class="p">(</span>del<span class="p">)</span><span class="o">==</span><span class="m">1</span><span class="p">){</span>
</span><span class="line">          <span class="kp">diag</span><span class="o">&lt;-</span>grid<span class="p">[</span>i<span class="p">]</span><span class="o">*</span><span class="kp">q</span><span class="o">*</span><span class="p">(</span><span class="kp">abs</span><span class="p">(</span>beta_prev<span class="p">[</span><span class="o">-</span>del<span class="p">])</span><span class="o">^</span><span class="p">(</span><span class="kp">q</span><span class="m">-2</span><span class="p">))</span><span class="o">/</span><span class="m">2</span>
</span><span class="line">        <span class="p">}</span><span class="kp">else</span><span class="p">{</span>
</span><span class="line">          <span class="kp">diag</span><span class="o">&lt;-</span><span class="kp">diag</span><span class="p">(</span>grid<span class="p">[</span>i<span class="p">]</span><span class="o">*</span><span class="kp">q</span><span class="o">*</span><span class="p">(</span><span class="kp">abs</span><span class="p">(</span>beta_prev<span class="p">[</span><span class="o">-</span>del<span class="p">])</span><span class="o">^</span><span class="p">(</span><span class="kp">q</span><span class="m">-2</span><span class="p">))</span><span class="o">/</span><span class="m">2</span><span class="p">)</span>
</span><span class="line">        <span class="p">}</span>
</span><span class="line">
</span><span class="line">        <span class="c1"># next beta</span>
</span><span class="line">        beta_curr<span class="o">&lt;-</span>beta_prev
</span><span class="line">        beta_curr<span class="p">[</span><span class="o">-</span>del<span class="p">]</span><span class="o">&lt;-</span><span class="kp">solve</span><span class="p">(</span><span class="kp">t</span><span class="p">(</span>x.new<span class="p">)</span><span class="o">%*%</span>x.new<span class="o">+</span><span class="kp">diag</span><span class="p">)</span><span class="o">%*%</span><span class="kp">t</span><span class="p">(</span>x.new<span class="p">)</span><span class="o">%*%</span>y
</span><span class="line">
</span><span class="line">        <span class="c1"># new converge value</span>
</span><span class="line">        converge<span class="o">&lt;-</span><span class="kp">sum</span><span class="p">((</span>beta_curr<span class="o">-</span>beta_prev<span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="p">)</span>
</span><span class="line">        <span class="c1"># next iteration</span>
</span><span class="line">        beta_prev<span class="o">&lt;-</span>beta_curr
</span><span class="line">
</span><span class="line">        <span class="c1"># if we don&#39;t need to remove the coefficients</span>
</span><span class="line">      <span class="p">}</span><span class="kp">else</span><span class="p">{</span>
</span><span class="line">        x.new<span class="o">&lt;-</span>x
</span><span class="line">        <span class="kp">diag</span><span class="o">&lt;-</span><span class="kp">diag</span><span class="p">(</span>grid<span class="p">[</span>i<span class="p">]</span><span class="o">*</span><span class="kp">q</span><span class="o">*</span><span class="p">(</span><span class="kp">abs</span><span class="p">(</span>beta_prev<span class="p">)</span><span class="o">^</span><span class="p">(</span><span class="kp">q</span><span class="m">-2</span><span class="p">))</span><span class="o">/</span><span class="m">2</span><span class="p">)</span>
</span><span class="line">        <span class="c1"># next beta</span>
</span><span class="line">        beta_curr<span class="o">&lt;-</span><span class="kp">solve</span><span class="p">(</span><span class="kp">t</span><span class="p">(</span>x.new<span class="p">)</span><span class="o">%*%</span>x.new<span class="o">+</span><span class="kp">diag</span><span class="p">)</span><span class="o">%*%</span><span class="kp">t</span><span class="p">(</span>x.new<span class="p">)</span><span class="o">%*%</span>y
</span><span class="line">
</span><span class="line">        <span class="c1"># new converge value</span>
</span><span class="line">        converge<span class="o">&lt;-</span><span class="kp">sum</span><span class="p">((</span>beta_curr<span class="o">-</span>beta_prev<span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="p">)</span>
</span><span class="line">        <span class="c1"># next iteration</span>
</span><span class="line">        beta_prev<span class="o">&lt;-</span>beta_curr
</span><span class="line">
</span><span class="line">      <span class="p">}</span>
</span><span class="line">    <span class="p">}</span>
</span><span class="line">
</span><span class="line">    beta.mat<span class="p">[,</span>i<span class="p">]</span><span class="o">&lt;-</span>beta_prev
</span><span class="line">  <span class="p">}</span>
</span><span class="line">  <span class="kp">colnames</span><span class="p">(</span>beta.mat<span class="p">)</span><span class="o">&lt;-</span>grid
</span><span class="line">  <span class="kr">return</span><span class="p">(</span>beta.mat<span class="p">)</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line"><span class="c1"># The function to find the optimal coefficients set</span>
</span><span class="line"><span class="c1"># that minimize mse of validation set</span>
</span><span class="line">bridge.opt<span class="o">&lt;-</span><span class="kr">function</span><span class="p">(</span>coef.matrix<span class="p">,</span> newx<span class="p">,</span> newy<span class="p">){</span>
</span><span class="line">  <span class="c1"># calculate the mse</span>
</span><span class="line">  mse.validate<span class="o">&lt;-</span><span class="kc">NULL</span>
</span><span class="line">  <span class="kr">for</span><span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>coef.matrix<span class="p">)){</span>
</span><span class="line">    mse.validate<span class="o">&lt;-</span><span class="kt">c</span><span class="p">(</span>mse.validate<span class="p">,</span> <span class="kp">mean</span><span class="p">((</span>newx<span class="o">%*%</span>coef.matrix<span class="p">[,</span>i<span class="p">]</span><span class="o">-</span>newy<span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="p">))</span>
</span><span class="line">  <span class="p">}</span>
</span><span class="line">  <span class="c1"># find the optimal coefficients set</span>
</span><span class="line">  coef.opt<span class="o">&lt;-</span>coef.matrix<span class="p">[,</span><span class="kp">which.min</span><span class="p">(</span>mse.validate<span class="p">)]</span>
</span><span class="line">  <span class="kr">return</span><span class="p">(</span>coef.opt<span class="p">)</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line">
</span><span class="line"><span class="c1">##################</span>
</span><span class="line"><span class="c1"># start simulation</span>
</span><span class="line"><span class="c1"># For this simulation, create three data sets </span>
</span><span class="line"><span class="c1"># consisting of 50/50/200 observations (training/validation/testing).</span>
</span><span class="line"><span class="c1"># Use validation data to select the tuning parameter (lambda).</span>
</span><span class="line">
</span><span class="line">
</span><span class="line"><span class="c1"># q settings</span>
</span><span class="line">q_seq<span class="o">&lt;-</span><span class="kt">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span><span class="m">0.5</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">)</span>
</span><span class="line"><span class="c1"># lambda grid</span>
</span><span class="line">grid<span class="o">=</span><span class="m">10</span><span class="o">^</span><span class="kp">seq</span><span class="p">(</span><span class="m">10</span><span class="p">,</span><span class="m">-2</span><span class="p">,</span>length<span class="o">=</span><span class="m">100</span><span class="p">)</span>
</span><span class="line"><span class="c1"># training and testing set</span>
</span><span class="line">ntrain<span class="o">&lt;-</span><span class="m">50</span>
</span><span class="line">ntest<span class="o">&lt;-</span><span class="m">200</span>
</span><span class="line"><span class="c1"># validation set size</span>
</span><span class="line">nvalidate<span class="o">&lt;-</span><span class="m">50</span>
</span><span class="line">
</span><span class="line"><span class="c1"># repeat number</span>
</span><span class="line">repeatnum<span class="o">&lt;-</span><span class="m">100</span>
</span><span class="line"><span class="c1"># initialize MSE</span>
</span><span class="line">MSE<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span> nrow<span class="o">=</span>repeatnum<span class="p">,</span>ncol<span class="o">=</span><span class="kp">length</span><span class="p">(</span>q_seq<span class="p">))</span>
</span><span class="line"><span class="kp">colnames</span><span class="p">(</span>MSE<span class="p">)</span><span class="o">&lt;-</span>q_seq
</span><span class="line">
</span><span class="line"><span class="kn">library</span><span class="p">(</span>MASS<span class="p">)</span>
</span><span class="line"><span class="c1"># the beta</span>
</span><span class="line"><span class="kp">beta</span><span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span><span class="kt">c</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="m">1.5</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">),</span>ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span>
</span><span class="line"><span class="c1"># covariance matrix of X</span>
</span><span class="line">cov<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="m">8</span><span class="p">,</span><span class="m">8</span><span class="p">)</span>
</span><span class="line"><span class="kr">for</span> <span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="m">8</span><span class="p">){</span>
</span><span class="line">  <span class="kr">for</span> <span class="p">(</span>j <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="m">8</span><span class="p">){</span>
</span><span class="line">    cov<span class="p">[</span>i<span class="p">,</span>j<span class="p">]</span><span class="o">&lt;-</span><span class="m">0.5</span><span class="o">^</span><span class="p">(</span><span class="kp">abs</span><span class="p">(</span>i<span class="o">-</span>j<span class="p">))</span>
</span><span class="line">  <span class="p">}</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line"><span class="kr">for</span> <span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span>repeatnum<span class="p">){</span>
</span><span class="line">  <span class="c1"># generate X</span>
</span><span class="line">  x.train<span class="o">&lt;-</span>mvrnorm<span class="p">(</span>n<span class="o">=</span>ntrain<span class="p">,</span><span class="kp">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">8</span><span class="p">),</span>cov<span class="p">)</span>
</span><span class="line">  x.test <span class="o">&lt;-</span>mvrnorm<span class="p">(</span>n<span class="o">=</span>ntest<span class="p">,</span><span class="kp">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">8</span><span class="p">),</span>cov<span class="p">)</span>
</span><span class="line">  x.validate <span class="o">&lt;-</span>mvrnorm<span class="p">(</span>n<span class="o">=</span>nvalidate<span class="p">,</span><span class="kp">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">8</span><span class="p">),</span>cov<span class="p">)</span>
</span><span class="line">  <span class="c1"># generate error term</span>
</span><span class="line">  err.train<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span>rnorm<span class="p">(</span>n<span class="o">=</span>ntrain<span class="p">,</span>mean<span class="o">=</span><span class="m">0</span><span class="p">,</span>sd<span class="o">=</span><span class="m">3</span><span class="p">),</span>ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span>
</span><span class="line">  err.test<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span>rnorm<span class="p">(</span>n<span class="o">=</span>ntest<span class="p">,</span>mean<span class="o">=</span><span class="m">0</span><span class="p">,</span>sd<span class="o">=</span><span class="m">3</span><span class="p">),</span>ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span>
</span><span class="line">  err.validate<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span>rnorm<span class="p">(</span>n<span class="o">=</span>nvalidate<span class="p">,</span>mean<span class="o">=</span><span class="m">0</span><span class="p">,</span>sd<span class="o">=</span><span class="m">3</span><span class="p">),</span>ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span>
</span><span class="line">  <span class="c1"># calculate Y</span>
</span><span class="line">  y.train<span class="o">&lt;-</span>x.train<span class="o">%*%</span><span class="kp">beta</span><span class="o">+</span>err.train
</span><span class="line">  y.test<span class="o">&lt;-</span>x.test<span class="o">%*%</span><span class="kp">beta</span><span class="o">+</span>err.test
</span><span class="line">  y.validate<span class="o">&lt;-</span>x.validate<span class="o">%*%</span><span class="kp">beta</span><span class="o">+</span>err.validate
</span><span class="line">  <span class="c1"># centralize Y and standardize X</span>
</span><span class="line">  y.train<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>y.train<span class="p">,</span>scale<span class="o">=</span><span class="bp">F</span><span class="p">);</span>y.test<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>y.test<span class="p">,</span>scale<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span><span class="line">  y.validate<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>y.validate<span class="p">,</span>scale<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</span><span class="line">  x.train<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>x.train<span class="p">);</span>x.test<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>x.test<span class="p">)</span>
</span><span class="line">  x.validate<span class="o">&lt;-</span><span class="kp">scale</span><span class="p">(</span>x.validate<span class="p">)</span>
</span><span class="line">
</span><span class="line">  <span class="c1"># run for each q</span>
</span><span class="line">  <span class="kr">for</span><span class="p">(</span>j <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="kp">length</span><span class="p">(</span>q_seq<span class="p">)){</span>
</span><span class="line">
</span><span class="line">    <span class="c1"># coefficients for trainning set</span>
</span><span class="line">    bridge.train<span class="o">&lt;-</span>bridge<span class="p">(</span>x.train<span class="p">,</span> y.train<span class="p">,</span> grid<span class="p">,</span> q<span class="o">=</span>q_seq<span class="p">[</span>j<span class="p">],</span> eta<span class="o">=</span><span class="m">0.001</span><span class="p">)</span>
</span><span class="line">    <span class="c1"># find the optimal coefficients set that minimize mse of validation set</span>
</span><span class="line">    coef.opt<span class="o">&lt;-</span>bridge.opt<span class="p">(</span>bridge.train<span class="p">,</span> x.validate<span class="p">,</span> y.validate<span class="p">)</span>
</span><span class="line">    <span class="c1"># MSE on the test set using the optimal model</span>
</span><span class="line">    MSE<span class="p">[</span>i<span class="p">,</span>j<span class="p">]</span><span class="o">&lt;-</span><span class="kp">mean</span><span class="p">((</span>x.test<span class="o">%*%</span>coef.opt<span class="o">-</span>y.test<span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="p">)</span>
</span><span class="line">  <span class="p">}</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line"><span class="c1"># mean of MSEs under different models</span>
</span><span class="line"><span class="kp">apply</span><span class="p">(</span>MSE<span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="kp">mean</span><span class="p">)</span>
</span><span class="line"><span class="c1"># sd of MSEs under different models</span>
</span><span class="line"><span class="kp">apply</span><span class="p">(</span>MSE<span class="p">,</span><span class="m">2</span><span class="p">,</span>sd<span class="p">)</span>
</span><span class="line"><span class="c1"># boxplot of MSEs under different models</span>
</span><span class="line">boxplot<span class="p">(</span>MSE<span class="p">,</span> ylab<span class="o">=</span><span class="s">&quot;MSE&quot;</span><span class="p">,</span> xlab<span class="o">=</span><span class="s">&quot;q&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><img src="http://bioops.info/images/uploads/2015/bridge.png" alt="bridge MSE plots" /></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/01/simulation-bernoulli-consitent-estimator/">The Consistent Estimator of Bernouli Distribution</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-01-04T01:53:14+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>4</span><span class='date-suffix'>th</span>, <span class='date-year'>2015</span></span> <span class='time'>1:53 am</span></time>
        
         | <a href="/2015/01/simulation-bernoulli-consitent-estimator/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>This is a simple post showing the basic knowledge of statistics, the consistency.</p>

<p>For Bernoulli distribution, $ Y \sim B(n,p) $, $ \hat{p}=Y/n $ is a consistent estimator of $ p $, because:</p>

<script type="math/tex; mode=display">% <![CDATA[
\lim_{n \to \infty} \left(p-\epsilon<\frac{Y}{n}<p+\epsilon\right)=1, %]]></script>

<p>for any positive number $ \epsilon $.</p>

<p>Here is the simulation to show the estimator is consitent.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># set parameters</span>
</span><span class="line">n<span class="o">&lt;-</span><span class="m">1000</span><span class="p">;</span>p<span class="o">&lt;-</span><span class="m">0.5</span><span class="p">;</span>
</span><span class="line"><span class="c1"># n Bernoulli trails</span>
</span><span class="line">obs<span class="o">&lt;-</span>rbinom<span class="p">(</span>n<span class="p">,</span><span class="m">1</span><span class="p">,</span>p<span class="p">)</span>
</span><span class="line"><span class="c1"># estimate p on different number of trials.</span>
</span><span class="line">phat<span class="o">&lt;-</span><span class="kp">cumsum</span><span class="p">(</span>obs<span class="p">)</span><span class="o">/</span><span class="kp">cumsum</span><span class="p">(</span><span class="kp">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span>n<span class="p">))</span>
</span><span class="line"><span class="c1"># the convergence plot</span>
</span><span class="line">plot<span class="p">(</span>phat<span class="p">,</span> type<span class="o">=</span><span class="s">&quot;l&quot;</span><span class="p">,</span> xlab<span class="o">=</span><span class="s">&quot;Trails&quot;</span><span class="p">)</span>
</span><span class="line">abline<span class="p">(</span>h<span class="o">=</span>p<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><img src="http://bioops.info/images/uploads/2015/Rplot01.png" alt="convergence plot" /></p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># then, 100 repetitions</span>
</span><span class="line">
</span><span class="line"><span class="c1"># set parameters</span>
</span><span class="line">n<span class="o">&lt;-</span><span class="m">1000</span><span class="p">;</span>p<span class="o">&lt;-</span><span class="m">0.5</span><span class="p">;</span>B<span class="o">&lt;-</span><span class="m">100</span><span class="p">;</span>
</span><span class="line"><span class="c1"># n*B Bernoulli trails</span>
</span><span class="line">obs<span class="o">&lt;-</span>rbinom<span class="p">(</span>n<span class="o">*</span>B<span class="p">,</span><span class="m">1</span><span class="p">,</span>p<span class="p">)</span>
</span><span class="line"><span class="c1"># convert n*B observations to a n*B matrix</span>
</span><span class="line">obs_mat<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span>obs<span class="p">,</span> nrow<span class="o">=</span>n<span class="p">,</span> ncol<span class="o">=</span>B<span class="p">)</span>
</span><span class="line"><span class="c1"># a function to estimate p on different number of trials</span>
</span><span class="line">est_p<span class="o">&lt;-</span><span class="kr">function</span><span class="p">(</span>x<span class="p">,</span>n<span class="p">)</span> <span class="kp">cumsum</span><span class="p">(</span>x<span class="p">)</span><span class="o">/</span><span class="kp">cumsum</span><span class="p">(</span><span class="kp">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span>n<span class="p">))</span>
</span><span class="line"><span class="c1"># estimate p on different number of trials for each repetition</span>
</span><span class="line">phat_mat<span class="o">&lt;-</span><span class="kp">apply</span><span class="p">(</span>obs_mat<span class="p">,</span><span class="m">2</span><span class="p">,</span> est_p<span class="p">,</span> n<span class="o">=</span>n<span class="p">)</span>
</span><span class="line"><span class="c1"># the convergence plot with 100 repetitions</span>
</span><span class="line">matplot<span class="p">(</span>phat_mat<span class="p">,</span>type<span class="o">=</span><span class="s">&quot;l&quot;</span><span class="p">,</span>lty<span class="o">=</span><span class="m">1</span><span class="p">,</span>xlab<span class="o">=</span><span class="s">&quot;Trials&quot;</span><span class="p">,</span>ylab<span class="o">=</span><span class="s">&quot;phat&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><img src="http://bioops.info/images/uploads/2015/Rplot02.png" alt="convergence plots" /></p>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/01/permutation-pca/">Permutation Test for Principal Component Analysis</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-01-02T05:35:51+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>2</span><span class='date-suffix'>nd</span>, <span class='date-year'>2015</span></span> <span class='time'>5:35 am</span></time>
        
         | <a href="/2015/01/permutation-pca/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>The procedure of permutation test for PCA is as follows:</p>

<p>For each replicate,</p>

<ol>
  <li>
    <p>Individually permute each column of the data matrix.</p>
  </li>
  <li>
    <p>Conduct the PCA and find the proportion of variance explained by each of the components 1 to s. Store this information.</p>
  </li>
  <li>
    <p>Repeat 1 and 2 R times.</p>
  </li>
</ol>

<p>At the end of this we will have a matrix with R rows and s columns that contains the proportion of variance explained by each component for each replicate.</p>

<p>Finally, compare the observed values from the original data to the set of values from the permutations in order to determine the approximate p-value.</p>

<p>The R code:</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>pca_perm.R</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
<span class="line-number">27</span>
<span class="line-number">28</span>
<span class="line-number">29</span>
<span class="line-number">30</span>
<span class="line-number">31</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># the fuction to assess the significance of the principal components.</span>
</span><span class="line">sign.pc<span class="o">&lt;-</span><span class="kr">function</span><span class="p">(</span>x<span class="p">,</span>R<span class="o">=</span><span class="m">1000</span><span class="p">,</span>s<span class="o">=</span><span class="m">10</span><span class="p">,</span> cor<span class="o">=</span><span class="bp">T</span><span class="p">,</span><span class="kc">...</span><span class="p">){</span>
</span><span class="line">  <span class="c1"># run PCA</span>
</span><span class="line">  pc.out<span class="o">&lt;-</span>princomp<span class="p">(</span>x<span class="p">,</span>cor<span class="o">=</span>cor<span class="p">,</span><span class="kc">...</span><span class="p">)</span>
</span><span class="line">  <span class="c1"># the proportion of variance of each PC</span>
</span><span class="line">  pve<span class="o">=</span><span class="p">(</span>pc.out<span class="o">$</span>sdev<span class="o">^</span><span class="m">2</span><span class="o">/</span><span class="kp">sum</span><span class="p">(</span>pc.out<span class="o">$</span>sdev<span class="o">^</span><span class="m">2</span><span class="p">))[</span><span class="m">1</span><span class="o">:</span>s<span class="p">]</span>
</span><span class="line">
</span><span class="line">  <span class="c1"># a matrix with R rows and s columns that contains</span>
</span><span class="line">  <span class="c1"># the proportion of variance explained by each pc</span>
</span><span class="line">  <span class="c1"># for each randomization replicate.</span>
</span><span class="line">  pve.perm<span class="o">&lt;-</span><span class="kt">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span>ncol<span class="o">=</span>s<span class="p">,</span>nrow<span class="o">=</span>R<span class="p">)</span>
</span><span class="line">  <span class="kr">for</span><span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span>R<span class="p">){</span>
</span><span class="line">    <span class="c1"># permutation each column</span>
</span><span class="line">    x.perm<span class="o">&lt;-</span><span class="kp">apply</span><span class="p">(</span>x<span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="kp">sample</span><span class="p">)</span>
</span><span class="line">    <span class="c1"># run PCA</span>
</span><span class="line">    pc.perm.out<span class="o">&lt;-</span>princomp<span class="p">(</span>x.perm<span class="p">,</span>cor<span class="o">=</span>cor<span class="p">,</span><span class="kc">...</span><span class="p">)</span>
</span><span class="line">    <span class="c1"># the proportion of variance of each PC.perm</span>
</span><span class="line">    pve.perm<span class="p">[</span>i<span class="p">,]</span><span class="o">=</span><span class="p">(</span>pc.perm.out<span class="o">$</span>sdev<span class="o">^</span><span class="m">2</span><span class="o">/</span><span class="kp">sum</span><span class="p">(</span>pc.perm.out<span class="o">$</span>sdev<span class="o">^</span><span class="m">2</span><span class="p">))[</span><span class="m">1</span><span class="o">:</span>s<span class="p">]</span>
</span><span class="line">  <span class="p">}</span>
</span><span class="line">  <span class="c1"># calcalute the p-values</span>
</span><span class="line">  pval<span class="o">&lt;-</span><span class="kp">apply</span><span class="p">(</span><span class="kp">t</span><span class="p">(</span>pve.perm<span class="p">)</span><span class="o">&gt;</span>pve<span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="kp">sum</span><span class="p">)</span><span class="o">/</span>R
</span><span class="line">  <span class="kr">return</span><span class="p">(</span><span class="kt">list</span><span class="p">(</span>pve<span class="o">=</span>pve<span class="p">,</span>pval<span class="o">=</span>pval<span class="p">))</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line">
</span><span class="line"><span class="c1"># apply the function</span>
</span><span class="line"><span class="kn">library</span><span class="p">(</span>RCurl<span class="p">)</span>
</span><span class="line">data <span class="o">&lt;-</span> getURL<span class="p">(</span><span class="s">&quot;https://raw.githubusercontent.com/bioops/mis_scripts/master/statistics/data/pca.txt&quot;</span><span class="p">)</span>
</span><span class="line">OCRdata <span class="o">&lt;-</span> read.table<span class="p">(</span>text <span class="o">=</span> data<span class="p">,</span> header<span class="o">=</span><span class="bp">T</span><span class="p">,</span>sep<span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
</span><span class="line">OCRdat<span class="o">&lt;-</span>OCRdata<span class="p">[,</span><span class="m">-1</span><span class="p">]</span> <span class="c1">#leave out location id column</span>
</span><span class="line">sign.pc<span class="p">(</span>OCRdat<span class="p">,</span>cor<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>The result:</p>

<pre>
$pve
    Comp.1     Comp.2     Comp.3     Comp.4     Comp.5     Comp.6     Comp.7     Comp.8 
0.23129378 0.14864525 0.11552865 0.06741744 0.06274641 0.05858431 0.05033795 0.04484122 
    Comp.9    Comp.10 
0.03873311 0.03431297 

$pval
 [1] 0.000 0.000 0.000 1.000 1.000 0.996 1.000 1.000 1.000 1.000
</pre>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/01/svm/">Demo of Support Vector Machine</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-01-01T20:43:07+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>1</span><span class='date-suffix'>st</span>, <span class='date-year'>2015</span></span> <span class='time'>8:43 pm</span></time>
        
         | <a href="/2015/01/svm/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>Demo of SVM</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>svm.R</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># load library</span>
</span><span class="line"><span class="kn">library</span><span class="p">(</span>e1071<span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># simulate x and y</span>
</span><span class="line">x1<span class="o">&lt;-</span>rnorm<span class="p">(</span><span class="m">100</span><span class="p">);</span>x2<span class="o">&lt;-</span>rnorm<span class="p">(</span><span class="m">100</span><span class="p">)</span>
</span><span class="line">y<span class="o">&lt;-</span><span class="kp">as.factor</span><span class="p">(</span><span class="kp">ifelse</span><span class="p">(</span>x1<span class="o">^</span><span class="m">2</span><span class="o">+</span>x2<span class="o">^</span><span class="m">2</span><span class="o">&lt;=</span><span class="m">1.6</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">-1</span><span class="p">))</span>
</span><span class="line">dat3<span class="o">&lt;-</span><span class="kt">data.frame</span><span class="p">(</span>x1<span class="p">,</span>x2<span class="p">,</span>y<span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># (a) tuning parameters</span>
</span><span class="line">cost<span class="o">&lt;-</span><span class="kt">c</span><span class="p">(</span><span class="m">0.001</span><span class="p">,</span> <span class="m">0.01</span><span class="p">,</span> <span class="m">0.1</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="m">5</span><span class="p">,</span> <span class="m">10</span><span class="p">,</span> <span class="m">100</span><span class="p">)</span>
</span><span class="line"><span class="kp">gamma</span><span class="o">&lt;-</span><span class="kp">seq</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">0.1</span><span class="p">)</span>
</span><span class="line">tune.out<span class="o">&lt;-</span>tune<span class="p">(</span>svm<span class="p">,</span> y<span class="o">~</span><span class="m">.</span><span class="p">,</span> data<span class="o">=</span>dat3<span class="p">,</span> kernel<span class="o">=</span><span class="s">&quot;radial&quot;</span><span class="p">,</span>
</span><span class="line">              ranges<span class="o">=</span><span class="kt">list</span><span class="p">(</span>cost<span class="o">=</span>cost<span class="p">,</span>gamma<span class="o">=</span><span class="kp">gamma</span><span class="p">))</span>
</span><span class="line">bestmod<span class="o">&lt;-</span>tune.out<span class="o">$</span>best.model
</span><span class="line"><span class="c1"># the best model</span>
</span><span class="line"><span class="kp">summary</span><span class="p">(</span>bestmod<span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># (b) test set</span>
</span><span class="line"><span class="c1"># simulate test set</span>
</span><span class="line">x1<span class="o">&lt;-</span>rnorm<span class="p">(</span><span class="m">100</span><span class="p">);</span>x2<span class="o">&lt;-</span>rnorm<span class="p">(</span><span class="m">100</span><span class="p">)</span>
</span><span class="line">y<span class="o">&lt;-</span><span class="kp">as.factor</span><span class="p">(</span><span class="kp">ifelse</span><span class="p">(</span>x1<span class="o">^</span><span class="m">2</span><span class="o">+</span>x2<span class="o">^</span><span class="m">2</span><span class="o">&lt;=</span><span class="m">1.6</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">-1</span><span class="p">))</span>
</span><span class="line">dat3.test<span class="o">&lt;-</span><span class="kt">data.frame</span><span class="p">(</span>x1<span class="p">,</span>x2<span class="p">,</span>y<span class="p">)</span>
</span><span class="line">ypred<span class="o">&lt;-</span>predict<span class="p">(</span>bestmod<span class="p">,</span>dat3.test<span class="p">)</span>
</span><span class="line"><span class="c1"># the confusion matrix</span>
</span><span class="line"><span class="kp">table</span><span class="p">(</span>predict<span class="o">=</span>ypred<span class="p">,</span> truth<span class="o">=</span>dat3.test<span class="o">$</span>y<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/01/lm-cv/">Linear Regression With Cross Validation</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-01-01T20:33:24+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>1</span><span class='date-suffix'>st</span>, <span class='date-year'>2015</span></span> <span class='time'>8:33 pm</span></time>
        
         | <a href="/2015/01/lm-cv/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>Cross validation for linear model and the bootstrap confidence interval for coefficients</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>linear_CV.R</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
<span class="line-number">27</span>
<span class="line-number">28</span>
<span class="line-number">29</span>
<span class="line-number">30</span>
<span class="line-number">31</span>
<span class="line-number">32</span>
<span class="line-number">33</span>
<span class="line-number">34</span>
<span class="line-number">35</span>
<span class="line-number">36</span>
<span class="line-number">37</span>
<span class="line-number">38</span>
<span class="line-number">39</span>
<span class="line-number">40</span>
<span class="line-number">41</span>
<span class="line-number">42</span>
<span class="line-number">43</span>
<span class="line-number">44</span>
<span class="line-number">45</span>
<span class="line-number">46</span>
<span class="line-number">47</span>
<span class="line-number">48</span>
<span class="line-number">49</span>
<span class="line-number">50</span>
<span class="line-number">51</span>
<span class="line-number">52</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># (a) Linear model</span>
</span><span class="line">
</span><span class="line"><span class="c1"># read data</span>
</span><span class="line"><span class="kn">library</span><span class="p">(</span>RCurl<span class="p">)</span>
</span><span class="line">data <span class="o">&lt;-</span> getURL<span class="p">(</span><span class="s">&quot;https://raw.githubusercontent.com/bioops/mis_scripts/master/statistics/data/prostateData.txt&quot;</span><span class="p">)</span>
</span><span class="line">prostate <span class="o">&lt;-</span> read.table<span class="p">(</span>text <span class="o">=</span> data<span class="p">,</span> header<span class="o">=</span><span class="bp">T</span><span class="p">,</span>sep<span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
</span><span class="line"><span class="c1"># remove the first column</span>
</span><span class="line">prostate<span class="o">&lt;-</span>prostate<span class="p">[,</span><span class="m">-1</span><span class="p">]</span>
</span><span class="line"><span class="c1"># run the linear model</span>
</span><span class="line">prostate_lm<span class="o">&lt;-</span>lm<span class="p">(</span>lpsa<span class="o">~</span><span class="m">.</span><span class="p">,</span>data<span class="o">=</span>prostate<span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># summary of the fit</span>
</span><span class="line"><span class="kp">summary</span><span class="p">(</span>prostate_lm<span class="p">)</span>
</span><span class="line">
</span><span class="line">
</span><span class="line"><span class="c1"># (b) Cross-validation</span>
</span><span class="line">
</span><span class="line"><span class="c1"># need boot package</span>
</span><span class="line"><span class="kn">library</span><span class="p">(</span>boot<span class="p">)</span>
</span><span class="line"><span class="c1"># fit the linear model</span>
</span><span class="line">glm.fit<span class="o">&lt;-</span>glm<span class="p">(</span>lpsa<span class="o">~</span>lcavol<span class="o">+</span>lweight<span class="o">+</span>age<span class="o">+</span>lbph<span class="o">+</span>svi<span class="o">+</span>lcp<span class="o">+</span>gleason<span class="o">+</span>pgg45<span class="p">,</span>data<span class="o">=</span>prostate<span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># LOOCV</span>
</span><span class="line">cv.glm<span class="p">(</span>prostate<span class="p">,</span> glm.fit<span class="p">)</span><span class="o">$</span>delta<span class="p">[</span><span class="m">1</span><span class="p">]</span>
</span><span class="line">
</span><span class="line"><span class="c1"># 10-fold CV</span>
</span><span class="line">cv.err.10<span class="o">&lt;-</span><span class="kp">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">10</span><span class="p">)</span>
</span><span class="line"><span class="kr">for</span><span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">){</span>
</span><span class="line">  cv.err.10<span class="p">[</span>i<span class="p">]</span><span class="o">&lt;-</span>cv.glm<span class="p">(</span>prostate<span class="p">,</span> glm.fit<span class="p">,</span> K<span class="o">=</span><span class="m">10</span><span class="p">)</span><span class="o">$</span>delta<span class="p">[</span><span class="m">1</span><span class="p">]</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line"><span class="c1"># mean error of 10-fold CV</span>
</span><span class="line"><span class="kp">mean</span><span class="p">(</span>cv.err.10<span class="p">)</span>
</span><span class="line">
</span><span class="line">
</span><span class="line"><span class="c1"># (c) Bootstrap</span>
</span><span class="line">
</span><span class="line"><span class="c1"># get ther residuals and fitted values from linear model</span>
</span><span class="line">prostate_new<span class="o">&lt;-</span><span class="kt">data.frame</span><span class="p">(</span>prostate<span class="p">,</span> res<span class="o">=</span>resid<span class="p">(</span>prostate_lm<span class="p">),</span> fitted<span class="o">=</span>fitted<span class="p">(</span>prostate_lm<span class="p">))</span>
</span><span class="line"><span class="c1"># a function to get the coefficients from each bootstrap</span>
</span><span class="line">prostate.fun<span class="o">&lt;-</span><span class="kr">function</span><span class="p">(</span>data<span class="p">,</span> i<span class="p">){</span>
</span><span class="line">  d<span class="o">&lt;-</span>data
</span><span class="line">  d<span class="o">$</span>lpsa<span class="o">&lt;-</span>d<span class="o">$</span>fitted<span class="o">+</span>d<span class="o">$</span>res<span class="p">[</span>i<span class="p">]</span>
</span><span class="line">  coef<span class="p">(</span>update<span class="p">(</span>prostate_lm<span class="p">,</span> data<span class="o">=</span>d<span class="p">))</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line"><span class="c1"># bootstrap</span>
</span><span class="line">prostate.lm.boot<span class="o">&lt;-</span>boot<span class="p">(</span>prostate_new<span class="p">,</span> prostate.fun<span class="p">,</span> R<span class="o">=</span><span class="m">1000</span><span class="p">)</span>
</span><span class="line">
</span><span class="line"><span class="c1"># 95% conficence level for lcavol</span>
</span><span class="line">boot.ci<span class="p">(</span>prostate.lm.boot<span class="p">,</span> index<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</span><span class="line"><span class="c1"># 95% conficence level for lweight</span>
</span><span class="line">boot.ci<span class="p">(</span>prostate.lm.boot<span class="p">,</span> index<span class="o">=</span><span class="m">3</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/01/gamma-mme-mle/">Estimate Gamma Distribution Parmaters Using MME and MLE</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-01-01T12:13:59+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>1</span><span class='date-suffix'>st</span>, <span class='date-year'>2015</span></span> <span class='time'>12:13 pm</span></time>
        
         | <a href="/2015/01/gamma-mme-mle/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>This post shows how to estimate gamma distribution parameters using (a) moment of estimation (MME) and (b) maximum likelihood estimate (MLE).</p>

<p>The probability density function of Gamma distribution is</p>

<script type="math/tex; mode=display">\frac{1}{\Gamma (\alpha) \beta ^{\alpha}} x^{\alpha - 1} e^{- \frac{x}{\beta}}</script>

<p>The MME:</p>

<script type="math/tex; mode=display">\hat{\alpha}=\frac{n\bar{X} ^2}{\sum_{i=1}^{n} (X_i-\bar{X})^2}</script>

<script type="math/tex; mode=display">\hat{\beta}=\frac{\sum_{i=1}^{n} (X_i-\bar{X})^2}{n \bar{X}}</script>

<p>We can calculate the MLE of $ \alpha $ using the Newton-Raphson method.</p>

<p>For $ k =1,2,…,$</p>

<script type="math/tex; mode=display">\hat{\alpha} ^{(k)}=\hat{\alpha} ^{(k-1)} - \frac{\ell'(\hat{\alpha} ^{(k-1)})}{\ell'' (\hat{\alpha} ^{(k-1)})}</script>

<p>where</p>

<script type="math/tex; mode=display">\ell' (\alpha) = n \log \left(\frac{\alpha}{\bar{X}}\right)-n \frac{\Gamma '(\alpha)}{\Gamma(\alpha)}+\sum_{i=1}^{n} \log X_i</script>

<script type="math/tex; mode=display">\ell'' (\alpha) = \frac{n}{\alpha} - n \left(\frac{\Gamma '(\alpha)}{\Gamma (\alpha)}\right)'</script>

<p>Use the MME for the initial value of $ \alpha^{(0)} $, and stop the approximation when $ \vert \hat{\alpha}^{(k)}-\hat{\alpha}^{(k-1)} \vert &lt; 0.0000001 $. The MLE of $ \beta $ can be found by $ \hat{\beta} = \bar{X} / \hat{\alpha} $.</p>

<p>Below is the R code.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span>gamma.R</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
<span class="line-number">27</span>
<span class="line-number">28</span>
<span class="line-number">29</span>
<span class="line-number">30</span>
<span class="line-number">31</span>
<span class="line-number">32</span>
<span class="line-number">33</span>
<span class="line-number">34</span>
<span class="line-number">35</span>
<span class="line-number">36</span>
<span class="line-number">37</span>
<span class="line-number">38</span>
<span class="line-number">39</span>
<span class="line-number">40</span>
<span class="line-number">41</span>
<span class="line-number">42</span>
<span class="line-number">43</span>
<span class="line-number">44</span>
<span class="line-number">45</span>
<span class="line-number">46</span>
<span class="line-number">47</span>
<span class="line-number">48</span>
<span class="line-number">49</span>
<span class="line-number">50</span>
<span class="line-number">51</span>
<span class="line-number">52</span>
</pre></td><td class="code"><pre><code class="R"><span class="line"><span class="c1"># (a) MME</span>
</span><span class="line">gamma_MME<span class="o">&lt;-</span><span class="kr">function</span><span class="p">(</span>x<span class="p">){</span>
</span><span class="line">  n<span class="o">&lt;-</span><span class="kp">length</span><span class="p">(</span>x<span class="p">)</span>
</span><span class="line">  mean_x<span class="o">&lt;-</span><span class="kp">mean</span><span class="p">(</span>x<span class="p">)</span>
</span><span class="line">  alpha<span class="o">&lt;-</span>n<span class="o">*</span><span class="p">(</span>mean_x<span class="o">^</span><span class="m">2</span><span class="p">)</span><span class="o">/</span><span class="kp">sum</span><span class="p">((</span>x<span class="o">-</span>mean_x<span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="p">)</span>
</span><span class="line">  <span class="kp">beta</span><span class="o">&lt;-</span><span class="kp">sum</span><span class="p">((</span>x<span class="o">-</span>mean_x<span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="p">)</span><span class="o">/</span>n<span class="o">/</span>mean_x
</span><span class="line">  estimate_MME<span class="o">&lt;-</span><span class="kt">data.frame</span><span class="p">(</span>alpha<span class="p">,</span><span class="kp">beta</span><span class="p">)</span>
</span><span class="line">  <span class="kr">return</span><span class="p">(</span>estimate_MME<span class="p">)</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line">
</span><span class="line"><span class="c1"># (b) MLE</span>
</span><span class="line">gamma_MLE<span class="o">&lt;-</span><span class="kr">function</span><span class="p">(</span>x<span class="p">){</span>
</span><span class="line">  n<span class="o">&lt;-</span><span class="kp">length</span><span class="p">(</span>x<span class="p">)</span>
</span><span class="line">  mean_x<span class="o">&lt;-</span><span class="kp">mean</span><span class="p">(</span>x<span class="p">)</span>
</span><span class="line">
</span><span class="line">  <span class="c1"># initiate the convergence and alpha value</span>
</span><span class="line">  converg<span class="o">&lt;-</span><span class="m">1000</span>
</span><span class="line">  alpha_prev<span class="o">&lt;-</span>gamma_MME<span class="p">(</span>x<span class="p">)</span><span class="o">$</span>alpha
</span><span class="line">
</span><span class="line">  <span class="c1"># initiate two vectors to store alpha and beta in each step</span>
</span><span class="line">  alpha_est<span class="o">&lt;-</span>alpha_prev
</span><span class="line">  beta_est<span class="o">&lt;-</span>mean_x<span class="o">/</span>alpha_prev
</span><span class="line">
</span><span class="line">  <span class="c1"># Newton-Raphson</span>
</span><span class="line">  <span class="kr">while</span><span class="p">(</span>converg<span class="o">&gt;</span><span class="m">0.0000001</span><span class="p">){</span>
</span><span class="line">    <span class="c1">#first derivative of alpha_k-1</span>
</span><span class="line">    der1<span class="o">&lt;-</span>n<span class="o">*</span><span class="kp">log</span><span class="p">(</span>alpha_prev<span class="o">/</span>mean_x<span class="p">)</span><span class="o">-</span>n<span class="o">*</span><span class="kp">digamma</span><span class="p">(</span>alpha_prev<span class="p">)</span><span class="o">+</span><span class="kp">sum</span><span class="p">(</span><span class="kp">log</span><span class="p">(</span>x<span class="p">))</span>
</span><span class="line">    <span class="c1">#second derivative of alpha_k-1</span>
</span><span class="line">    der2<span class="o">&lt;-</span>n<span class="o">/</span>alpha_prev<span class="o">-</span>n<span class="o">*</span><span class="kp">trigamma</span><span class="p">(</span>alpha_prev<span class="p">)</span>
</span><span class="line">    <span class="c1">#calculate next alpha</span>
</span><span class="line">    alpha_next<span class="o">&lt;-</span>alpha_prev<span class="o">-</span>der1<span class="o">/</span>der2
</span><span class="line">    <span class="c1"># get the convergence value</span>
</span><span class="line">    converg<span class="o">&lt;-</span><span class="kp">abs</span><span class="p">(</span>alpha_next<span class="o">-</span>alpha_prev<span class="p">)</span>
</span><span class="line">    <span class="c1"># store estimators in each step</span>
</span><span class="line">    alpha_est<span class="o">&lt;-</span><span class="kt">c</span><span class="p">(</span>alpha_est<span class="p">,</span> alpha_next<span class="p">)</span>
</span><span class="line">    beta_est<span class="o">&lt;-</span><span class="kt">c</span><span class="p">(</span>beta_est<span class="p">,</span> mean_x<span class="o">/</span>alpha_next<span class="p">)</span>
</span><span class="line">    <span class="c1"># go to next alpha</span>
</span><span class="line">    alpha_prev<span class="o">&lt;-</span>alpha_next
</span><span class="line">  <span class="p">}</span>
</span><span class="line">
</span><span class="line">  alpha<span class="o">&lt;-</span>alpha_next
</span><span class="line">  <span class="kp">beta</span><span class="o">&lt;-</span>mean_x<span class="o">/</span>alpha_next
</span><span class="line">  estimate_MLE<span class="o">&lt;-</span><span class="kt">data.frame</span><span class="p">(</span>alpha<span class="p">,</span><span class="kp">beta</span><span class="p">)</span>
</span><span class="line">
</span><span class="line">  <span class="kr">return</span><span class="p">(</span>estimate_MLE<span class="p">)</span>
</span><span class="line"><span class="p">}</span>
</span><span class="line">
</span><span class="line"><span class="c1"># apply</span>
</span><span class="line">x<span class="o">&lt;-</span>rgamma<span class="p">(</span><span class="m">100</span><span class="p">,</span><span class="m">2</span><span class="p">,</span>scale<span class="o">=</span><span class="m">5</span><span class="p">)</span>
</span><span class="line">gammma_MME<span class="p">(</span>x<span class="p">)</span>
</span><span class="line">gamma_MLE<span class="p">(</span>x<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2015/01/2015/">2015</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2015-01-01T02:31:05+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>1</span><span class='date-suffix'>st</span>, <span class='date-year'>2015</span></span> <span class='time'>2:31 am</span></time>
        
         | <a href="/2015/01/2015/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>Happy new year!</p>

<p>Hopefully, I will <a href="http://bioops.info/">write</a> and <a href="https://github.com/bioops">code</a> more often in 2015.</p>

<p>Stay tuned!</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2014/07/cfda-approved-bgis-next-generation-sequencing-diagnostic-products/">CFDA Approved BGI&#8217;s Next Generation Sequencing Diagnostic Products</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2014-07-04T14:32:11+00:00'><span class='date'><span class='date-month'>Jul</span> <span class='date-day'>4</span><span class='date-suffix'>th</span>, <span class='date-year'>2014</span></span> <span class='time'>2:32 pm</span></time>
        
         | <a href="/2014/07/cfda-approved-bgis-next-generation-sequencing-diagnostic-products/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>只收集一些相关资料，不评论。
<a href="http://www.sda.gov.cn/WS01/CL0051/102239.html">http://www.sda.gov.cn/WS01/CL0051/102239.html</a>
2014年6月30日，国家食品药品监督管理总局经审查，批准了BGISEQ-1000基因测序仪、BGISEQ-100基因测序仪和胎儿染色体非整倍体（T21、T18、T13）检测试剂盒（联合探针锚定连接测序法）、胎儿染色体非整倍体（T21、T18、T13）检测试剂盒（半导体测序法）医疗器械注册。这是国家食品药品监督管理总局首次批准注册的第二代基因测序诊断产品。</p>

<p>该批产品可通过对孕周12周以上的高危孕妇外周血血浆中的游离基因片段进行基因测序，对胎儿染色体非整倍体疾病21-三体综合征、18-三体综合征和13-三体综合征进行无创产前检查和辅助诊断。</p>

<p><a href="http://www.knowgene.com/question/677">http://www.knowgene.com/question/677</a>
BGISEQ-1000基因测序仪基于Complete Genomics平台，配套的试剂盒为胎儿染色体非整倍体（T21、T18、T13）检测试剂盒（联合探针锚定连接测序法）。CG平台的特点是通量高，但周期较长，因此BGISEQ-1000应该主要会应用于全国范围内的样品，集中测序分析；</p>

<p>BGISEQ-100基因测序仪基于Ion Torrent平台，配套的试剂盒为胎儿染色体非整倍体（T21、T18、T13）检测试剂盒（半导体测序法）。Ion Torrent平台的特点是测序周期短，可灵活部署，BGISEQ-100有很大可能会被部署到有一定业务量的大中型医院，就地采样、测序、分析并出具报告.</p>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2014/01/stat-learning-starts/">Online &#8216;Statistical Learning&#8217; Course Starts Jan 21 2014</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2014-01-20T10:30:00+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>20</span><span class='date-suffix'>th</span>, <span class='date-year'>2014</span></span> <span class='time'>10:30 am</span></time>
        
         | <a href="/2014/01/stat-learning-starts/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>The free online course <a href="https://class.stanford.edu/courses/HumanitiesScience/StatLearning/Winter2014/about">Statistical Learning</a> will start tomorrow. It will be taught by Rob Tibshirani and Trevor Hastie. It is an excellent opportunity to learn directly from the two famous professors in the field and the authors of two great textbooks on statistical learning, <a href="http://www-bcf.usc.edu/~gareth/ISL/">An Introduction to Statistical Learning, with Applications in R</a> and <a href="http://statweb.stanford.edu/~tibs/ElemStatLearn/">The Elements of Statistical Learning</a>.</p>
</div>
  
  


    </article>
  
  <div class="pagination">
    
      <a class="prev" href="/page2">&larr; Older</a>
    
    <a href="/blog/archives">Blog Archives</a>
    
  </div>
</div>
<aside class="sidebar">
  
    
  
</aside>

    </div>
  </div>
  <footer role="contentinfo"><!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>

<p>
  Copyright &copy; 2016 - Bioops -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a> | Themed with <a href="https://github.com/lucaslew/whitespace">Whitespace</a></span>
</p>


</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'bioopsgithub';
      
        
        var disqus_script = 'count.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>






<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
