
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Bioops</title>
  <meta name="author" content="Bioops">

  
  <meta name="description" content="Nature Review Genetics doi:10.1038/nrg3068 一篇入门文章，值得一看，包括引用的一些review Abstract Transcriptomics studies often rely on partial reference transcriptomes &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bioops.info/page9">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="Bioops" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="/javascripts/libs/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Fjalla+One" rel="stylesheet" type="text/css">
<!--- MathJax Configuration -->
<script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-21431888-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   class="collapse-sidebar sidebar-footer" >
  <header role="banner"><hgroup>
  <h1><a href="/">Bioops</a></h1>
  
    <h2>Bioinformatics=(ACGAAG->AK)+(#!/bin/sh)+(P(A|B)=P(B|A)*P(A)/P(B))</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscribe" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 13.310204,73.332654 C 5.967347,73.332654 0,79.322448 0,86.621428 c 0,7.338776 5.967347,13.262246 13.310204,13.262246 7.370408,0 13.328572,-5.92245 13.328572,-13.262246 0,-7.29898 -5.958164,-13.288774 -13.328572,-13.288774 z M 0.01530612,33.978572 V 53.143878 C 12.493878,53.143878 24.229592,58.02347 33.068368,66.865306 41.894898,75.685714 46.767346,87.47449 46.767346,100 h 19.25 C 66.017346,63.592858 36.4,33.979592 0.01530612,33.978572 l 0,0 z M 0.03877552,0 V 19.17449 C 44.54796,19.17551 80.77551,55.437756 80.77551,100 H 100 C 100,44.87653 55.15102,0 0.03877552,0 z"></path></svg></a></li>
  
</ul>
  
  
  
  
  
<ul class="subscribe">
  <li><a href="https://github.com/bioops" rel="subscribe-github" title="@bioops on GitHub"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 50,0 C 22.385714,0 0,22.385714 0,50 0,77.614286 22.385714,100 50,100 77.614286,100 100,77.614286 100,50 100,22.385714 77.614286,0 50,0 z m 29.692858,79.692858 c -3.859184,3.859182 -8.351022,6.887754 -13.35,9.00306 -1.27041,0.536736 -2.560204,1.009184 -3.867348,1.415306 v -7.493878 c 0,-3.938774 -1.35102,-6.835714 -4.053062,-8.690816 1.692858,-0.163264 3.24694,-0.390816 4.663266,-0.683672 1.416326,-0.292858 2.913266,-0.716328 4.491838,-1.27041 1.57857,-0.55408 2.994896,-1.213264 4.247958,-1.97755 1.253062,-0.765306 2.458164,-1.758164 3.613266,-2.978572 1.155102,-1.220408 2.12449,-2.604082 2.905102,-4.15 0.780612,-1.545918 1.4,-3.40204 1.855102,-5.566326 0.455102,-2.164286 0.683674,-4.54898 0.683674,-7.153062 0,-5.045918 -1.643878,-9.341836 -4.931634,-12.890816 C 77.44796,33.35 77.285714,29.10204 75.463266,24.512244 l -1.22143,-0.145918 c -0.845918,-0.09796 -2.368366,0.260204 -4.565306,1.07449 -2.196938,0.814286 -4.663264,2.14796 -7.396938,4.004082 -3.87449,-1.07449 -7.893878,-1.611224 -12.061224,-1.611224 -4.19898,0 -8.203062,0.536734 -12.012246,1.611224 -1.72449,-1.17245 -3.361224,-2.139796 -4.907142,-2.905102 C 31.753062,25.77449 30.516326,25.254082 29.587756,24.97653 28.660204,24.7 27.79796,24.528572 27,24.463266 c -0.79796,-0.0653 -1.310204,-0.08062 -1.537756,-0.04898 -0.22755,0.03164 -0.390816,0.0653 -0.487754,0.09796 -1.82347,4.62245 -1.985714,8.87143 -0.487756,12.743878 -3.287754,3.54796 -4.931632,7.844898 -4.931632,12.890816 0,2.604082 0.227552,4.988776 0.683674,7.153062 0.456122,2.164286 1.07449,4.020408 1.855102,5.566326 0.780612,1.545918 1.75,2.929592 2.905102,4.15 1.155102,1.220408 2.360204,2.213266 3.613264,2.978572 1.253062,0.766326 2.669388,1.42449 4.24796,1.97755 1.578572,0.554082 3.07551,0.976532 4.491836,1.27041 1.416328,0.292856 2.970408,0.521428 4.663266,0.683672 -2.669388,1.82347 -4.004082,4.720408 -4.004082,8.690816 v 7.639796 C 36.536734,89.818368 35.083674,89.3 33.656122,88.695918 c -4.99898,-2.115306 -9.490816,-5.143878 -13.35,-9.00306 -3.859184,-3.859184 -6.887754,-8.351022 -9.00306,-13.35 C 9.1163263,61.171428 8.0071428,55.67347 8.0071428,50 c 0,-5.67347 1.1091835,-11.171428 3.2969392,-16.342858 2.115306,-4.998978 5.143878,-9.490816 9.00306,-13.35 3.859184,-3.859182 8.351022,-6.887754 13.35,-9.00306 C 38.828572,9.1163266 44.32653,8.0071428 50,8.0071428 c 5.67347,0 11.171428,1.1091838 16.342858,3.2969392 5,2.115306 9.490816,5.143878 13.35,9.00306 3.859182,3.859184 6.887754,8.351022 9.00306,13.35 2.186736,5.17245 3.295918,10.67041 3.295918,16.342858 0,5.672448 -1.109182,11.171428 -3.296938,16.342858 -2.115306,4.998978 -5.143878,9.490816 -9.00204,13.35 l 0,0 z"></path></svg></a></li>
</ul>
  
  
  
  
  
  
  
    
      <form action="http://google.com/search" method="get">
        <fieldset role="search">
          <input type="hidden" name="sitesearch" value="bioops.info" />
    
          <input class="search" type="text" name="q" results="0" placeholder="Search"/>
        </fieldset>
      </form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div class="blog-index">
  
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/09/next-generation-transcriptome-assembly/">Next-generation Transcriptome Assembly</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-09-08T00:00:00+00:00'><span class='date'><span class='date-month'>Sep</span> <span class='date-day'>8</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/09/next-generation-transcriptome-assembly/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>Nature Review Genetics <a href="http://www.nature.com/nrg/journal/vaop/ncurrent/full/nrg3068.html" target="_blank">doi:10.1038/nrg3068</a></p>

<p>一篇入门文章，值得一看，包括引用的一些review</p>

<p>Abstract</p>

<p>Transcriptomics studies often rely on partial reference transcriptomes that fail to capture the full catalogue of transcripts and their variations. Recent advances in sequencing technologies and assembly algorithms have facilitated the reconstruction of the entire transcriptome by deep RNA sequencing (RNA-seq), even without a reference genome. However, transcriptome assembly from billions of RNA-seq reads, which are often very short, poses a significant informatics challenge. This Review summarizes the recent developments in transcriptome assembly approaches — reference-based, de novo and combined strategies — along with some perspectives on transcriptome assembly in the near future.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/09/bioperl-howtos-translation/">Bioperl HOWTOs中文翻译小组招募</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-09-03T00:00:00+00:00'><span class='date'><span class='date-month'>Sep</span> <span class='date-day'>3</span><span class='date-suffix'>rd</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/09/bioperl-howtos-translation/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>Bioperl HOWTOs中文翻译小组招募</p>

<p><a href="https://skydrive.live.com/redir.aspx?cid=3c28835207352b3c&amp;resid=3C28835207352B3C!154" target="_blank">https://skydrive.live.com/redir.aspx?cid=3c28835207352b3c&amp;resid=3C28835207352B3C!154</a></p>

<p>有兴趣加入的请留hotmail</p>

<p>具体问题可留言询问</p>

<p>谢谢！</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/08/parse-blast/">perl脚本提取BLAST结果中的信息【以tblastn为例】</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-08-27T00:00:00+00:00'><span class='date'><span class='date-month'>Aug</span> <span class='date-day'>27</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/08/parse-blast/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>用blast中<a href="http://www.pangloss.com/wiki/Blast" target="_blank"> -m 8</a>参数可以得到类似的信息，但满足不了所有的需求。用bioperl可以得到任意想要的信息，但我用的cluster上没有bioperl，只好自己写了一个。只是用了一些正则表达式，不适用于其他blast（比如blastn、blastp），但只需要稍微改一下就可以了，暂时将就着用吧。欢迎留言讨论。</p>

<figure class='code'><figcaption><span>parse_blast.pl</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
<span class='line-number'>36</span>
<span class='line-number'>37</span>
<span class='line-number'>38</span>
<span class='line-number'>39</span>
<span class='line-number'>40</span>
<span class='line-number'>41</span>
<span class='line-number'>42</span>
<span class='line-number'>43</span>
<span class='line-number'>44</span>
<span class='line-number'>45</span>
<span class='line-number'>46</span>
<span class='line-number'>47</span>
<span class='line-number'>48</span>
<span class='line-number'>49</span>
<span class='line-number'>50</span>
<span class='line-number'>51</span>
<span class='line-number'>52</span>
<span class='line-number'>53</span>
<span class='line-number'>54</span>
<span class='line-number'>55</span>
<span class='line-number'>56</span>
<span class='line-number'>57</span>
<span class='line-number'>58</span>
<span class='line-number'>59</span>
<span class='line-number'>60</span>
<span class='line-number'>61</span>
<span class='line-number'>62</span>
<span class='line-number'>63</span>
<span class='line-number'>64</span>
<span class='line-number'>65</span>
<span class='line-number'>66</span>
<span class='line-number'>67</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1">#!/usr/bin/perl</span>
</span><span class='line'><span class="c1"># Parse alignments from tBLASTn output file</span>
</span><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="n">warnings</span><span class="p">;</span>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;query\tquery_length\thit\tscore\texpect\tidentity\tquery_strand\tsubject_strand\tcoverage_query\tcoverage_subject\tquery_start\tquery_end\tsubject_start\tsubject_end\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$input</span><span class="o">=</span><span class="nv">$ARGV</span><span class="p">[</span><span class="mi">0</span><span class="p">];</span>
</span><span class='line'><span class="k">my</span> <span class="p">(</span><span class="nv">$eachline</span><span class="p">,</span><span class="nv">$query_name</span><span class="p">,</span><span class="nv">$query_length</span><span class="p">,</span><span class="nv">$hit_name</span><span class="p">,</span><span class="nv">$hsp_num</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="p">(</span><span class="nv">$query_start_temp</span><span class="p">,</span><span class="nv">$query_end_temp</span><span class="p">,</span><span class="nv">$subject_start_temp</span><span class="p">,</span><span class="nv">$subject_end_temp</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="p">(</span><span class="nv">$query_start</span><span class="p">,</span><span class="nv">$query_end</span><span class="p">,</span><span class="nv">$subject_start</span><span class="p">,</span><span class="nv">$subject_end</span><span class="p">,</span><span class="nv">$query_seq</span><span class="p">,</span><span class="nv">$subject_seq</span><span class="p">,</span><span class="nv">$query_seq_temp</span><span class="p">,</span><span class="nv">$subject_seq_temp</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="p">(</span><span class="nv">$score</span><span class="p">,</span><span class="nv">$expect</span><span class="p">,</span><span class="nv">$coverage_query</span><span class="p">,</span><span class="nv">$coverage_subject</span><span class="p">,</span><span class="nv">$identity_query</span><span class="p">,</span><span class="nv">$identity_subject</span><span class="p">,</span><span class="nv">$identity</span><span class="p">,</span><span class="nv">$query_strand</span><span class="p">,</span><span class="nv">$subject_strand</span><span class="p">);</span>
</span><span class='line'><span class="nv">$hsp_num</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="nb">open</span><span class="p">(</span><span class="n">IN</span><span class="p">,</span><span class="s">&quot;&lt;$input&quot;</span><span class="p">)</span> <span class="ow">or</span> <span class="nb">die</span> <span class="p">(</span><span class="s">&quot;no file $input !&quot;</span><span class="p">);</span>
</span><span class='line'><span class="k">while</span> <span class="p">(</span><span class="nb">defined</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=</span><span class="sr">&lt;IN&gt;</span><span class="p">)){</span>
</span><span class='line'>  <span class="nv">$eachline</span><span class="o">=~</span><span class="sr">s/[\r\n]//g</span><span class="p">;</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^Query= /</span><span class="p">){</span>
</span><span class='line'>    <span class="p">(</span><span class="nv">$query_name</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^Query= (\S+)/</span><span class="p">);</span>
</span><span class='line'>    <span class="nv">$eachline</span><span class="o">=</span><span class="sr">&lt;IN&gt;</span><span class="p">;</span>
</span><span class='line'>    <span class="p">(</span><span class="nv">$query_length</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/\((\S+) letters\)/</span><span class="p">);</span>
</span><span class='line'>    <span class="nv">$query_length</span><span class="o">=~</span><span class="sr">s/,//g</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^&gt;/</span><span class="p">){</span>
</span><span class='line'>    <span class="p">(</span><span class="nv">$hit_name</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^&gt;(\S+)/</span><span class="p">);</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^ Score/</span><span class="p">){</span>
</span><span class='line'>    <span class="p">(</span><span class="nv">$score</span><span class="p">,</span><span class="nv">$expect</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> /Score = (.*) bits.*Expect.* = (\S+),/</span><span class="p">);</span>
</span><span class='line'>    <span class="nv">$hsp_num</span><span class="o">++</span><span class="p">;</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$hsp_num</span><span class="o">&gt;</span><span class="mi">1</span><span class="p">){</span>
</span><span class='line'>      <span class="nv">$coverage_query</span><span class="o">=</span><span class="nv">$query_end</span><span class="o">-</span><span class="nv">$query_start</span><span class="o">+</span><span class="mi">1</span><span class="p">;</span>
</span><span class='line'>      <span class="nv">$coverage_subject</span><span class="o">=</span><span class="nv">$subject_end</span><span class="o">-</span><span class="nv">$subject_start</span><span class="o">+</span><span class="mi">1</span><span class="p">;</span>
</span><span class='line'>      <span class="k">print</span> <span class="s">&quot;$coverage_query\t$coverage_subject\t$query_start\t$query_end\t$subject_start\t$subject_end\t$query_seq\t$subject_seq\n&quot;</span><span class="p">;</span>
</span><span class='line'>    <span class="p">}</span>
</span><span class='line'>    <span class="nv">$query_start</span><span class="o">=</span><span class="mi">1000000000</span><span class="p">;</span><span class="nv">$query_end</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'>    <span class="nv">$subject_start</span><span class="o">=</span><span class="mi">1000000000</span><span class="p">;</span><span class="nv">$subject_end</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'>    <span class="nv">$query_seq</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">;</span><span class="nv">$subject_seq</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">;</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;$query_name\t$query_length\t$hit_name\t&quot;</span><span class="p">;</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;$score\t$expect\t&quot;</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^ Identities/</span><span class="p">){</span>
</span><span class='line'>    <span class="p">(</span><span class="nv">$identity_query</span><span class="p">,</span><span class="nv">$identity_subject</span><span class="p">,</span><span class="nv">$identity</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> /Identities = (\d+)\/(\d+) \((\d+)%\)/</span><span class="p">);</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;$identity\t&quot;</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^ Frame = /</span><span class="p">){</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> / Frame = -/</span><span class="p">){</span><span class="nv">$subject_strand</span><span class="o">=</span><span class="s">&#39;Minus&#39;</span><span class="p">;}</span><span class="k">elsif</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> / Frame = +/</span><span class="p">){</span><span class="nv">$subject_strand</span><span class="o">=</span><span class="s">&#39;Plus&#39;</span><span class="p">;}</span>
</span><span class='line'>    <span class="nv">$query_strand</span><span class="o">=</span><span class="s">&#39;Plus&#39;</span><span class="p">;</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;$query_strand\t$subject_strand\t&quot;</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^Query: (\d+)/</span><span class="p">){</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$query_strand</span> <span class="ow">eq</span> <span class="s">&#39;Plus&#39;</span><span class="p">){(</span><span class="nv">$query_start_temp</span><span class="p">,</span><span class="nv">$query_seq_temp</span><span class="p">,</span><span class="nv">$query_end_temp</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> /(\d+)(\D+)(\d+)/</span><span class="p">);}</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$query_strand</span> <span class="ow">eq</span> <span class="s">&#39;Minus&#39;</span><span class="p">){(</span><span class="nv">$query_end_temp</span><span class="p">,</span><span class="nv">$query_seq_temp</span><span class="p">,</span><span class="nv">$query_start_temp</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> /(\d+)(\D+)(\d+)/</span><span class="p">);}</span>
</span><span class='line'>    <span class="nv">$query_seq_temp</span><span class="o">=~</span><span class="sr">s/[ -]//g</span><span class="p">;</span><span class="nv">$query_seq</span><span class="o">.=</span><span class="nv">$query_seq_temp</span><span class="p">;</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$query_start</span><span class="o">&gt;</span><span class="nv">$query_start_temp</span><span class="p">){</span><span class="nv">$query_start</span><span class="o">=</span><span class="nv">$query_start_temp</span><span class="p">;}</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$query_end</span><span class="o">&lt;</span><span class="nv">$query_end_temp</span><span class="p">){</span><span class="nv">$query_end</span><span class="o">=</span><span class="nv">$query_end_temp</span><span class="p">;}</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^Sbjct: (\d+)/</span><span class="p">){</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$subject_strand</span> <span class="ow">eq</span> <span class="s">&#39;Plus&#39;</span><span class="p">){(</span><span class="nv">$subject_start_temp</span><span class="p">,</span><span class="nv">$subject_seq_temp</span><span class="p">,</span><span class="nv">$subject_end_temp</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> /(\d+)(\D+)(\d+)/</span><span class="p">);}</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$subject_strand</span> <span class="ow">eq</span> <span class="s">&#39;Minus&#39;</span><span class="p">){(</span><span class="nv">$subject_end_temp</span><span class="p">,</span><span class="nv">$subject_seq_temp</span><span class="p">,</span><span class="nv">$subject_start_temp</span><span class="p">)</span><span class="o">=</span><span class="p">(</span><span class="nv">$eachline</span> <span class="o">=~</span><span class="sr"> /(\d+)(\D+)(\d+)/</span><span class="p">);}</span>
</span><span class='line'>    <span class="nv">$subject_seq_temp</span><span class="o">=~</span><span class="sr">s/[ -]//g</span><span class="p">;</span><span class="nv">$subject_seq</span><span class="o">.=</span><span class="nv">$subject_seq_temp</span><span class="p">;</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$subject_start</span><span class="o">&gt;</span><span class="nv">$subject_start_temp</span><span class="p">){</span><span class="nv">$subject_start</span><span class="o">=</span><span class="nv">$subject_start_temp</span><span class="p">;}</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$subject_end</span><span class="o">&lt;</span><span class="nv">$subject_end_temp</span><span class="p">){</span><span class="nv">$subject_end</span><span class="o">=</span><span class="nv">$subject_end_temp</span><span class="p">;}</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$eachline</span><span class="o">=~</span><span class="sr">/^Matrix: /</span><span class="p">){</span>
</span><span class='line'>    <span class="nv">$coverage_query</span><span class="o">=</span><span class="nv">$query_end</span><span class="o">-</span><span class="nv">$query_start</span><span class="o">+</span><span class="mi">1</span><span class="p">;</span>
</span><span class='line'>    <span class="nv">$coverage_subject</span><span class="o">=</span><span class="nv">$subject_end</span><span class="o">-</span><span class="nv">$subject_start</span><span class="o">+</span><span class="mi">1</span><span class="p">;</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;$coverage_query\t$coverage_subject\t$query_start\t$query_end\t$subject_start\t$subject_end\t$query_seq\t$subject_seq\n&quot;</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="nb">close</span> <span class="n">IN</span><span class="p">;</span>
</span></code></pre></td></tr></table></div></figure>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/08/linux-command-bioinformatics/">做生物信息常用到的linux命令</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-08-27T00:00:00+00:00'><span class='date'><span class='date-month'>Aug</span> <span class='date-day'>27</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/08/linux-command-bioinformatics/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>1，统计一个序列文件中的序列个数（grep用好了可以非常快捷方便地处理一些数据）
<pre>grep -c '&gt;' seqfile</pre>
2, 查看大文件头几行或最后几行
<pre>head seqfile
tail seqfile</pre>
3，文件行数
<pre>wc -l seqfile</pre>
4，矩阵格式的文件，提取其中的某几列（例如<a href="http://www.pangloss.com/wiki/Blast" target="_blank">blast -m 8</a>）
<pre> cut -f 1,2,11 seq.cblast &gt; seq.abc</pre>
5，awk和sed</p>

<p>先学的perl，后知道awk和sed，认识到很多事情用awk和sed解决比写perl脚本方便多了</p>

<p>比如fastq转换fasta文件：
<pre>awk ‘NR % 4 == 1 || NR % 4 == 2′ myfile.fastq | sed -e ‘s/@/&gt;/’ &gt; myfile.fasta</pre>
6，screen管理远程任务，可以在远程会话断开后继续在后台运行，详见<a href="http://www.path8.net/tn/archives/3746" target="_blank">此文</a>。</p>

<p>7，vi/vim就不必细说了，编程必备。(emacs党自动替换成emacs)</p>

<p>8，暂时就这么多了。好长时间没做过东西了。等用到或者想到了再加。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/08/born-smart/">聪明是天生的么？</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-08-12T00:00:00+00:00'><span class='date'><span class='date-month'>Aug</span> <span class='date-day'>12</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/08/born-smart/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><p>Nature旗下Molecular Psychiatry的一篇<a href="http://www.nature.com/mp/journal/vaop/ncurrent/full/mp201185a.html" target="_blank">文章</a>。</p>
<p>用基因组关联研究（<a href="http://en.wikipedia.org/wiki/Genome-wide_association_study" target="_blank">GWAS</a>）的方法，分析了3511个人的549692个<a href="http://en.wikipedia.org/wiki/Single-nucleotide_polymorphism" target="_blank">SNP</a>与智力的关联程度。结论：40%的crystallized-type intelligence差异和51%的fluid-type intelligence差异可以认为是受遗传因素影响。（crystallized-type intelligence：所具备的知识，和记忆力有关；fluid-type intelligence：解决问题、逻辑思维的能力。 具体解释见<a href="http://en.wikipedia.org/wiki/Fluid_and_crystallized_intelligence" target="_blank">wikipedia</a>）</p>
<p>有兴趣的可读<a href="http://www.nature.com/mp/journal/vaop/ncurrent/full/mp201185a.html" target="_blank">原文</a>或相关评论文章（<a href="http://www.bioon.com/biology/genetics/501012.shtml" target="_blank">生物谷</a>、<a href="http://www.genomeweb.com/blog/born-smart" target="_blank">genomeweb</a>和<a href="http://articles.boston.com/2011-08-11/news/29877046_1_genes-eric-turkheimer-intelligence" target="_blank">boston.com</a>）</p>
<p>&nbsp;</p>
</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/07/structural-variation-detection-by-whole-genome-de-novo-assembly/">Structural Variation Detection by Whole Genome De Novo Assembly</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-07-29T00:00:00+00:00'><span class='date'><span class='date-month'>Jul</span> <span class='date-day'>29</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/07/structural-variation-detection-by-whole-genome-de-novo-assembly/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://www.nature.com/nbt/journal/vaop/ncurrent/full/nbt.1904.html" target="_blank">Structural variation in two human genomes mapped at single-nucleotide resolution by whole genome de novo assembly</a></p>

<p>Here we use whole-genome de novo assembly of second-generation sequencing reads to map structural variation (SV) in an Asian genome and an African genome. Our approach identifies small- and intermediate-size homozygous variants (1–50 kb) including insertions, deletions, inversions and their precise breakpoints, and in contrast to other methods, can resolve complex rearrangements. In total, we identified 277,243 SVs ranging in length from 1–23 kb. Validation using computational and experimental methods suggests that we achieve overall &lt;6% false-positive rate and &lt;10% false-negative rate in genomic regions that can be assembled, which outperforms other methods. Analysis of the SVs in the genomes of 106 individuals sequenced as part of the 1000 Genomes Project suggests that SVs account for a greater fraction of the diversity between individuals than do single-nucleotide polymorphisms (SNPs). These findings demonstrate that whole-genome de novo assembly is a feasible approach to deriving more comprehensive maps of genetic variation.</p>

<p>BGI的文章。BGI，也就是华大，有钱的主，拿钱砸出来好多高IF的文章，此为其一。</p>

<p>通过whole genome de novo assembly来确定Structural variation（SV）。</p>

<p>用拼接好的scaffolds和human reference genome来比对。BGI利用<a href="http://genome.ucsc.edu/cgi-bin/hgBlat?command=start" target="_blank">BLAT</a>和<a href="http://www.bx.psu.edu/~rsharris/lastz/" target="_blank">LASTZ</a>，加上一些自己开发的脚本（scripts）建立了一个SV detection的流程（pipeline），叫做<a href="http://soap.genomics.org.cn/SOAPsv.html" target="_blank">SOAPsv</a>。这篇文章等于是这个pipeline的首次实践。</p>

<p>验证（validation）是一个必要的程序，不像<a href="http://www.1000genomes.org/" target="_blank">千人基因组</a> （<a href="http://www.nature.com/nature/journal/v467/n7319/full/nature09534.html" target="_blank">paper</a>）用实验方法（wet），他们同时用了实验（wet）和电脑计算（dry）两种方法。关于computational方法，是用<a href="http://bio-bwa.sourceforge.net/" target="_blank">BWA</a>或者<a href="http://soap.genomics.org.cn/soapaligner.html" target="_blank">SOAPaligner</a>来把原始的reads align到reference genome上。为什么不能用这个align的结果来做SV detection呢？其他的软件基本都是这样做的。为了“创新”，另辟蹊径，用de novo assembly，但同时还需要align to reference，是不是有点走弯路？。另外一个疑问：如果align的结果和de novo assembly的序列不一样呢？</p>

<p>也和另外两个软件做了比较，<a href="http://breakdancer.sourceforge.net/" target="_blank">BreakDancer</a> （<a href="http://www.nature.com/nmeth/journal/v6/n9/abs/nmeth.1363.html" target="_blank">paper</a>）和<a href="http://www.ebi.ac.uk/~kye/pindel/" target="_blank">Pinde</a>l （<a href="http://bioinformatics.oxfordjournals.org/content/25/21/2865.long" target="_blank">paper</a>），“据说”是比这两个软件的精确性要高。（“similar sensitivity but improved precision”）但是，为什么不和其他软件比较呢？</p>

<p>最后要提另一篇重量级的文章：<a href="http://www.nature.com/nature/journal/v470/n7332/full/nature09708.html" target="_blank">Mapping copy number variation by population-scale genome sequencing</a>，使用了几乎所有的SV detection的软件，并实验验证，还探讨了一些SV发生机制。是一篇集大成的文章，需要仔细研究，有空写篇博客介绍一下。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/07/comparison-assembly-next-generation-sequencing/">Comparison of De Novo Assembly Tools for Next-generation Sequencing</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-07-21T00:00:00+00:00'><span class='date'><span class='date-month'>Jul</span> <span class='date-day'>21</span><span class='date-suffix'>st</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/07/comparison-assembly-next-generation-sequencing/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><strong><a href="http://bioinformatics.oxfordjournals.org/content/early/2011/06/02/bioinformatics.btr319.abstract" target="_blank">Comparative Studies of <em>de novo</em> Assembly Tools for Next-generation Sequencing Technologies</a></strong></p>

<p>Yong Lin, Jian Li, Hui Shen, Lei Zhang, Christopher J Papasian and Hong-Wen Deng</p>

<p><strong>Motivation:</strong> Several new de novo assembly tools have been developed recently to assemble short sequencing reads generated by next-generation sequencing platforms. However, the performance of these tools under various conditions has not been fully investigated, and sufficient information is not currently available for informed decisions to be made regarding the tool that would be most likely to produce the best performance under a specific set of conditions.</p>

<p><strong>Results:</strong> We studied and compared the performance of commonly used de novo assembly tools specifically designed for next-generation sequencing data, including SSAKE, VCAKE, Euler-sr, Edena, Velvet, ABySS and SOAPdenovo. Tools were compared using several performance criteria, including N50 length, sequence cover-age, and assembly accuracy. Various properties of read data, including single-end/paired-end, sequence GC content, depth of coverage and base calling error rates, were investigated for their effects on the performance of different assembly tools. We also compared the computation time and memory usage of these seven tools. Based on the results of our comparison, the relative perform-ance of individual tools are summarized and tentative guidelines for optimal selection of different assembly tools, under different condi-tions, are provided.</p>

<p>拿流行的几个<em>de novo</em> assembly tools for next-generation sequencing做了系统的比较，是一篇很不错的文章。曾经有多少人徘徊于各种assembly软件中，不知道选择哪一种，这篇文章提供了很好的帮助。</p>

<p>作者比较了<a href="http://www.bcgsc.ca/platform/bioinfo/software/ssake" target="_blank">SSAKE</a>, <a href="http://vcake.sourceforge.net/" target="_blank">VCAKE</a>, <a href="http://euler-assembler.ucsd.edu/portal/" target="_blank">Euler-sr</a>, <a href="http://www.genomic.ch/edena.php" target="_blank">Edena</a>, <a href="http://www.ebi.ac.uk/~zerbino/velvet/" target="_blank">Velvet</a>, <a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss" target="_blank">ABySS</a> and <a href="http://soap.genomics.org.cn/soapdenovo.html" target="_blank">SOAPdenovo</a>，（竟然没有<a href="https://www.broadinstitute.org/scientific-community/science/programs/genome-sequencing-and-analysis/computational-rd/computational-" target="_blank">ALLPATHS-LG</a>！）。</p>

<p>先用大到人类（其实最多只用了100Mb序列片段）小到<em>E. coli</em>的不同GC含量的基因组，模拟出single-end和paired-end、以及不同测序深度、不同测序片段长度的短序列片段，然后使用各个软件拼接，缺省设置，在各有2.40GB双核+12GB的8个nodes上运行，然后比较拼接结果：<a href="http://bioops.info/2011/03/n50/" target="_blank">N50</a>、基因组覆盖率（sequence coverage）、拼接错误（assembly error）和base  calling error rate (BCER)等，还有软件运行时的硬件消耗。</p>

<p><strong>最终的结论：（<span style="color: #ff0000;">结论比较复杂，不同条件和要求下，各个软件表现有些差异。总的来看ABySS和SOAPdenovo比较“万金油”。</span>）</strong></p>

<p><a href="http://bioops.info/images/uploads/2011/07/20110721212646.gif"><img class="alignnone size-full wp-image-480" title="20110721212646" src="http://bioops.info/images/uploads/2011/07/20110721212646.gif" alt="" width="1040" height="408" /></a></p>

<p><strong>再看非常重要的硬件消耗（<span style="color: #ff0000;">还是SOAPdenovo和abyss给力啊，看来我眼光不错，我就是用的这两个，还有ALLPATHS-LG！</span>）</strong></p>

<p><a href="http://bioops.info/images/uploads/2011/07/20110721212958.gif"><img class="alignnone size-full wp-image-481" title="20110721212958" src="http://bioops.info/images/uploads/2011/07/20110721212958.gif" alt="" width="436" height="634" /></a></p>

<p><span style="color: #ff0000;"><strong>本人看法：SOAPdenovo能拼接<a href="http://bioops.info/2011/03/scaffold-or-supercontig/" target="_blank">scaffolds</a>，能产生更长的N50，<strong>而ABySS只能拼接到<a href="http://bioops.info/2011/03/contig/" target="_blank">contigs</a>；</strong>但ABySS基于<a href="https://computing.llnl.gov/tutorials/parallel_comp/" target="_blank">MPI并行</a>，对硬件需求低。<strong>ALLPATHS-LG和<strong>SOAPdenovo相似。</strong></strong></strong></span></p>

<p><strong>
</strong></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/07/hello/">Hello 点滴记录</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-07-05T00:00:00+00:00'><span class='date'><span class='date-month'>Jul</span> <span class='date-day'>5</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/07/hello/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>今天发现有人转载我的文章，（虽然我的大部分也是转载别人的），而且有人把我的网址收藏成“有用网站”。哈哈！</p>

<p>为答谢网友，分享一个关于RNA-Seq的高水平blog，<a href="http://rna-seqblog.com/">http://rna-seqblog.com/</a>，还有一个关于人类基因组测序的blog，<a href="http://www.genomesunzipped.org/">http://www.genomesunzipped.org/</a>。</p>

<p>今后要向两位博主看齐，多整点专业的、有深度的文章。（时间啊，就像那个XX，挤一下总会有的！）</p>

<p>今天又被自己鄙视了一下，眼高手低啊。被人问到专业而具体的问题，一问三不知。罪过！还敢号称是做“技术”的！打算拿已发表的Next-generation sequencing data，把这些主要技术流程亲自下手过一遍，<em>de novo</em> assembly、reference assembly、RNA-Seq map、transcriptome assembly、SNP discovery、structural variation/CNV detection等等。</p>

<p>还有一件事要记录一下。有人请我审一篇关于identification of SNPs without the reference genome的文章。惭愧！我自己还没发过一篇一作文章。老板在后面撑腰，说，审吧，你审完了我给把把关。（老板原话：“you can review.  if you like, i can look at your review before you send it in.”）</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/05/perl-multi-threads/">Perl多线程</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-05-23T00:00:00+00:00'><span class='date'><span class='date-month'>May</span> <span class='date-day'>23</span><span class='date-suffix'>rd</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/05/perl-multi-threads/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><strong>先转一下多线程的概念：</strong></p>

<p>线程是一个单一的执行流程，它是所有程序执行过程中最小的控制单位，即能被 CPU 所调度的最小任务单元。线程与进程之间既有联系，又完全不同。简单地说，一个线程必然属于某一个进程，而一个进程包含至少一个或者多个线程。早期的计算机系统一次只能运行一个程序，因此，当有多个程序需要执行的时候，唯一的办法就是让它们排成队，按顺序串行执行。进程的出现打破了这种格局，CPU 资源按时间片被分割开来，分配给不同的进程使用。这样一来，从微观上看进程的执行虽然仍是串行的，但是从宏观上看，不同的程序已经是在并行执行了。如果我们把同样的思想运用到进程上，很自然地就会把进程再细分成更小的执行单位，即线程。由于一个进程又往往需要同时执行多个类似的任务，因此这些被细分的线程之间可以共享相同的代码段，数据段和文件句柄等资源。有了进程，我们可以在一台单 CPU 计算机系统上同时运行 Firefox 和 Microsoft Office Word 等多个程序；有了线程，我们可以使 Firefox 在不同的标签里同时加载多个不同的页面，在 Office Word 里编辑文档的同时进行语法错误检查。因此，线程给我们带来了更高的 CPU 利用率、更快速的程序响应、更经济地资源使用方式和对多 CPU 的体系结构更良好的适应性。关于多线程的详细讲解，可参看：perl 线程模型讲解（<a href="http://it.chinawin.net/softwaredev/article-124a1.html" target="_blank">http://it.chinawin.net/softwaredev/article-124a1.html</a>）</p>

<p>————————————————————–</p>

<p>perl中的多线程模块</p>

<p>5.8以后的版本的多线程模块可参看perldoc（<a href="http://perldoc.perl.org/threads.html" target="_blank">http://perldoc.perl.org/threads.html</a>）</p>

<p>—————————————————————</p>

<p>perl的多线程实例：</p>

<p>涉及语言：Perl<br />
所用模块：threads<br />
模块中的方法： threads-&gt;create（），<br />
创建一个新线程；threads-&gt;join()，<br />
收割已经创建的线程；threads-&gt;list(threads::all)，<br />
返回所有已经创建的线程；threads-&gt;is_joinable()，<br />
返回目标线程是否已经完成，等待join；<br />
其他的在perldoc上了。</p>

<p>——————————————————–</p>

<figure class='code'><figcaption><span>multithread.pl</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="n">threads</span><span class="p">;</span>           <span class="c1">#声明模块</span>
</span><span class='line'><span class="k">use</span> <span class="n">warnings</span><span class="p">;</span><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">print</span> <span class="nb">localtime</span><span class="p">(</span><span class="nb">time</span><span class="p">),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>  <span class="c1">#输出系统时间；</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$j</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$thread</span><span class="p">;</span>
</span><span class='line'><span class="k">while</span><span class="p">()</span>
</span><span class='line'><span class="p">{</span>
</span><span class='line'><span class="k">last</span> <span class="k">if</span><span class="p">(</span><span class="nv">$j</span><span class="o">&gt;=</span><span class="mi">10</span><span class="p">)</span><span class="err">；</span><span class="c1">#这里控制一下任务数量，共10个；</span>
</span><span class='line'><span class="c1">#控制创建的线程数，这里是5，scalar函数返回列表threads-&gt;list()元素的个数；</span>
</span><span class='line'><span class="k">while</span><span class="p">(</span><span class="nb">scalar</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">())</span><span class="o">&lt;</span><span class="mi">5</span><span class="p">)</span>  <span class="p">{</span>    <span class="nv">$j</span><span class="o">++</span><span class="p">;</span> <span class="c1">#创建一个线程，这个线程其实就是调用（引用）函数“ss”； #函数‘ss’包含两个参数（$j和$j）；       threads-&gt;new(\&amp;ss,$j,$j);</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="k">foreach</span> <span class="nv">$thread</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">(</span><span class="nn">threads::</span><span class="n">all</span><span class="p">))</span>
</span><span class='line'><span class="p">{</span>    <span class="k">if</span><span class="p">(</span><span class="nv">$thread</span><span class="o">-&gt;</span><span class="n">is_joinable</span><span class="p">())</span>      <span class="c1">#判断线程是否运行完成；</span>
</span><span class='line'><span class="p">{</span>    <span class="nv">$thread</span><span class="o">-&gt;</span><span class="nb">join</span><span class="p">();</span>
</span><span class='line'><span class="c1">#输出中间结果；</span>
</span><span class='line'><span class="k">print</span> <span class="nb">scalar</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">()),</span><span class="s">&quot;\t$j\t&quot;</span><span class="p">,</span><span class="nb">localtime</span><span class="p">(</span><span class="nb">time</span><span class="p">),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="c1">#join掉剩下的线程（因为在while中当j=10时，还有4个线程正在运行，但是此时程序将退出while循，所以在这里需要额外程序join掉剩下的4个线程）</span>
</span><span class='line'><span class="k">foreach</span> <span class="nv">$thread</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">(</span><span class="nn">threads::</span><span class="n">all</span><span class="p">))</span>
</span><span class='line'><span class="p">{</span>    <span class="nv">$thread</span><span class="o">-&gt;</span><span class="nb">join</span><span class="p">();</span>
</span><span class='line'>     <span class="k">print</span> <span class="nb">scalar</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">()),</span><span class="s">&quot;\t$j\t&quot;</span><span class="p">,</span><span class="nb">localtime</span><span class="p">(</span><span class="nb">time</span><span class="p">),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="c1">#输出程序结束的时间，和程序开始运行时间比较，看程序运行性能；</span>
</span><span class='line'><span class="k">print</span> <span class="nb">localtime</span><span class="p">(</span><span class="nb">time</span><span class="p">),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="c1">#下面就是每个线程引用的函数；</span>
</span><span class='line'><span class="k">sub </span><span class="nf">ss</span><span class="p">()</span>
</span><span class='line'><span class="p">{</span>   <span class="k">my</span> <span class="p">(</span><span class="nv">$t</span><span class="p">,</span><span class="nv">$s</span><span class="p">)</span><span class="o">=</span><span class="nv">@_</span><span class="p">;</span>
</span><span class='line'><span class="nb">sleep</span><span class="p">(</span><span class="nv">$t</span><span class="p">);</span>       <span class="c1">#sleep函数，睡觉；以秒为单位；</span>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;$s\t&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>—————————————————</p>

<p><strong>结果：</strong></p>

<p><strong><a href="http://bioops.info/images/uploads/2011/05/http_imgload.png"><img class="alignnone size-medium wp-image-439" title="http_imgload" src="http://bioops.info/images/uploads/2011/05/http_imgload-300x175.png" alt="" width="300" height="175" /></a>
</strong></p>

<p>第一列表示程序已经完成的任务数，第二列表示正在运行的线程数-1（join掉一个了），第三列表示在收割掉一个线程后新添加的任务，最后一列表示完成一个线程时的系统时间。</p>

<p>————————————————————</p>

<p>多线程运行性能</p>

<p>如果单独运行这10个任务，所需要的时间为：1+2+3+4++10=55s；<br />
采用多线程运行（5个）的话，需要的时间为：54-39=16s；</p>

<p>————————————————————-</p>

<p>运行过程</p>

<p>简要描述一下程序运行过程，以便更深入理解多线程的概念。<br />
程序共要运行10个任务，第一个任务的作用是暂停程序1s（sleep（1））；第二个任务是暂停程序2s（sleep（2））；以此类推，第十个任务是暂停程序10s；<br />
时间(s)       任务<br />
0          1,2,3,4,5（程序初始，5个线程同时运行，需要时间最长的是线程5（5s））<br />
1          2,3,4,5,6(经过1s后，第一个任务已经完成，被join掉，同时添加新任务6)<br />
2          3,4,5,6,7（同上）<br />
3          4,5,6,7,8<br />
4          5,6,7,8,9<br />
5          6,7,8,9,10<br />
7-end      join所有剩下的线程（所有任务都已经添加，程序中while循环退出）</p>

<p>方法$thread-&gt;is_joinable()的作用</p>

<p>前面已经说了，这个方法是用来判断线程是否已经运行完成，处于等待join的状态。当需要处理多个任务，但这些任务完成需要的时间又不一样时，这个方法就显得特别重要。</p>

<p>还是以上面的程序为例。程序初始运行时创建5个线程。第一个线程所需时间最短，为1s。第五个线程所需时间最长5s。如果不适用$thread-&gt;is_joinable()而直接join这五个线程的话，如下：</p>

<pre>foreach $thread(threads->list(threads::all))

{   $thread->join();

}</pre>

<p>结果是：主程序处于等待状态。在1s后，第一个线程被join，主程序依然处于等待，2s后第二个线程被join，主程序等待。知道5s后第五个线程被join，主程序通畅，重新创建下一组线程（5个）。显然这个过程不能最大话利用CPU的资源。当第一个线程被join后，虽然程序中只有4个线程在运行，但是由于主程序处于等待状态，新的线程不会被创建。</p>

<p>最佳的方法就是判断线程是否可以被join。如上面的程序所写的。这样可以保证程序运行过程中始终是5个线程，最大化的利用CPU资源。</p>

<p>——————————————————-</p>

<p><strong>实例</strong></p>

<p>说了这么多，多线程在生物信息中到底可以怎么来运用，下面给一个简单的实例。从KEGG数据库（http://www.genome.jp/kegg/）上搜索同源序列。</p>

<p>所需文件：seqname.txt(用于存放需要搜索的序列KEGG名称)；</p>

<p><strong>源码：</strong></p>

<figure class='code'><figcaption><span>keggmultithread.pl</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
<span class='line-number'>36</span>
<span class='line-number'>37</span>
<span class='line-number'>38</span>
<span class='line-number'>39</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span><span class="k">use</span> <span class="n">warnings</span><span class="p">;</span><span class="k">use</span> <span class="n">threads</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">SOAP::</span><span class="n">Lite</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="n">Cwd</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$path</span><span class="o">=</span><span class="n">getcwd</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$wsdl</span> <span class="o">=</span> <span class="s">&#39;http://soap.genome.jp/KEGG.wsdl&#39;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$serv</span> <span class="o">=</span> <span class="nn">SOAP::</span><span class="n">Lite</span><span class="o">-&gt;</span><span class="n">service</span><span class="p">(</span><span class="nv">$wsdl</span><span class="p">);</span>
</span><span class='line'><span class="nb">open</span><span class="p">(</span><span class="n">F</span><span class="p">,</span><span class="s">&quot;K00006.txt&quot;</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">@names</span><span class="o">=</span><span class="p">;</span>
</span><span class='line'><span class="nb">chomp</span> <span class="nv">@names</span><span class="p">;</span>
</span><span class='line'><span class="nb">close</span><span class="p">(</span><span class="n">F</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$i</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$thread</span><span class="p">;</span>
</span><span class='line'><span class="k">print</span> <span class="nb">localtime</span><span class="p">(</span><span class="nb">time</span><span class="p">);</span>
</span><span class='line'><span class="k">while</span><span class="p">(</span><span class="nv">$i</span><span class="o">&lt;</span><span class="nb">scalar</span><span class="p">(</span><span class="nv">@names</span><span class="p">))</span>  <span class="p">{</span>    <span class="k">while</span><span class="p">(</span><span class="nb">scalar</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">())</span><span class="o">&lt;</span><span class="mi">10</span><span class="p">)</span>  <span class="p">{</span>    <span class="n">threads</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="o">\&amp;</span><span class="n">orgfile</span><span class="p">,</span><span class="nv">$names</span><span class="p">[</span><span class="nv">$i</span><span class="p">]);</span>
</span><span class='line'><span class="nv">$i</span><span class="o">++</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="k">foreach</span> <span class="nv">$thread</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">(</span><span class="nn">threads::</span><span class="n">all</span><span class="p">))</span>
</span><span class='line'><span class="p">{</span>    <span class="k">if</span><span class="p">(</span><span class="nv">$thread</span><span class="o">-&gt;</span><span class="n">is_joinable</span><span class="p">())</span>
</span><span class='line'><span class="p">{</span>    <span class="nv">$thread</span><span class="o">-&gt;</span><span class="nb">join</span><span class="p">();</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="k">foreach</span> <span class="nv">$thread</span><span class="p">(</span><span class="n">threads</span><span class="o">-&gt;</span><span class="n">list</span><span class="p">(</span><span class="nn">threads::</span><span class="n">all</span><span class="p">))</span>
</span><span class='line'><span class="p">{</span>    <span class="nv">$thread</span><span class="o">-&gt;</span><span class="nb">join</span><span class="p">();</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="k">print</span> <span class="nb">localtime</span><span class="p">(</span><span class="nb">time</span><span class="p">);</span>
</span><span class='line'><span class="k">sub </span><span class="nf">orgfile</span>
</span><span class='line'><span class="p">{</span>    <span class="k">my</span><span class="p">(</span><span class="nv">$seq</span><span class="p">)</span><span class="o">=</span><span class="nv">@_</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$offset</span> <span class="o">=</span> <span class="mi">1</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$limit</span> <span class="o">=</span> <span class="mi">100</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$top5</span> <span class="o">=</span> <span class="nv">$serv</span><span class="o">-&gt;</span><span class="n">get_best_neighbors_by_gene</span><span class="p">(</span><span class="nv">$seq</span><span class="p">,</span> <span class="nv">$offset</span><span class="p">,</span><span class="nv">$limit</span><span class="p">);</span>
</span><span class='line'><span class="nv">$seq</span><span class="o">=~</span><span class="sr">s/://</span><span class="p">;</span>
</span><span class='line'><span class="nb">open</span><span class="p">(</span><span class="n">F</span><span class="p">,</span><span class="s">&quot;&gt;$seq.txt&quot;</span><span class="p">);</span>
</span><span class='line'><span class="k">foreach</span> <span class="k">my</span> <span class="nv">$hit</span> <span class="p">(</span><span class="nv">@</span><span class="p">{</span><span class="nv">$top5</span><span class="p">})</span>
</span><span class='line'><span class="p">{</span>    <span class="k">print</span> <span class="n">F</span> <span class="s">&quot;$hit-&gt;{genes_id1}\t$hit-&gt;{genes_id2}\t$hit-&gt;{sw_score}\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="nb">close</span><span class="p">(</span><span class="n">F</span><span class="p">);</span>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;$seq\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

<p>————————————————</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/05/paper-genome-analysis-toolkit/">[Paper]A Framework for Variation Discovery and Genotyping Using Next-generation DNA Sequencing Data</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-05-22T00:00:00+00:00'><span class='date'><span class='date-month'>May</span> <span class='date-day'>22</span><span class='date-suffix'>nd</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/05/paper-genome-analysis-toolkit/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://www.nature.com/ng/journal/v43/n5/full/ng.806.html">http://www.nature.com/ng/journal/v43/n5/full/ng.806.html</a></p>

<p>Recent advances in sequencing technology make it possible to comprehensively catalog genetic variation in population samples, creating a foundation for understanding human disease, ancestry and evolution. The amounts of raw data produced are prodigious, and many computational steps are required to translate this output into high-quality variant calls. We present a unified analytic framework to discover and genotype variation among multiple samples simultaneously that achieves sensitive and specific results across five sequencing technologies and three distinct, canonical experimental designs. Our process includes (i) initial read mapping; (ii) local realignment around indels; (iii) base quality score recalibration; (iv) SNP discovery and genotyping to find all potential variants; and (v) machine learning to separate true segregating variation from machine artifacts common to next-generation sequencing technologies. We here discuss the application of these tools, instantiated in the Genome Analysis Toolkit, to deep whole-genome, whole-exome capture and multi-sample low-pass (~4×) <a title="1000genomes" href="http://www.1000genomes.org/" target="_blank">1000 Genomes Project</a> datasets.</p>
</div>
  
  


    </article>
  
  <div class="pagination">
    
      <a class="prev" href="/page10">&larr; Older</a>
    
    <a href="/blog/archives">Blog Archives</a>
    
    <a class="next" href="/page8">Newer &rarr;</a>
    
  </div>
</div>
<aside class="sidebar">
  
    
  
</aside>

    </div>
  </div>
  <footer role="contentinfo"><!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>

<p>
  Copyright &copy; 2016 - Bioops -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a> | Themed with <a href="https://github.com/lucaslew/whitespace">Whitespace</a></span>
</p>


</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'bioopsgithub';
      
        
        var disqus_script = 'count.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>






<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
