
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Bioops</title>
  <meta name="author" content="Bioops">

  
  <meta name="description" content="http://www.nature.com/nbt/journal/vaop/ncurrent/full/nbt.1883.html Massively parallel sequencing of cDNA has enabled deep and efficient probing of &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bioops.info/page10">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="Bioops" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="/javascripts/libs/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Fjalla+One" rel="stylesheet" type="text/css">
<!--- MathJax Configuration -->
<script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-21431888-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   class="collapse-sidebar sidebar-footer" >
  <header role="banner"><hgroup>
  <h1><a href="/">Bioops</a></h1>
  
    <h2>Bioinformatics=(ACGAAG->AK)+(#!/bin/sh)+(P(A|B)=P(B|A)*P(A)/P(B))</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscribe" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 13.310204,73.332654 C 5.967347,73.332654 0,79.322448 0,86.621428 c 0,7.338776 5.967347,13.262246 13.310204,13.262246 7.370408,0 13.328572,-5.92245 13.328572,-13.262246 0,-7.29898 -5.958164,-13.288774 -13.328572,-13.288774 z M 0.01530612,33.978572 V 53.143878 C 12.493878,53.143878 24.229592,58.02347 33.068368,66.865306 41.894898,75.685714 46.767346,87.47449 46.767346,100 h 19.25 C 66.017346,63.592858 36.4,33.979592 0.01530612,33.978572 l 0,0 z M 0.03877552,0 V 19.17449 C 44.54796,19.17551 80.77551,55.437756 80.77551,100 H 100 C 100,44.87653 55.15102,0 0.03877552,0 z"></path></svg></a></li>
  
</ul>
  
  
  
  
  
<ul class="subscribe">
  <li><a href="https://github.com/bioops" rel="subscribe-github" title="@bioops on GitHub"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 50,0 C 22.385714,0 0,22.385714 0,50 0,77.614286 22.385714,100 50,100 77.614286,100 100,77.614286 100,50 100,22.385714 77.614286,0 50,0 z m 29.692858,79.692858 c -3.859184,3.859182 -8.351022,6.887754 -13.35,9.00306 -1.27041,0.536736 -2.560204,1.009184 -3.867348,1.415306 v -7.493878 c 0,-3.938774 -1.35102,-6.835714 -4.053062,-8.690816 1.692858,-0.163264 3.24694,-0.390816 4.663266,-0.683672 1.416326,-0.292858 2.913266,-0.716328 4.491838,-1.27041 1.57857,-0.55408 2.994896,-1.213264 4.247958,-1.97755 1.253062,-0.765306 2.458164,-1.758164 3.613266,-2.978572 1.155102,-1.220408 2.12449,-2.604082 2.905102,-4.15 0.780612,-1.545918 1.4,-3.40204 1.855102,-5.566326 0.455102,-2.164286 0.683674,-4.54898 0.683674,-7.153062 0,-5.045918 -1.643878,-9.341836 -4.931634,-12.890816 C 77.44796,33.35 77.285714,29.10204 75.463266,24.512244 l -1.22143,-0.145918 c -0.845918,-0.09796 -2.368366,0.260204 -4.565306,1.07449 -2.196938,0.814286 -4.663264,2.14796 -7.396938,4.004082 -3.87449,-1.07449 -7.893878,-1.611224 -12.061224,-1.611224 -4.19898,0 -8.203062,0.536734 -12.012246,1.611224 -1.72449,-1.17245 -3.361224,-2.139796 -4.907142,-2.905102 C 31.753062,25.77449 30.516326,25.254082 29.587756,24.97653 28.660204,24.7 27.79796,24.528572 27,24.463266 c -0.79796,-0.0653 -1.310204,-0.08062 -1.537756,-0.04898 -0.22755,0.03164 -0.390816,0.0653 -0.487754,0.09796 -1.82347,4.62245 -1.985714,8.87143 -0.487756,12.743878 -3.287754,3.54796 -4.931632,7.844898 -4.931632,12.890816 0,2.604082 0.227552,4.988776 0.683674,7.153062 0.456122,2.164286 1.07449,4.020408 1.855102,5.566326 0.780612,1.545918 1.75,2.929592 2.905102,4.15 1.155102,1.220408 2.360204,2.213266 3.613264,2.978572 1.253062,0.766326 2.669388,1.42449 4.24796,1.97755 1.578572,0.554082 3.07551,0.976532 4.491836,1.27041 1.416328,0.292856 2.970408,0.521428 4.663266,0.683672 -2.669388,1.82347 -4.004082,4.720408 -4.004082,8.690816 v 7.639796 C 36.536734,89.818368 35.083674,89.3 33.656122,88.695918 c -4.99898,-2.115306 -9.490816,-5.143878 -13.35,-9.00306 -3.859184,-3.859184 -6.887754,-8.351022 -9.00306,-13.35 C 9.1163263,61.171428 8.0071428,55.67347 8.0071428,50 c 0,-5.67347 1.1091835,-11.171428 3.2969392,-16.342858 2.115306,-4.998978 5.143878,-9.490816 9.00306,-13.35 3.859184,-3.859182 8.351022,-6.887754 13.35,-9.00306 C 38.828572,9.1163266 44.32653,8.0071428 50,8.0071428 c 5.67347,0 11.171428,1.1091838 16.342858,3.2969392 5,2.115306 9.490816,5.143878 13.35,9.00306 3.859182,3.859184 6.887754,8.351022 9.00306,13.35 2.186736,5.17245 3.295918,10.67041 3.295918,16.342858 0,5.672448 -1.109182,11.171428 -3.296938,16.342858 -2.115306,4.998978 -5.143878,9.490816 -9.00204,13.35 l 0,0 z"></path></svg></a></li>
</ul>
  
  
  
  
  
  
  
    
      <form action="http://google.com/search" method="get">
        <fieldset role="search">
          <input type="hidden" name="sitesearch" value="bioops.info" />
    
          <input class="search" type="text" name="q" results="0" placeholder="Search"/>
        </fieldset>
      </form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div class="blog-index">
  
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/05/paper-de-novo-full-length-transcriptome-assembly/">[Paper]Full-length Transcriptome Assembly From RNA-Seq Data Without a Reference Genome</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-05-22T00:00:00+00:00'><span class='date'><span class='date-month'>May</span> <span class='date-day'>22</span><span class='date-suffix'>nd</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/05/paper-de-novo-full-length-transcriptome-assembly/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://www.nature.com/nbt/journal/vaop/ncurrent/full/nbt.1883.html">http://www.nature.com/nbt/journal/vaop/ncurrent/full/nbt.1883.html</a></p>

<p>Massively parallel sequencing of cDNA has enabled deep and efficient probing of transcriptomes. Current approaches for transcript reconstruction from such data often rely on aligning reads to a reference genome, and are thus unsuitable for samples with a partial or missing reference genome. Here we present the Trinity method for <em>de novo</em> assembly of full-length transcripts and evaluate it on samples from fission yeast, mouse and whitefly, whose reference genome is not yet available. By efficiently constructing and analyzing sets of de Bruijn graphs, Trinity fully reconstructs a large fraction of transcripts, including alternatively spliced isoforms and transcripts from recently duplicated genes. Compared with other <em>de novo</em> transcriptome assemblers, Trinity recovers more full-length transcripts across a broad range of expression levels, with a sensitivity similar to methods that rely on genome alignments. Our approach provides a unified solution for transcriptome reconstruction in any sample, especially in the absence of a reference genome.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/05/paper-de-novo-assembly-and-analysis-rna-seq-data/">[Paper]De Novo Assembly and Analysis of RNA-seq Data</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-05-22T00:00:00+00:00'><span class='date'><span class='date-month'>May</span> <span class='date-day'>22</span><span class='date-suffix'>nd</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/05/paper-de-novo-assembly-and-analysis-rna-seq-data/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://www.nature.com/nmeth/journal/v7/n11/full/nmeth.1517.html">http://www.nature.com/nmeth/journal/v7/n11/full/nmeth.1517.html</a></p>

<p>We describe Trans-ABySS, a <em>de novo</em> short-read transcriptome assembly and analysis pipeline that addresses variation in local read densities by assembling read substrings with varying stringencies and then merging the resulting contigs before analysis. Analyzing 7.4 gigabases of 50-base-pair paired-end Illumina reads from an adult mouse liver poly(A) RNA library, we identified known, new and alternative structures in expressed transcripts, and achieved high sensitivity and specificity relative to reference-based assembly methods.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/05/gatk-the-genome-analysis-toolkit/">GATK: The Genome Analysis Toolkit</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-05-22T00:00:00+00:00'><span class='date'><span class='date-month'>May</span> <span class='date-day'>22</span><span class='date-suffix'>nd</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/05/gatk-the-genome-analysis-toolkit/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>The <a title="GATK" href="http://www.broadinstitute.org/gsa/wiki/index.php/The_Genome_Analysis_Toolkit" target="_blank">GATK</a> is a structured software library that makes writing efficient analysis tools using next-generation sequencing data very easy, and second it&#8217;s a suite of tools for working with human medical resequencing projects such as <a title="1000genomes" href="http://www.1000genomes.org/" target="_blank">1000 Genomes</a> and <a title="cancergenome" href="http://cancergenome.nih.gov/" target="_blank">The Cancer Genome Atlas</a>. These tools include things like a depth of coverage analyzers, a quality score recalibrator, a SNP/indel caller and a local realigner.</p>

<p>We aim to work well with both samtools and Picard by providing complementary tools to those available in those two packages. Our SNP calling pipeline (Q score recalibration -&gt; multiple sequence realignment -&gt; snp/index calling) is a particular area of focus, and have been pushing to make these capabilities as general-purpose and powerful as possible. My group&#8217;s mandate is to ensure the success of the human medical resequencing projects we&#8217;ve undertaken at the Broad over the next 2-3 years, which involves providing a robust, production-quality development library that underlies tools for common analysis problems (like SNP calling) as well as enabling exploratory research on NGS data.</p>

<p>Take a look at <a title="File:CBBO 100709 v3.pptx.pdf" href="http://www.broadinstitute.org/gsa/wiki/index.php/File:CBBO_100709_v3.pptx.pdf">File:CBBO 100709 v3.pptx.pdf</a> to view a presentation that provides an introduction to some of the capabilities of the GATK and its application to the <a title="1000genomes" href="http://www.1000genomes.org/" target="_blank">1000 Genomes project</a>.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/05/aa-to-dna-aln-perl/">Transfer Protein Alignment to DNA Alignment by Bioperl</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-05-20T00:00:00+00:00'><span class='date'><span class='date-month'>May</span> <span class='date-day'>20</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/05/aa-to-dna-aln-perl/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><figure class='code'><figcaption><span>aaln_to_daln.pl</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1"># it needs two files as input,</span>
</span><span class='line'><span class="c1"># 1) protein alignment result</span>
</span><span class='line'><span class="c1"># 2)cds sequences of the proteins</span>
</span><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::</span><span class="n">AlignIO</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::Align::</span><span class="n">Utilities</span> <span class="sx">qw(aa_to_dna_aln)</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$alignio</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">AlignIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;NEXUS&#39;</span><span class="p">,</span>
</span><span class='line'><span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="s">&#39;pro.nxs&#39;</span><span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$aa_aln</span> <span class="o">=</span> <span class="nv">$alignio</span><span class="o">-&gt;</span><span class="n">next_aln</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqdata</span><span class="o">=</span> <span class="s">&#39;cds.fa&#39;</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seqio</span> <span class="o">=</span> <span class="k">new</span> <span class="nn">Bio::</span><span class="n">SeqIO</span><span class="p">(</span><span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="nv">$seqdata</span><span class="p">,</span>
</span><span class='line'><span class="o">-</span><span class="nb">format</span> <span class="o">=&gt;</span> <span class="s">&#39;fasta&#39;</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">%seqs</span><span class="p">;</span>
</span><span class='line'><span class="c1"># process each sequence</span>
</span><span class='line'><span class="k">while</span> <span class="p">(</span> <span class="k">my</span> <span class="nv">$seq</span> <span class="o">=</span> <span class="nv">$seqio</span><span class="o">-&gt;</span><span class="n">next_seq</span> <span class="p">)</span> <span class="p">{</span>
</span><span class='line'><span class="nv">$seqs</span><span class="p">{</span><span class="nv">$seq</span><span class="o">-&gt;</span><span class="n">display_id</span><span class="p">}</span> <span class="o">=</span> <span class="nv">$seq</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$dna_aln</span> <span class="o">=</span> <span class="n">aa_to_dna_aln</span><span class="p">(</span><span class="nv">$aa_aln</span><span class="p">,</span> <span class="nv">%seqs</span><span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$out</span> <span class="o">=</span> <span class="nn">Bio::</span><span class="n">AlignIO</span><span class="o">-&gt;</span><span class="k">new</span><span class="p">(</span><span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span> <span class="s">&quot;&gt;cds.phylip&quot;</span> <span class="p">,</span> <span class="s">&#39;-format&#39;</span> <span class="o">=&gt;</span> <span class="s">&#39;NEXUS&#39;</span><span class="p">);</span>
</span><span class='line'><span class="nv">$out</span> <span class="o">-&gt;</span><span class="n">write_aln</span><span class="p">(</span><span class="nv">$dna_aln</span><span class="p">);</span>
</span></code></pre></td></tr></table></div></figure>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/04/kegg/">KEGG: Kyoto Encyclopedia of Genes and Genomes</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-04-05T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>5</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/04/kegg/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>The increasing amount of genome sequence data is the basis for understanding life as a molecular system and for developing medical, pharmaceutical, and other practical applications. Since 1995 we have been developing knowledge-based methods for uncovering higher-order systemic behaviors of the cell and the organism from genomic and molecular information. The reference knowledge is stored in <a href="http://www.genome.jp/kegg/" target="_blank">KEGG</a>, Kyoto Encyclopedia of Genes and Genomes, and associated bioinformatics technologies are being developed both for basic research and practical applications.</p>

<p>Example for searching KEGG with Perl:</p>

<p>See <a href="http://www.genome.jp/kegg/soap/" target="_blank">this</a> for more information.</p>

<figure class='code'><figcaption><span>kegg.pl</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1">#!/usr/bin/perl</span>
</span><span class='line'><span class="k">use</span> <span class="nn">SOAP::</span><span class="n">Lite</span><span class="p">;</span>
</span><span class='line'><span class="nv">$wsdl</span>   <span class="o">=</span> <span class="s">&#39;http://soap.genome.jp/KEGG.wsdl&#39;</span><span class="p">;</span>
</span><span class='line'><span class="nv">$serv</span>   <span class="o">=</span> <span class="nn">SOAP::</span><span class="n">Lite</span><span class="o">-&gt;</span><span class="n">service</span><span class="p">(</span><span class="nv">$wsdl</span><span class="p">);</span>
</span><span class='line'><span class="nv">$offset</span> <span class="o">=</span> <span class="mi">1</span><span class="p">;</span>
</span><span class='line'><span class="nv">$limit</span>  <span class="o">=</span> <span class="mi">5</span><span class="p">;</span>
</span><span class='line'><span class="nv">$top5</span>   <span class="o">=</span> <span class="nv">$serv</span><span class="o">-&gt;</span><span class="n">get_best_neighbors_by_gene</span><span class="p">(</span><span class="s">&#39;vvi:100261203&#39;</span><span class="p">,</span> <span class="nv">$offset</span><span class="p">,</span> <span class="nv">$limit</span><span class="p">);</span>
</span><span class='line'><span class="k">foreach</span> <span class="nv">$hit</span> <span class="p">(</span><span class="nv">@</span><span class="p">{</span><span class="nv">$top5</span><span class="p">})</span> <span class="p">{</span>
</span><span class='line'> <span class="k">print</span> <span class="s">&quot;$hit-&gt;{genes_id1}\t$hit-&gt;{genes_id2}\t$hit-&gt;{sw_score}\n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/04/interpro/">InterPro</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-04-05T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>5</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/04/interpro/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://bioops.info/2011/04/interpro/" target="_blank">InterPro</a> is an integrated database of predictive protein &#8220;signatures&#8221; used for the classification and automatic annotation of proteins and genomes. InterPro classifies sequences at superfamily, family and subfamily levels, predicting the occurrence of functional domains, repeats and important sites. InterPro adds in-depth annotation, including GO terms, to the protein signatures.</p>

<p>Search InterPro with Perl:
<a href="http://www.ebi.ac.uk/Tools/webservices/download_clients/perl/soaplite/iprscan_soaplite.pl" target="_blank"> iprscan_soaplite.pl</a> (SOAP::Lite)
<a href="http://www.ebi.ac.uk/Tools/webservices/download_clients/perl/xmlcompile/iprscan_xmlcompile.pl" target="_blank"> iprscan_xmlcompile.pl</a> (XML::Compile::SOAP)<br />
usage:
<pre>perl iprscan_soaplite.pl Seqfile --email mymai@mymail.com</pre>
see <a href="http://www.ebi.ac.uk/Tools/webservices/services/pfa/iprscan_soap" target="_blank">this</a> for more information.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/04/google-search-perl/">google.pl - Command Line Google Search in a Shell With Perl</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-04-05T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>5</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/04/google-search-perl/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><figure class='code'><figcaption><span>google.pl</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
<span class='line-number'>36</span>
<span class='line-number'>37</span>
<span class='line-number'>38</span>
<span class='line-number'>39</span>
<span class='line-number'>40</span>
<span class='line-number'>41</span>
<span class='line-number'>42</span>
<span class='line-number'>43</span>
<span class='line-number'>44</span>
<span class='line-number'>45</span>
<span class='line-number'>46</span>
<span class='line-number'>47</span>
<span class='line-number'>48</span>
<span class='line-number'>49</span>
<span class='line-number'>50</span>
<span class='line-number'>51</span>
<span class='line-number'>52</span>
<span class='line-number'>53</span>
<span class='line-number'>54</span>
<span class='line-number'>55</span>
<span class='line-number'>56</span>
<span class='line-number'>57</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1">#!/usr/bin/perl</span>
</span><span class='line'><span class="c1"># google.pl - command line tool to search google</span>
</span><span class='line'><span class="c1"># 2009 by Stefan Grothkopp, this code is public domain use it as you wish!</span>
</span><span class='line'>
</span><span class='line'><span class="k">use</span> <span class="nn">LWP::</span><span class="n">Simple</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Term::</span><span class="n">ANSIColor</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># change this to false for b/w output</span>
</span><span class='line'><span class="nv">$use_color</span> <span class="o">=</span> <span class="n">true</span><span class="p">;</span>
</span><span class='line'><span class="c1">#result size: large=8, small=4</span>
</span><span class='line'><span class="nv">$result_size</span> <span class="o">=</span> <span class="s">&quot;large&quot;</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># unescape unicode characters in&quot; content&quot;</span>
</span><span class='line'><span class="k">sub </span><span class="nf">unescape</span> <span class="p">{</span>
</span><span class='line'><span class="k">my</span><span class="p">(</span><span class="nv">$str</span><span class="p">)</span> <span class="o">=</span> <span class="nb">splice</span><span class="p">(</span><span class="nv">@_</span><span class="p">);</span>
</span><span class='line'><span class="nv">$str</span> <span class="o">=~</span> <span class="sr">s/\u(.{4})/chr(hex($1))/eg</span><span class="p">;</span>
</span><span class='line'><span class="k">return</span> <span class="nv">$str</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># number of command line args</span>
</span><span class='line'><span class="nv">$numArgs</span> <span class="o">=</span> <span class="nv">$#ARGV</span> <span class="o">+</span> <span class="mi">1</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">if</span><span class="p">(</span><span class="nv">$numArgs</span> <span class="o">==</span><span class="mi">0</span><span class="p">){</span>
</span><span class='line'><span class="c1"># print usage info if no argument is given</span>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;Usage:n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;$0 &lt;searchterm&gt;n&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span> <span class="k">else</span> <span class="p">{</span>
</span><span class='line'><span class="c1"># use first argument as query string</span>
</span><span class='line'><span class="nv">$q</span> <span class="o">=</span> <span class="nv">$ARGV</span><span class="p">[</span><span class="mi">0</span><span class="p">];</span>
</span><span class='line'><span class="c1"># url encode query string</span>
</span><span class='line'><span class="nv">$q</span> <span class="o">=~</span> <span class="sr">s/([^A-Za-z0-9])/sprintf(&quot;%%%02X&quot;, ord($1))/seg</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># get json encoded search result from google ajax api</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$content</span> <span class="o">=</span> <span class="n">get</span><span class="p">(</span><span class="s">&quot;http://ajax.googleapis.com/ajax/services/search/web?v=1.0&amp;start=0&amp;rsz=$result_size&amp;q=$q&quot;</span><span class="p">);</span>
</span><span class='line'><span class="c1">#Get web page in content</span>
</span><span class='line'><span class="nb">die</span> <span class="s">&quot;get failed&quot;</span> <span class="k">if</span> <span class="p">(</span><span class="o">!</span><span class="nb">defined</span> <span class="nv">$content</span><span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># ugly result parsing (did not want to depend on a parser lib for this quick hack)</span>
</span><span class='line'><span class="k">while</span><span class="p">(</span><span class="nv">$content</span> <span class="o">=~</span> <span class="sr">s/&quot;unescapedUrl&quot;:&quot;([^&quot;]*)&quot;.*?&quot;titleNoFormatting&quot;:&quot;([^&quot;]*)&quot;.*?&quot;content&quot;:&quot;([^&quot;]*)&quot;//</span><span class="p">){</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># those three data items are extrated, there are more</span>
</span><span class='line'><span class="nv">$title</span> <span class="o">=</span> <span class="n">unescape</span><span class="p">(</span><span class="nv">$2</span><span class="p">);</span>
</span><span class='line'><span class="nv">$desc</span> <span class="o">=</span> <span class="n">unescape</span><span class="p">(</span><span class="nv">$3</span><span class="p">);</span>
</span><span class='line'><span class="nv">$url</span> <span class="o">=</span> <span class="n">unescape</span><span class="p">(</span><span class="nv">$1</span><span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="c1"># print result</span>
</span><span class='line'><span class="k">if</span><span class="p">(</span><span class="nv">$use_color</span><span class="p">){</span>
</span><span class='line'><span class="k">print</span> <span class="n">colored</span> <span class="p">[</span><span class="s">&#39;blue&#39;</span><span class="p">],</span> <span class="s">&quot;$titlen&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;$descn&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">print</span> <span class="n">colored</span> <span class="p">[</span><span class="s">&#39;green&#39;</span><span class="p">],</span> <span class="s">&quot;$urlnn&quot;</span><span class="p">;</span>
</span><span class='line'><span class="k">print</span> <span class="n">color</span> <span class="s">&#39;reset&#39;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="k">else</span><span class="p">{</span>
</span><span class='line'><span class="k">print</span> <span class="s">&quot;$titlen$descn$urlnn&quot;</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/04/fedora-vncserver/">Fedora下配置VNC服务器</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-04-05T00:00:00+00:00'><span class='date'><span class='date-month'>Apr</span> <span class='date-day'>5</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/04/fedora-vncserver/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>1、
<pre>rpm -qa | grep vnc</pre>
可以查找到<br />
vnc-1.0.0-2.fc11.i586<br />
vnc-server-1.0.0-2.fc11.i586<br />
否则安装vnc和vncserver</p>

<p>2、
<pre>env</pre>
假设可找到HOME=/home/user用户</p>

<p>3、
<pre>ifconfig</pre>
假设ip=192.168.119.131</p>

<p>4、
<pre>vncserver:10</pre>
指令创建用户，开启10为终端</p>

<p>5、
<pre>vi /home/user/.vnc/xstartup</pre>
修改最后一行tmp&amp; 为gnome&amp;</p>

<p>6、root权限
<pre>vi /etc/sysconfig/vncservers</pre>
修改：<br />
VNCSERVERS=&#8221;10：user&#8221;<br />
VNCSERVERARGS[10]=&#8221;-geometry 800x600&#8221;</p>

<p>7、root权限
<pre>vi /etc/sysconfig/iptables</pre>
添加<br />
-A INPUT -m state &#8211;state NEW -m tcp -p tcp &#8211;dport 5910 -j ACCEPT<br />
-A INPUT -m state &#8211;state NEW -m udp -p udp &#8211;dport 5910 -j ACCEPT</p>

<p>8、修改selinux
<pre>vi /etc/selinux/config</pre>
修改：<br />
SELINUX=disabed</p>

<p>9、root
<pre>service xinetd restart
service iptables restart
chkconfig --level 345 vncserver on//开机自启动
service vncserver restart</pre>
至此服务器已经启动<br />
在其他电脑上<br />
ping 192.169.119.131 没问题<br />
安装vncviewer<br />
输入192.168.119.131:5910<br />
恭喜成功</p>

<p>10、VNC黑屏问题<br />
将黑屏用户的xstartup（一般为：/home/user用户名称/.vnc/xstartup）文件的属性修改为755（rwxr-xr-x）.
<pre>chmod 755 /home/user/.vnc/xstartup</pre>
重启vncserver服务即可！
<pre>service vncserver restart</pre></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/03/mysql-common-command/">MySql常用命令总结</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-03-24T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>24</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/03/mysql-common-command/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><strong>一、连接MYSQL。</strong></p>

<p>格式： mysql -h主机地址 -u用户名 －p用户密码</p>

<p>1、例1：连接到本机上的MYSQL。</p>

<p>首先在打开DOS窗口，然后进入目录 mysqlbin，再键入命令mysql -uroot -p，回车后提示你输密码，如果刚安装好MYSQL，超级用户root是没有密码的，故直接回车即可进入到MYSQL中了，MYSQL的提示符是：mysql&gt;</p>

<p>2、例2：连接到远程主机上的MYSQL。假设远程主机的IP为：110.110.110.110，用户名为root,密码为abcd123。则键入以下命令：</p>

<p>mysql -h110.110.110.110 -uroot -pabcd123</p>

<p>（注:u与root可以不用加空格，其它也一样）</p>

<p>3、退出MYSQL命令： exit （回车）</p>

<p>other:
<pre>
Use yum to install both mysql command line tool and the server:
yum -y install mysql mysql-server
Enable the MySQL service:
/sbin/chkconfig mysqld on
Start the MySQL server:
/sbin/service mysqld start
Set the MySQL root password:
mysqladmin -u root password 'new-password'The quotes around the new password are required.
log on mysql:
mysql -h 127.0.0.1 -u user_name -p
mysqld --basedir=/ --datadir=/usr/local/mysql/data --user=mysql --pid-file=/usr/local/mysql/data/RFSIM.pid --skip-external-locking --port=3306 --socket=/tmp/mysql.sock
</pre></p>

<p>出现&#8221;/var/lib/mysql/mysql.sock“不存在的解决方法<br />
1、<br />
创建/修改文件 /etc/my.cnf，至少增加/修改一行<br />
[mysql]<br />
[client]<br />
socket = /tmp/mysql.sock<br />
#在这里写上你的mysql.sock的正确位置，通常不是在 /tmp/ 下就是在 /var/lib/mysql/ 下<br />
2、<br />
指定IP地址，使用tcp方式连接mysql，而不使用本地sock方式<br />
#mysql -h127.0.0.1 -uuser -ppassword<br />
3、<br />
为 mysql.sock 加个连接，比如说实际的mysql.sock在 /tmp/ 下，则<br />
#ln -s /tmp/mysql.sock /var/lib/mysql/mysql.sock即可</p>

<p><strong>二、修改密码。</strong></p>

<p>格式：mysqladmin -u用户名 -p旧密码 password 新密码</p>

<p>1、例1：给root加个密码ab12。首先在DOS下进入目录mysqlbin，然后键入以下命令</p>

<p>mysqladmin -uroot -password ab12</p>

<p>注：因为开始时root没有密码，所以-p旧密码一项就可以省略了。</p>

<p>2、例2：再将root的密码改为djg345。</p>

<p>mysqladmin -uroot -pab12 password djg345</p>

<p><strong>三、增加新用户。</strong>（注意：和上面不同，下面的因为是MYSQL环境中的命令，所以后面都带一个分号作为命令结束符）</p>

<p>格式：grant select on 数据库.* to 用户名@登录主机 identified by &#8220;密码&#8221;</p>

<p>例1、增加一个用户test1密码为abc，让他可以在任何主机上登录，并对所有数据库有查询、插入、修改、删除的权限。首先用以root用户连入MYSQL，然后键入以下命令：</p>

<p>grant select,insert,update,delete on *.* to test1@&#8221;%&#8221; Identified by &#8220;abc&#8221;;</p>

<p>但例1增加的用户是十分危险的，你想如某个人知道test1的密码，那么他就可以在internet上的任何一台电脑上登录你的mysql数据库并对你的数据可以为所欲为了，解决办法见例2。</p>

<p>例2、增加一个用户test2密码为abc,让他只可以在localhost上登录，并可以对数据库mydb进行查询、插入、修改、删除的操作（localhost指本地主机，即MYSQL数据库所在的那台主机），这样用户即使用知道test2的密码，他也无法从internet上直接访问数据库，只能通过MYSQL主机上的web页来访问了。</p>

<p>grant select,insert,update,delete on mydb.* to test2@localhost identified by &#8220;abc&#8221;;</p>

<p>如果你不想test2有密码，可以再打一个命令将密码消掉。</p>

<p>grant select,insert,update,delete on mydb.* to test2@localhost identified by &#8220;&#8221;;</p>

<p>在上篇我们讲了登录、增加用户、密码更改等问题。下篇我们来看看MYSQL中有关数据库方面的操作。注意：你必须首先登录到MYSQL中，以下操作都是在MYSQL的提示符下进行的，而且每个命令以分号结束。</p>

<p><strong>一、操作技巧</strong></p>

<p>1、如果你打命令时，回车后发现忘记加分号，你无须重打一遍命令，只要打个分号回车就可以了。也就是说你可以把一个完整的命令分成几行来打，完后用分号作结束标志就OK。</p>

<p>2、你可以使用光标上下键调出以前的命令。但以前我用过的一个MYSQL旧版本不支持。我现在用的是mysql-3.23.27-beta-win。</p>

<p><strong>二、显示命令</strong></p>

<p>1、显示数据库列表。</p>

<p>show databases;</p>

<p>刚开始时才两个数据库：mysql和test。mysql库很重要它里面有MYSQL的系统信息，我们改密码和新增用户，实际上就是用这个库进行操作。</p>

<p>2、显示库中的数据表：</p>

<p>use mysql； ／／打开库，学过FOXBASE的一定不会陌生吧</p>

<p>show tables;</p>

<p>3、显示数据表的结构：</p>

<p>describe 表名;</p>

<p>4、建库：</p>

<p>create database 库名;</p>

<p>5、建表：</p>

<p>use 库名；</p>

<p>create table 表名 (字段设定列表)；</p>

<p>6、删库和删表:</p>

<p>drop database 库名;</p>

<p>drop table 表名；</p>

<p>7、将表中记录清空：</p>

<p>delete from 表名;</p>

<p>8、显示表中的记录：</p>

<p>select * from 表名;</p>

<p><strong>三、一个建库和建表以及插入数据的实例</strong></p>

<p>&nbsp;</p>

<p><pre>drop database if exists school; //如果存在SCHOOL则删除 
create database school; //建立库SCHOOL 
use school; //打开库SCHOOL 
create table teacher //建立表TEACHER 
( 
id int(3) auto_increment not null primary key, 
name char(10) not null, 
address varchar(50) default '深圳', 
year date 
); //建表结束 
//以下为插入字段 
insert into teacher values('','glchengang','深圳一中','1976-10-10'); 
insert into teacher values('','jack','深圳一中','1975-12-23');</pre></p>

<p>注：在建表中（1）将ID设为长度为3的数字字段:int(3)并让它每个记录自动加一:auto_increment并不能为空:not null而且让他成为主字段primary key（2）将NAME设为长度为10的字符字段（3）将ADDRESS设为长度50的字符字段，而且缺省值为深圳。varchar和char有什么区别呢，只有等以后的文章再说了。（4）将YEAR设为日期字段。</p>

<p>如果你在mysql提示符键入上面的命令也可以，但不方便调试。你可以将以上命令原样写入一个文本文件中假设为school.sql，然后复制到c:\下，并在DOS状态进入目录\mysql\bin，然后键入以下命令：</p>

<p>mysql -uroot -p密码 &lt; c:\school.sql</p>

<p>如果成功，空出一行无任何显示；如有错误，会有提示。（以上命令已经调试，你只要将//的注释去掉即可使用）。</p>

<p><strong>四、将文本数据转到数据库中</strong></p>

<p>1、文本数据应符合的格式：字段数据之间用tab键隔开，null值用\n来代替.</p>

<p>例：</p>

<p>3 rose 深圳二中 1976-10-10</p>

<p>4 mike 深圳一中 1975-12-23</p>

<p>2、数据传入命令 load data local infile &#8220;文件名&#8221; into table 表名;</p>

<p>注意：你最好将文件复制到\mysql\bin目录下，并且要先用use命令打表所在的库。</p>

<p><strong>五、备份数据库：</strong>（命令在DOS的\mysql\bin目录下执行）</p>

<p>mysqldump &#8211;opt school&gt;school.bbb</p>

<p>注释:将数据库school备份到school.bbb文件，school.bbb是一个文本文件，文件名任取，打开看看你会有新发现。</p>

<p>后记：其实MYSQL的对数据库的操作与其它的SQL类数据库大同小异，您最好找本将SQL的书看看。我在这里只介绍一些基本的，其实我也就只懂这些了，呵呵。最好的MYSQL教程还是&#8221;晏子&#8221;译的&#8221;MYSQL中文参考手册&#8221;不仅免费每个相关网站都有下载，而且它是最权威的。可惜不是象&#8221;PHP4中文手册&#8221;那样是chm的格式，在查找函数命令的时候不太方便。</p>

<p>&nbsp;</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2011/03/linux-system-command/">Linux 查看系统信息命令</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-03-24T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>24</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2011/03/linux-system-command/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><strong>系统</strong>
<pre>uname -a               # 查看内核/操作系统/CPU信息
head -n 1 /etc/issue   # 查看操作系统版本
cat /proc/cpuinfo      # 查看CPU信息
hostname               # 查看计算机名
lspci -tv              # 列出所有PCI设备
lsusb -tv              # 列出所有USB设备
lsmod                  # 列出加载的内核模块
env                    # 查看环境变量</pre>
<strong>资源</strong>
<pre>free -m                # 查看内存使用量和交换区使用量
df -h                  # 查看各分区使用情况
du -sh &lt;目录名&gt;        # 查看指定目录的大小
grep MemTotal /proc/meminfo   # 查看内存总量
grep MemFree /proc/meminfo    # 查看空闲内存量
uptime                 # 查看系统运行时间、用户数、负载
cat /proc/loadavg      # 查看系统负载</pre>
<strong>磁盘和分区</strong>
<pre>mount | column -t      # 查看挂接的分区状态
fdisk -l               # 查看所有分区
swapon -s              # 查看所有交换分区
hdparm -i /dev/hda     # 查看磁盘参数(仅适用于IDE设备)
dmesg | grep IDE       # 查看启动时IDE设备检测状况</pre>
<strong>网络</strong>
<pre>ifconfig               # 查看所有网络接口的属性
iptables -L            # 查看防火墙设置
route -n               # 查看路由表
netstat -lntp          # 查看所有监听端口
netstat -antp          # 查看所有已经建立的连接
netstat -s             # 查看网络统计信息</pre>
<strong>进程</strong>
<pre>ps -ef                 # 查看所有进程
top                    # 实时显示进程状态</pre>
<strong>用户</strong>
<pre>w                      # 查看活动用户
id &lt;用户名&gt;            # 查看指定用户信息
last                   # 查看用户登录日志
cut -d: -f1 /etc/passwd   # 查看系统所有用户
cut -d: -f1 /etc/group    # 查看系统所有组
crontab -l             # 查看当前用户的计划任务</pre>
<strong>服务</strong>
<pre>chkconfig --list       # 列出所有系统服务
chkconfig --list | grep on    # 列出所有启动的系统服务</pre>
<strong>程序</strong>
<pre>rpm -qa                # 查看所有安装的软件包</pre>
另外<a href="http://www.pixelbeat.org/cmdline.html">这里</a>还有非常多的命令， 可以作为参考。</p>
</div>
  
  


    </article>
  
  <div class="pagination">
    
      <a class="prev" href="/page11">&larr; Older</a>
    
    <a href="/blog/archives">Blog Archives</a>
    
    <a class="next" href="/page9">Newer &rarr;</a>
    
  </div>
</div>
<aside class="sidebar">
  
    
  
</aside>

    </div>
  </div>
  <footer role="contentinfo"><!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>

<p>
  Copyright &copy; 2016 - Bioops -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a> | Themed with <a href="https://github.com/lucaslew/whitespace">Whitespace</a></span>
</p>


</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'bioopsgithub';
      
        
        var disqus_script = 'count.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>






<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
