
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Bioops</title>
  <meta name="author" content="Bioops">

  
  <meta name="description" content="Efficient de novo assembly of large genomes using compressed data structures Jared T. Simpson and Richard Durbin Wellcome Trust Sanger Institute, &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bioops.info/page5">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="Bioops" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="/javascripts/libs/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Fjalla+One" rel="stylesheet" type="text/css">
<!--- MathJax Configuration -->
<script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-21431888-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   class="collapse-sidebar sidebar-footer" >
  <header role="banner"><hgroup>
  <h1><a href="/">Bioops</a></h1>
  
    <h2>Bioinformatics=(ACGAAG->AK)+(#!/bin/sh)+(P(A|B)=P(B|A)*P(A)/P(B))</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscribe" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 13.310204,73.332654 C 5.967347,73.332654 0,79.322448 0,86.621428 c 0,7.338776 5.967347,13.262246 13.310204,13.262246 7.370408,0 13.328572,-5.92245 13.328572,-13.262246 0,-7.29898 -5.958164,-13.288774 -13.328572,-13.288774 z M 0.01530612,33.978572 V 53.143878 C 12.493878,53.143878 24.229592,58.02347 33.068368,66.865306 41.894898,75.685714 46.767346,87.47449 46.767346,100 h 19.25 C 66.017346,63.592858 36.4,33.979592 0.01530612,33.978572 l 0,0 z M 0.03877552,0 V 19.17449 C 44.54796,19.17551 80.77551,55.437756 80.77551,100 H 100 C 100,44.87653 55.15102,0 0.03877552,0 z"></path></svg></a></li>
  
</ul>
  
  
  
  
  
<ul class="subscribe">
  <li><a href="https://github.com/bioops" rel="subscribe-github" title="@bioops on GitHub"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 50,0 C 22.385714,0 0,22.385714 0,50 0,77.614286 22.385714,100 50,100 77.614286,100 100,77.614286 100,50 100,22.385714 77.614286,0 50,0 z m 29.692858,79.692858 c -3.859184,3.859182 -8.351022,6.887754 -13.35,9.00306 -1.27041,0.536736 -2.560204,1.009184 -3.867348,1.415306 v -7.493878 c 0,-3.938774 -1.35102,-6.835714 -4.053062,-8.690816 1.692858,-0.163264 3.24694,-0.390816 4.663266,-0.683672 1.416326,-0.292858 2.913266,-0.716328 4.491838,-1.27041 1.57857,-0.55408 2.994896,-1.213264 4.247958,-1.97755 1.253062,-0.765306 2.458164,-1.758164 3.613266,-2.978572 1.155102,-1.220408 2.12449,-2.604082 2.905102,-4.15 0.780612,-1.545918 1.4,-3.40204 1.855102,-5.566326 0.455102,-2.164286 0.683674,-4.54898 0.683674,-7.153062 0,-5.045918 -1.643878,-9.341836 -4.931634,-12.890816 C 77.44796,33.35 77.285714,29.10204 75.463266,24.512244 l -1.22143,-0.145918 c -0.845918,-0.09796 -2.368366,0.260204 -4.565306,1.07449 -2.196938,0.814286 -4.663264,2.14796 -7.396938,4.004082 -3.87449,-1.07449 -7.893878,-1.611224 -12.061224,-1.611224 -4.19898,0 -8.203062,0.536734 -12.012246,1.611224 -1.72449,-1.17245 -3.361224,-2.139796 -4.907142,-2.905102 C 31.753062,25.77449 30.516326,25.254082 29.587756,24.97653 28.660204,24.7 27.79796,24.528572 27,24.463266 c -0.79796,-0.0653 -1.310204,-0.08062 -1.537756,-0.04898 -0.22755,0.03164 -0.390816,0.0653 -0.487754,0.09796 -1.82347,4.62245 -1.985714,8.87143 -0.487756,12.743878 -3.287754,3.54796 -4.931632,7.844898 -4.931632,12.890816 0,2.604082 0.227552,4.988776 0.683674,7.153062 0.456122,2.164286 1.07449,4.020408 1.855102,5.566326 0.780612,1.545918 1.75,2.929592 2.905102,4.15 1.155102,1.220408 2.360204,2.213266 3.613264,2.978572 1.253062,0.766326 2.669388,1.42449 4.24796,1.97755 1.578572,0.554082 3.07551,0.976532 4.491836,1.27041 1.416328,0.292856 2.970408,0.521428 4.663266,0.683672 -2.669388,1.82347 -4.004082,4.720408 -4.004082,8.690816 v 7.639796 C 36.536734,89.818368 35.083674,89.3 33.656122,88.695918 c -4.99898,-2.115306 -9.490816,-5.143878 -13.35,-9.00306 -3.859184,-3.859184 -6.887754,-8.351022 -9.00306,-13.35 C 9.1163263,61.171428 8.0071428,55.67347 8.0071428,50 c 0,-5.67347 1.1091835,-11.171428 3.2969392,-16.342858 2.115306,-4.998978 5.143878,-9.490816 9.00306,-13.35 3.859184,-3.859182 8.351022,-6.887754 13.35,-9.00306 C 38.828572,9.1163266 44.32653,8.0071428 50,8.0071428 c 5.67347,0 11.171428,1.1091838 16.342858,3.2969392 5,2.115306 9.490816,5.143878 13.35,9.00306 3.859182,3.859184 6.887754,8.351022 9.00306,13.35 2.186736,5.17245 3.295918,10.67041 3.295918,16.342858 0,5.672448 -1.109182,11.171428 -3.296938,16.342858 -2.115306,4.998978 -5.143878,9.490816 -9.00204,13.35 l 0,0 z"></path></svg></a></li>
</ul>
  
  
  
  
  
  
  
    
      <form action="http://google.com/search" method="get">
        <fieldset role="search">
          <input type="hidden" name="sitesearch" value="bioops.info" />
    
          <input class="search" type="text" name="q" results="0" placeholder="Search"/>
        </fieldset>
      </form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div class="blog-index">
  
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/03/paper-sga-assembler/">[Paper] Efficient De Novo Assembly of Large Genomes Using Compressed Data Structures</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-03-01T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>1</span><span class='date-suffix'>st</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/03/paper-sga-assembler/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><strong>Efficient de novo assembly of large genomes using compressed data structures</strong></p>

<p>Jared T. Simpson and Richard Durbin</p>

<p>Wellcome Trust Sanger Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SA, United Kingdom</p>

<p><strong>Abstract</strong></p>

<p>De novo genome sequence assembly is important both to generate new sequence assemblies for previously uncharacterized genomes and to identify the genome sequence of individuals in a reference-unbiased way. We present memory efficient data structures and algorithms for assembly using the FM-index derived from the compressed Burrows-Wheeler transform, and a new assembler based on these called <a href="https://github.com/jts/sga" target="_blank">SGA </a>(String Graph Assembler). We describe algorithms to error-correct, assemble, and scaffold large sets of sequence data. SGA uses the overlap-based string graph model of assembly, unlike most de novo assemblers that rely on de Bruijn graphs, and is simply parallelizable. We demonstrate the error correction and assembly performance of SGA on 1.2 billion sequence reads from a human genome, which we are able to assemble using 54 GB of memory. The resulting contigs are highly accurate and contiguous, while covering 95% of the reference genome (excluding contigs &lt;200 bp in length). Because of the low memory requirements and parallelization without requiring inter-process communication, SGA provides the first practical assembler to our knowledge for a mammalian-sized genome on a low-end computing cluster.</p>

<p><a href="http://genome.cshlp.org/content/22/3/549.full?rss=1" target="_blank">fulltext</a></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/03/paper-gage-a-critical-evaluation-of-genome-assemblies-and-assembly-algorithms/">[Paper] GAGE: A Critical Evaluation of Genome Assemblies and Assembly Algorithms</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-03-01T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>1</span><span class='date-suffix'>st</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/03/paper-gage-a-critical-evaluation-of-genome-assemblies-and-assembly-algorithms/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><strong>GAGE: A critical evaluation of genome assemblies and assembly algorithms</strong></p>

<p>Abstract</p>

<p>New sequencing technology has dramatically altered the landscape of whole-genome sequencing, allowing scientists to initiate numerous projects to decode the genomes of previously unsequenced organisms. The lowest-cost technology can generate deep coverage of most species, including mammals, in just a few days. The sequence data generated by one of these projects consist of millions or billions of short DNA sequences (reads) that range from 50 to 150 nt in length. These sequences must then be assembled de novo before most genome analyses can begin. Unfortunately, genome assembly remains a very difficult problem, made more difficult by shorter reads and unreliable long-range linking information. In this study, we evaluated several of the leading de novo assembly algorithms on four different short-read data sets, all generated by Illumina sequencers. Our results describe the relative performance of the different assemblers as well as other significant differences in assembly difficulty that appear to be inherent in the genomes themselves. Three overarching conclusions are apparent: first, that data quality, rather than the assembler itself, has a dramatic effect on the quality of an assembled genome; second, that the degree of contiguity of an assembly varies enormously among different assemblers and different genomes; and third, that the correctness of an assembly also varies widely and is not well correlated with statistics on contiguity. To enable others to replicate our results, all of our data and methods are freely available, as are all assemblers used in this study.</p>

<p><a href="http://genome.cshlp.org/content/22/3/557.full?rss=1" target="_blank">fulltext</a></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/03/paper-bgi-single-cell-sequencing-cell-papers/">[Paper] BGI单细胞基因组测序</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-03-01T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>1</span><span class='date-suffix'>st</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/03/paper-bgi-single-cell-sequencing-cell-papers/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>转自<a href="http://www.weibo.com/hibgi" target="_blank">BGI华大基因（微博）</a></p>

<p>【Cell文章“双生子”诞生】华大基因在肿瘤研究上获得突破性进展，于国际著名学术期刊《细胞》上同时发表两篇研究论文。研究团队研发了一种解析单细胞基因组的新方法，并将该方法应用于原发性血小板增多症ET和肾透明细胞癌ccRCC的肿瘤内部遗传特征研究。</p>

<p>1. 研究人员发现此次研究的肾癌病例并非由常见的两个突变基因VHL和PBRM1导致，提示在癌症分析和诊断过程中进行个性化治疗的重要性。此外还发现了一些与此肾癌发生相关的重要功能基因。<a href="http://www.cell.com/abstract/S0092-8674(12)00225-5" target="_blank"> Single-Cell Exome Sequencing Reveals Single-Nucleotide Mutation Characteristics of a Kidney Tumor</a></p>

<p>2为了探究肿瘤的演化及遗传特性，研究人员对取自一例典型的JAK2阴性ET病人的90个单细胞进行了全外显子测序，揭示了此ET在肿瘤发生中遵循单克隆演化模型，并鉴定了一些与ET的发生、发展相关的突变基因。<a href="http://www.cell.com/abstract/S0092-8674(12)00228-0" target="_blank">Single-Cell Exome Sequencing and Monoclonal Evolution of a JAK2-Negative Myeloproliferative Neoplasm</a></p>

<p>News from BGI website: <a href="http://en.genomics.cn/navigation/show_news.action?newsContent.id=8990" target="_blank">Cell highlights BGI studies on single-cell sequencing, leading to a new era of cancer research</a></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/02/sciencemag-news-bgi/">新一期Science上有关BGI的通讯</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-02-02T00:00:00+00:00'><span class='date'><span class='date-month'>Feb</span> <span class='date-day'>2</span><span class='date-suffix'>nd</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/02/sciencemag-news-bgi/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>看看BGI在政府支持下，如何起家、发展并在全球扩张的。</p>

<p><a href="http://www.sciencemag.org/content/335/6068/516.full" target="_blank">http://www.sciencemag.org/content/335/6068/516.full</a></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/01/phd-thesis-topic-decided/">转方向啦</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-01-29T00:00:00+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>29</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/01/phd-thesis-topic-decided/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>PhD研究的基本方向初步定下来了。会继续关注测序技术，不过会更关注基因组、转录组、表观遗传组、蛋白质组、代谢组等组学数据的获取，处理和分析，特别是对如何整合多个组学、组学间调控关系的分析。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/01/life-technologies-illumina-new-machine-competitio/">Life Technologies和Illumina竞相发布新测序机器</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-01-11T00:00:00+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>11</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/01/life-technologies-illumina-new-machine-competitio/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>1月10日，Life Technologies（Ion Torrent）发布Ion Proton™ Sequencer。就在同一天，Illumina也发布Hiseq 2500。测序技术的竞争已经进入白热化，都在瞄准1000美元测个体基因组的目标。Ion Torrent说年末就能达到，而Illumina反击说他们在夏天就能实现。让我们拭目以待吧。</p>

<p>（目前的形势是Illumina一直领先，从reads长度、测序通量、价钱等方面都要优于Ion Torrent。但Ion Torrent采用的是不同的测序技术（Semiconductor Sequencing），所有测序进程都在一个芯片上完成。如果要升级机器，只需要更换新的芯片即可。Ion Torrent很有可能后来居上。）</p>

<p>（最近比较忙，先发布出来，以后再慢慢分析，有兴趣的直接看英文报道。）</p>

<p><a href="http://www.genomeweb.com/sequencing/life-tech-plans-new-ion-torrent-instrument-deliver-whole-genomes-hours-1k" target="_blank">http://www.genomeweb.com/sequencing/life-tech-plans-new-ion-torrent-instrument-deliver-whole-genomes-hours-1k</a></p>

<p><a href="http://www.lifetechnologies.com/us/en/home/about-us/news-gallery/press-releases/2012/life-techologies-itroduces-the-bechtop-io-proto.html" target="_blank">http://www.lifetechnologies.com/us/en/home/about-us/news-gallery/press-releases/2012/life-techologies-itroduces-the-bechtop-io-proto.html</a></p>

<p><a href="http://www.genomeweb.com/sequencing/illumina-introduces-one-day-genome-sequencer-new-business-groups" target="_blank">http://www.genomeweb.com/sequencing/illumina-introduces-one-day-genome-sequencer-new-business-groups</a></p>

<p><a href="http://pathogenomics.bham.ac.uk/blog/2012/01/ion-torrent-proton-the-chip-is-not-the-machine/ " target="_blank">http://pathogenomics.bham.ac.uk/blog/2012/01/ion-torrent-proton-the-chip-is-not-the-machine/</a></p>

<p><a href="http://blogs.nature.com/news/2012/01/the-1000-genome-are-we-there-yet.html" target="_blank">http://blogs.nature.com/news/2012/01/the-1000-genome-are-we-there-yet.html</a></p>

<p><a href="http://core-genomics.blogspot.com/2012/01/agbt-previews.html" target="_blank">http://core-genomics.blogspot.com/2012/01/agbt-previews.html</a></p>

<p><a href="http://omicsomics.blogspot.com/2012/01/sequencing-technology-fireworks.html" target="_blank">http://omicsomics.blogspot.com/2012/01/sequencing-technology-fireworks.html</a></p>

<p><a href="http://pathogenomics.bham.ac.uk/blog/2012/01/ion-torrent-proton-the-chip-is-not-the-machine/" target="_blank">http://pathogenomics.bham.ac.uk/blog/2012/01/ion-torrent-proton-the-chip-is-not-the-machine/</a></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/01/crack-wifi-wpa-password-reaver/">用Reaver破解WPA加密的WIFI密码</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-01-09T00:00:00+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>9</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/01/crack-wifi-wpa-password-reaver/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><a href="http://lifehacker.com/5873407/how-to-crack-a-wi+fi-networks-wpa-password-with-reaver" target="_blank">How to Crack a Wi-Fi Network’s WPA Password with Reaver</a></p>

<p>不用字典，而是利用一个WPS的漏洞进行暴力破解，时长2-10小时不等。</p>

<p>很早就有关于此漏洞的理论研究（<a href="http://cnbeta.com/articles/168498.htm" target="_blank">参考</a>），Reaver将其变成现实。</p>

<p>ENJOY!!!</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/01/cortex-nature-paper/">[Paper] De Novo Assembly and Genotyping of Variants Using Colored De Bruijn Graphs</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-01-09T00:00:00+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>9</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/01/cortex-nature-paper/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p><strong><a href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.1028.html" target="_blank"><em>De novo</em> assembly and genotyping of variants using colored de Bruijn graphs</a></strong></p>

<p>Detecting genetic variants that are highly divergent from a reference sequence remains a major challenge in genome sequencing. We introduce <em>de novo</em>assembly algorithms using colored de Bruijn graphs for detecting and genotyping simple and complex genetic variants in an individual or population. We provide an efficient software implementation, <a href="http://cortexassembler.sourceforge.net/" target="_blank">Cortex</a>, the first <em>de novo </em>assembler capable of assembling multiple eukaryotic genomes simultaneously. Four applications of Cortex are presented. First, we detect and validate both simple and complex structural variations in a high-coverage human genome. Second, we identify more than 3 Mb of sequence absent from the human reference genome, in pooled low-coverage population sequence data from the 1000 Genomes Project. Third, we show how population information from ten chimpanzees enables accurate variant calls without a reference sequence. Last, we estimate classical human leukocyte antigen (HLA) genotypes at <em>HLA-B</em>, the most variable gene in the human genome.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/01/bgi-gpu-sequencing/">华大基因利用GPU加速基因组数据分析</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-01-09T00:00:00+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>9</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/01/bgi-gpu-sequencing/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>基因组数据爆炸式地增长，带来计算上极大的困难，让超级计算机也捉襟见肘。要解决起来，一方面是软件，包括算法的优化和数据压缩技术等；另一方面是硬件，GPU计算、云计算等等。</p>

<p>前一段时间看到华大网站上SOAP系列的软件都了GPU的版本，现在wired出了篇通讯，报道华大用GPU加速基因组数据分析。面向大众的，没有什么专业术语，有兴趣的看看。</p>

<p><a href="http://www.wired.com/wiredenterprise/2012/01/genomes-and-gpus/all/1" target="_blank">Chinese Crunch Human Genome With Videogame Chips</a></p>

<p>主要讲的是：世界最大基因测序中心<a href="http://www.genomics.cn/navigation/index.action">华大基因</a>（BGI）和NVIDIA合作将基因分析工具移植到NVIDIA的GPU架构，将基因组分析时间降低一个数量级。以前分析个人的基因组数据需要4天，现在缩短到6小时。</p>

<p>另外：genome research新出一篇<a href="http://genome.cshlp.org/content/early/2012/01/05/gr.131383.111.full.pdf+html?rss=1" target="_blank">论文</a>，说是序列质量比不同拼接软件对拼接结果影响要大。顿时让我联想到种种关于下一代测序技术的论文和报道。下一代测序技术的开发和应用，看似如火如荼，其实非常不成熟，问题很多，还很难用此测序技术做出突破性的新发现。不过，话说回来，不成熟，有问题，很多人都在关注，正好说明了可以做一些东西，且能灌几篇高IF的水文（虽然在科学上不算什么新突破）。</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/2012/01/predictions-supercomputing-2012/">展望2012超级计算技术</a></h1>
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2012-01-04T00:00:00+00:00'><span class='date'><span class='date-month'>Jan</span> <span class='date-day'>4</span><span class='date-suffix'>th</span>, <span class='date-year'>2012</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="/2012/01/predictions-supercomputing-2012/#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


  <div class="entry-content"><p>By Eng Lim Goh, Translated by <a href="http://bioops.info" target="_blank">Bioops</a></p>

<p>只做了简要翻译，有兴趣请查看<a href="http://www.bio-itworld.com/2012/01/04/predictions-supercomputing-2012.html" target="_blank">原文</a>。</p>

<p>十个关于2012年高性能计算（high performance computing）的预测：</p>

<p>1 个人医疗成真。个人基因组测序话费变得和磁共振成像一样便宜。药物设计和开方会根据个人基因组的不同而因地制宜（译者注：不敢苟同，前一句可能成真，后一句还有很长的路要走。）</p>

<p>2 预报恶劣天气更加精准。气象学上对于恶劣天气方位和时间的预测上有了长足的发展。</p>

<p>3 共享式内存系统会有预想不到的发现。共享式系统仍会是下年的主流。随着计算能力的扩大，必然会有新的预想不到的发现（“expect to find the unexpected”）</p>

<p>4 分布式系统的快速增长。分布式系统将成为各个需要处理大批量非结构性数据的企业和政府机构的首选。</p>

<p>5. 对协处理器的兴趣增加。有那么一群人会对协处理器感兴趣，并关注如何平衡协处理器所带来的性能增强与编程困难。（译者注：目前很火的GPU就是一个例子。）</p>

<p>6 开放式的云计算服务。目前使用云计算的主要是政府机构，但下年，类似Amazon、Facebook和微软等提供公用云计算的公司会越来越多。</p>

<p>7 数据中心的绿色化。数据中心低能耗，冷却水的重新利用，等等。</p>

<p>8 计算流体力学（computational fluid dynamics (CFD)）。通过计算机的模拟，可以对小到自行车，大到航天飞机进行流体力学的研究。同时CFD也被用于医学上对血液流动的模拟。下一年CFD将在系统生物学方面大展身手。</p>

<p>9 全球变暖的预测更加精确。更准确得预测下一个百年全球气候变化，并指出全球不同地区的关键“热点”（hot spots）</p>

<p>10 关于宇宙的部分理论将被证明。霍金关于黑洞和宇宙形成的关键理论会被实验和观测证实。（译者注：有点瞎扯淡）</p>

<p>在未来，我们更关注的是使用高性能计算技术能给其他学科带来新的发现和发展，而不只是用计算机做分析和写报告。</p>
</div>
  
  


    </article>
  
  <div class="pagination">
    
      <a class="prev" href="/page6">&larr; Older</a>
    
    <a href="/blog/archives">Blog Archives</a>
    
    <a class="next" href="/page4">Newer &rarr;</a>
    
  </div>
</div>
<aside class="sidebar">
  
    
  
</aside>

    </div>
  </div>
  <footer role="contentinfo"><!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>

<p>
  Copyright &copy; 2016 - Bioops -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a> | Themed with <a href="https://github.com/lucaslew/whitespace">Whitespace</a></span>
</p>


</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'bioopsgithub';
      
        
        var disqus_script = 'count.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>






<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
