<<<<<<< HEAD:source/_posts/2011-02-17-synteny-relevant-links.html
---
layout: post
title: Synteny relevant links
categories:
- Bioinformatics
- Genomics
- Software
tags:
- Bioinformatics
- Comparative Genomics
- Evolution
- Genomics
- Software
- Synteny
published: true
comments: true
---
<p>Copy from <a href="http://www.symapdb.org/docs/links.html" target="_blank">http://www.symapdb.org/docs/links.html</a></p>
=======

<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Synteny Relevant Links - Bioops</title>
  <meta name="author" content="Bioops">

  
  <meta name="description" content="Copy from http://www.symapdb.org/docs/links.html Contents: Information On-line comparative sequenced genome databases Synteny downloadable programs A &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bioops.info/2011/02/synteny-relevant-links">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="Bioops" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="/javascripts/libs/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Fjalla+One" rel="stylesheet" type="text/css">
<!--- MathJax Configuration -->
<script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-21431888-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   class="collapse-sidebar sidebar-footer" >
  <header role="banner"><hgroup>
  <h1><a href="/">Bioops</a></h1>
  
    <h2>Bioinformatics=(ACGAAG->AK)+(#!/bin/sh)+(P(A|B)=P(B|A)*P(A)/P(B))</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscribe" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 13.310204,73.332654 C 5.967347,73.332654 0,79.322448 0,86.621428 c 0,7.338776 5.967347,13.262246 13.310204,13.262246 7.370408,0 13.328572,-5.92245 13.328572,-13.262246 0,-7.29898 -5.958164,-13.288774 -13.328572,-13.288774 z M 0.01530612,33.978572 V 53.143878 C 12.493878,53.143878 24.229592,58.02347 33.068368,66.865306 41.894898,75.685714 46.767346,87.47449 46.767346,100 h 19.25 C 66.017346,63.592858 36.4,33.979592 0.01530612,33.978572 l 0,0 z M 0.03877552,0 V 19.17449 C 44.54796,19.17551 80.77551,55.437756 80.77551,100 H 100 C 100,44.87653 55.15102,0 0.03877552,0 z"></path></svg></a></li>
  
</ul>
  
  
  
  
  
<ul class="subscribe">
  <li><a href="https://github.com/bioops" rel="subscribe-github" title="@bioops on GitHub"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 50,0 C 22.385714,0 0,22.385714 0,50 0,77.614286 22.385714,100 50,100 77.614286,100 100,77.614286 100,50 100,22.385714 77.614286,0 50,0 z m 29.692858,79.692858 c -3.859184,3.859182 -8.351022,6.887754 -13.35,9.00306 -1.27041,0.536736 -2.560204,1.009184 -3.867348,1.415306 v -7.493878 c 0,-3.938774 -1.35102,-6.835714 -4.053062,-8.690816 1.692858,-0.163264 3.24694,-0.390816 4.663266,-0.683672 1.416326,-0.292858 2.913266,-0.716328 4.491838,-1.27041 1.57857,-0.55408 2.994896,-1.213264 4.247958,-1.97755 1.253062,-0.765306 2.458164,-1.758164 3.613266,-2.978572 1.155102,-1.220408 2.12449,-2.604082 2.905102,-4.15 0.780612,-1.545918 1.4,-3.40204 1.855102,-5.566326 0.455102,-2.164286 0.683674,-4.54898 0.683674,-7.153062 0,-5.045918 -1.643878,-9.341836 -4.931634,-12.890816 C 77.44796,33.35 77.285714,29.10204 75.463266,24.512244 l -1.22143,-0.145918 c -0.845918,-0.09796 -2.368366,0.260204 -4.565306,1.07449 -2.196938,0.814286 -4.663264,2.14796 -7.396938,4.004082 -3.87449,-1.07449 -7.893878,-1.611224 -12.061224,-1.611224 -4.19898,0 -8.203062,0.536734 -12.012246,1.611224 -1.72449,-1.17245 -3.361224,-2.139796 -4.907142,-2.905102 C 31.753062,25.77449 30.516326,25.254082 29.587756,24.97653 28.660204,24.7 27.79796,24.528572 27,24.463266 c -0.79796,-0.0653 -1.310204,-0.08062 -1.537756,-0.04898 -0.22755,0.03164 -0.390816,0.0653 -0.487754,0.09796 -1.82347,4.62245 -1.985714,8.87143 -0.487756,12.743878 -3.287754,3.54796 -4.931632,7.844898 -4.931632,12.890816 0,2.604082 0.227552,4.988776 0.683674,7.153062 0.456122,2.164286 1.07449,4.020408 1.855102,5.566326 0.780612,1.545918 1.75,2.929592 2.905102,4.15 1.155102,1.220408 2.360204,2.213266 3.613264,2.978572 1.253062,0.766326 2.669388,1.42449 4.24796,1.97755 1.578572,0.554082 3.07551,0.976532 4.491836,1.27041 1.416328,0.292856 2.970408,0.521428 4.663266,0.683672 -2.669388,1.82347 -4.004082,4.720408 -4.004082,8.690816 v 7.639796 C 36.536734,89.818368 35.083674,89.3 33.656122,88.695918 c -4.99898,-2.115306 -9.490816,-5.143878 -13.35,-9.00306 -3.859184,-3.859184 -6.887754,-8.351022 -9.00306,-13.35 C 9.1163263,61.171428 8.0071428,55.67347 8.0071428,50 c 0,-5.67347 1.1091835,-11.171428 3.2969392,-16.342858 2.115306,-4.998978 5.143878,-9.490816 9.00306,-13.35 3.859184,-3.859182 8.351022,-6.887754 13.35,-9.00306 C 38.828572,9.1163266 44.32653,8.0071428 50,8.0071428 c 5.67347,0 11.171428,1.1091838 16.342858,3.2969392 5,2.115306 9.490816,5.143878 13.35,9.00306 3.859182,3.859184 6.887754,8.351022 9.00306,13.35 2.186736,5.17245 3.295918,10.67041 3.295918,16.342858 0,5.672448 -1.109182,11.171428 -3.296938,16.342858 -2.115306,4.998978 -5.143878,9.490816 -9.00204,13.35 l 0,0 z"></path></svg></a></li>
</ul>
  
  
  
  
  
  
  
    
      <form action="http://google.com/search" method="get">
        <fieldset role="search">
          <input type="hidden" name="sitesearch" value="bioops.info" />
    
          <input class="search" type="text" name="q" results="0" placeholder="Search"/>
        </fieldset>
      </form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      
        <h1 class="entry-title">Synteny Relevant Links</h1>
      
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-02-17T00:00:00+00:00'><span class='date'><span class='date-month'>Feb</span> <span class='date-day'>17</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


<div class="entry-content"><p>Copy from <a href="http://www.symapdb.org/docs/links.html" target="_blank">http://www.symapdb.org/docs/links.html</a></p>
>>>>>>> d80cd8fa3e1fb5461144707ba04f7385ec6726a7:2011/02/synteny-relevant-links/index.html

<p><strong>Contents:</strong>
<ul>
	<li>Information</li>
	<li>On-line comparative sequenced genome databases</li>
	<li>Synteny downloadable programs</li>
	<li>A few multiple alignment programs</li>
</ul>
<a name="info"></a>
<h3>Information</h3>
<table border="1" cellpadding="5" width="95%">
<tbody>
<tr>
<td><strong>Information</strong></td>
<td><strong>Comments</strong></td>
</tr>
<tr>
<td><a href="http://en.wikipedia.org/wiki/Homology_(biology)" target="_blank">Homology</a></td>
<td>Wiki</td>
</tr>
<tr>
<td><a href="http://www.nature.com/nrg/journal/v5/n10/glossary/nrg1449.html#df11" target="_blank">Useful definitions</a></td>
<td>Nature Reviews Genetics</td>
</tr>
<tr>
<td><a href="http://en.wikipedia.org/wiki/List_of_sequence_alignment_softwaretarget=%22_blank%22" target="_blank">Alignment software</a></td>
<td>Wiki</td>
</tr>
<tr>
<td><a href="http://www.plants.usda.gov/" target="_blank">Plant database</a></td>
<td>USDA</td>
</tr>
<tr>
<td><a href="http://www.ncbi.nlm.nih.gov/sites/entrez?cmd=Search&amp;db=genomeprj&amp;term=%22Viridiplantae%22[Organism]" target="_blank">Plant genomes</a></td>
<td>GenBanks</td>
</tr>
<tr>
<td><a href="http://www.mobot.org/MOBOT/research/APweb/welcome.html" target="_blank">Angliosperm Phylogeny</a></td>
<td>Missouri Botanical Garden</td>
</tr>
<tr>
<td><a href="http://tolweb.org/" target="_blank">Tree of Life</a></td>
<td>Collaborative effort to provide information about biodiversity</td>
</tr>
</tbody>
</table>
<a name="databases"></a>
<h3>On-line comparative sequenced genomes databases</h3>
<table border="1" cellpadding="5" width="95%">
<tbody>
<tr>
<td><strong>Database</strong></td>
<td><strong>Species;graphics;computation</strong></td>
<td><strong>Publication</strong></td>
</tr>
<tr>
<td><a href="http://www.phytozome.org/" target="_blank">Phytozome</a></td>
<td>Plants; Genome browsers;</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1347408/?tool=pubmed" target="_blank">Hartmann et al. 2006</a></td>
</tr>
<tr>
<td><a href="http://bioinformatics.psb.ugent.be/plaza/" target="_blank">PLAZA</a></td>
<td>Plants; Gene region alignments, Dotplot, Chromosome alignment, Multiplicon; blastp &amp; OrthoMCL method</td>
<td><a href="http://www.plantcell.org/cgi/content/full/21/12/3718" target="_blank">Proost et al. 2009</a><a></a></td>
</tr>
<tr>
<td><a href="http://chibba.agtec.uga.edu/duplication" target="_blank">Plant Genome Duplication</a></td>
<td>Plants; Dotplots and detail; blastp &amp; MCSCAN</td>
<td><a href="http://genome.cshlp.org/content/18/12/1944.long" target="_blank">Tang et al. 2008</a><a></a></td>
</tr>
<tr>
<td><a href="http://www.gramene.org/" target="_blank">Gramene</a></td>
<td>Grasses; Ensembl;</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2238951/?tool=pubmed" target="_blank">Liang et al. 2008</a></td>
</tr>
<tr>
<td><a href="http://www.comparative-legumes.org/" target="_blank">Legume Information System</a></td>
<td>Legumes; CMap;</td>
<td>--</td>
</tr>
<tr>
<td><a href="http://www.jgi.doe.gov/genome-projects" target="_blank">JGI genome projects</a></td>
<td>Various; Vista and links to Phytozome;</td>
<td>--</td>
</tr>
<tr>
<td><a href="http://genome.lbl.gov/vista/index.shtml" target="_blank">Vista</a></td>
<td>Various; Java display of regions; Shuffle-LAGAN &amp; BLAT in pipeline, allows user input.</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC441596/?tool=pubmed" target="_blank">Frazer et al. 2004</a></td>
</tr>
<tr>
<td><a href="http://narcisse.toulouse.inra.fr/" target="_blank">Narcisse</a></td>
<td>Various; Dotplots, circos, chromosome-based gene alignment; blastp &amp; custom script.</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2238891/?tool=pubmed" target="_blank">Courcelle et al. 2008</a></td>
</tr>
<tr>
<td><a href="http://cinteny.cchmc.org/" target="_blank">Cinteny</a></td>
<td>Various; Genome/chromosome view; custom algorithm</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1821339/?tool=pubmed" target="_blank">Sinha &amp; Meller 2007</a></td>
</tr>
<tr>
<td><a href="http://synteny.cnr.berkeley.edu/CoGe/" target="_blank">CoGe</a></td>
<td>Various; Dotplot (SynMap) and ajax Genome Browser; DAGchainer, Blast</td>
<td><a href="http://www3.interscience.wiley.com/journal/119410740/abstract?CRETRY=1&amp;SRETRY=0" target="_blank">Lyons &amp; Freeling 2008</a></td>
</tr>
<tr>
<td><a href="http://genome.sfu.ca/orthoclusterdb/" target="_blank">OrthoClusterDB</a></td>
<td>No plants; Genome chromosome view, link to Gbrowse; OrthoCluster, allows user input.</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2711082/?tool=pubmed" target="_blank">Ng et al. 2009</a></td>
</tr>
<tr>
<td><a href="http://teleost.cs.uoregon.edu/synteny_db" target="_blank">The Synteny Database</a></td>
<td>No plants; Various views; Blastp &amp; custom pipeline, emphasis on duplications and ohnologs.</td>
<td><a href="http://genome.cshlp.org/content/19/8/1497.long" target="_blank">Catchen et al. 2009</a></td>
</tr>
<tr>
<td><a href="http://www.ensembl.org/" target="_blank">Ensembl</a></td>
<td>No plants; ComparaMart, DAS visualization; TreeBeST, Pecan.</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2686571/?tool=pubmed" target="_blank">Hubbard et al. 2009</a></td>
</tr>
<tr>
<td><a href="http://genome.ucsc.edu/" target="_blank">UCSC Genome Browser</a></td>
<td>No plants;</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2099589/?tool=pubmed" target="_blank">Miller et. al. 2007</a></td>
</tr>
<tr>
<td><a href="http://inparanoid.sbc.su.se/">Inparanoid</a></td>
<td>Various, produce lists of orthologs</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2808972/?tool=pubmed">Ostlund et al. 2009</a></td>
</tr>
<tr>
<td><a href="http://www.cbil.upenn.edu/gene-family/">OrthoMCL</a></td>
<td>Various, produce lists of orthologs</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/12952885?dopt=Citation">Li et al. 2003</a></td>
</tr>
</tbody>
</table>
<h3>Synteny Computation and/or Display Software</h3>
<strong>Key</strong>: Perfect - conserved order in block, Imperfect - allows micro-rearrangments in block, Duplications - allows duplicated regions, Multiplicon - mutually homologous segment between &gt;2 genomes.
<table border="1" cellpadding="5" width="95%">
<tbody>
<tr>
<td><strong>Software</strong></td>
<td><strong>Input</strong></td>
<td><strong>Synteny</strong></td>
<td><strong>Graphics</strong></td>
<td><strong>Publication</strong></td>
</tr>
<tr>
<td><a href="http://bioops.info/2011/03/i-adhore-detect-synteny-regions/" target="_blank">i-ADHoRe</a></td>
<td>N gene lists</td>
<td>Perfect, duplications, multiplicon</td>
<td>Generates plots using Perl GD</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/12421767?dopt=Citation" target="_blank">Vandepoele et al. 2002</a> <a href="http://bioinformatics.oxfordjournals.org/content/24/1/127.long" target="_blank">Simmillion et al. 2007</a></td>
</tr>
<tr>
<td><a href="http://users.soe.ucsc.edu/~kent/">AXTCHAIN</a></td>
<td>BLASTZ alignments</td>
<td></td>
<td></td>
<td><a href="http://www.pnas.org/content/100/20/11484.full">Kent et al. 2003</a></td>
</tr>
<tr>
<td><a href="http://www.tc.umn.edu/~cann0010/Software.html">DiagHunter</a></td>
<td>Similarity file</td>
<td>Imperfect</td>
<td><a href="http://www.atgc.org/GenoPix_2D_Plotter/GenoPix_2D_Plotter_Download.html" target="_blank">GenoPix</a> Tcl/Tk</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/14519203?dopt=Citation" target="_blank">Cannon et al. 2003</a></td>
</tr>
<tr>
<td><a href="http://www-cse.ucsd.edu/groups/bioinformatics/GRIMM" target="_blank">GRIMM-synteny</a></td>
<td>Similarity file</td>
<td>Imperfect, breakpoints</td>
<td>Web</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/11934753?dopt=Citation" target="_blank">Tessler 2002</a> and<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC164646/?tool=pubmed" target="_blank">Pevzner and Tessler 2003</a></td>
</tr>
<tr>
<td><a href="http://www.bio.unc.edu/faculty/vision/lab/FISH/" target="_blank">FISH</a></td>
<td>Ordered marker file per contig and pairs files</td>
<td>Imperfect</td>
<td>no</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/12855440?dopt=Citation" target="_blank">Calabrese et al. 2003</a></td>
</tr>
<tr>
<td><a href="http://dagchainer.sourceforge.net/" target="_blank">DAGchainer</a></td>
<td>Gene pairs with chromosome positions</td>
<td>Perfect</td>
<td>Dot plot</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/15247098?dopt=Citation" target="_blank">Haas et al. 2004</a></td>
</tr>
<tr>
<td><a href="http://colinear.cbi.pku.edu.cn/" target="_blank">ColinearScan</a></td>
<td>Protein sequences with positions and pairs file</td>
<td>Perfect, estimates gap parameters.</td>
<td>no</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1626491/?tool=pubmed" target="_blank">Wang et al. 2006</a><a></a></td>
</tr>
<tr>
<td><a href="http://osfinder.dna.bio.keio.ac.jp/" target="_blank">OSfinder</a></td>
<td>Homologous markers with locations (provide scripts to generate this file from different sources)</td>
<td>Perfect, estimates all parameters, no duplication</td>
<td>PNG file of blocks and of dotplot</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/19188192?dopt=Citation" target="_blank">Hachiya et al. 2009</a></td>
</tr>
<tr>
<td><a href="http://krono.act.uji.es/noticias/orthocluster-a-new-tool-for-mining-syntenic-blocks" target="_blank">OrthoCluster</a></td>
<td>Imperfect</td>
<td></td>
<td></td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/19728289?dopt=Citation" target="_blank">Vergara &amp; CHen 2009</a></td>
</tr>
<tr>
<td><a href="http://www.bioinformatics.org/cyntenator/wiki/">Cyntenator</a></td>
<td>guided tree &amp; similarity files</td>
<td>Conserved synteny between n genomes</td>
<td>no</td>
<td><a href="http://www.plosone.org/article/info:doi%2F10.1371%2Fjournal.pone.0008861" target="_blank">Rodelsperger &amp; Dieterich 2010</a></td>
</tr>
<tr>
<td><a href="http://asap.ahabs.wisc.edu/mauve" target="_blank">Mauve</a></td>
<td>Multiple sequences (small genomes or regions)</td>
<td>Conserved, aligns multiple sequences, computes synteny &amp; breakpoints</td>
<td>Java graphics &amp; manager</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/15231754?dopt=Citation" target="_blank">Darling et al. 2004</a></td>
</tr>
<tr>
<td><a href="http://ews.uiuc.edu/~xuling/mcmusec/" target="_blank">MCMuSeC</a></td>
<td>Homologs for multiple genomes &amp; phylogenetic tree</td>
<td>Detects gene clusters</td>
<td>no</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/19158161?dopt=Citation" target="_blank">Ling et al. 2009</a></td>
</tr>
<tr>
<td><a href="http://chibba.agtec.uga.edu/duplication/mcscan/" target="_blank">MCscan</a><a></a></td>
<td>blast &amp; gff file</td>
<td>DAGchainer + consensus sequence builder</td>
<td>no</td>
<td><a href="http://genome.cshlp.org/content/18/12/1944.long" target="_blank">Tang et al. 2008</a><a></a></td>
</tr>
<tr>
<td><a href="http://titus.bio.uci.edu/lineup/" target="_blank">LineUp</a></td>
<td>Genetic maps</td>
<td></td>
<td>no</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/12695327?dopt=Citation" target="_blank">Hampson et al. 2003</a></td>
</tr>
<tr>
<td><a href="http://www.ige.tohoku.ac.jp/joho/gmProject/gmdownload.html" target="_blank">GenomeMatcher</a></td>
<td>BLAST, MUMmer</td>
<td>no</td>
<td>Dotplots and closeup (see<a href="http://www.ige.tohoku.ac.jp/joho/gmProject/gallery.html" target="_blank">gallery</a>)</td>
<td><a href="http://www.biomedcentral.com/1471-2105/9/376" target="_blank">Ohtsubo et al. 2008</a></td>
</tr>
<tr>
<td><a href="http://gmod.org/wiki/CMap" target="_blank">CMap</a><a></a></td>
<td>Maps (genetic, FPC, sequence)</td>
<td>no</td>
<td>Perl/CGI (see<a href="http://gmod.org/wiki/CMap#Screenshots" target="_blank">screen shots</a>)</td>
<td><a href="http://www.symapdb.org/docs/%3Ca" target="_blank">Youens-Clark et al. 2009</a></td>
</tr>
<tr>
<td><a href="http://gmod.org/wiki/GBrowse_syn" target="_blank">GBrowse_syn</a><a></a></td>
<td>Alignments</td>
<td>no</td>
<td>Perl/CGI GBrowse-based</td>
<td>-</td>
</tr>
<tr>
<td><a href="http://mkweb.bcgsc.ca/circos" target="_blank">Circos</a></td>
<td>GFF files</td>
<td>no</td>
<td>Perl/CGI</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pubmed/19541911?dopt=Citation" target="_blank">Krzywinski et al. 2009</a></td>
</tr>
</tbody>
</table>
<h4>A few alignment programs</h4>
<table border="1" cellpadding="5" width="95%">
<tbody>
<tr>
<td><strong>Software</strong></td>
<td><strong>Input</strong></td>
<td><strong>Computation</strong></td>
<td><strong>Graphics</strong></td>
<td><strong>Publications</strong></td>
</tr>
<tr>
<td><a href="http://mummer.sourceforge.net/">Mummer</a></td>
<td>Genome sequence</td>
<td>Anchors</td>
<td>Java graphical interface</td>
<td><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Citation&amp;list_uids=14759262">Kurtz et al. 2004</a></td>
</tr>
<tr>
<td><a href="http://lagan.stanford.edu/lagan_web/index.shtml" target="_blank">LAGAN</a></td>
<td>Genome Sequence</td>
<td></td>
<td>Vista</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC430967/?tool=pubmed" target="_blank">Brudno et al. 2007</a></td>
</tr>
<tr>
<td><a href="http://www.lbl.gov/Tech-Transfer/techs/lbnl1690.html" target="_blank">AVID</a></td>
<td>Genome sequence</td>
<td>Aligns and orders</td>
<td>Vista</td>
<td><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Citation&amp;list_uids=12529311">Bray et al. 2003</a></td>
</tr>
<tr>
<td><a href="http://genome.lbl.gov/vista/index.shtml" target="_blank">Vista</a></td>
<td>Alignments</td>
<td>LAGAN or AVID</td>
<td>Java (see <a href="http://pipeline.lbl.gov/vgb2help.shtml" target="_blank">screenshot</a>)</td>
<td><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC441596/?tool=pubmed" target="_blank">Frazer et al. 2004</a></td>
</tr>
</tbody>
</table>
<a name="details"></a></p>
<<<<<<< HEAD:source/_posts/2011-02-17-synteny-relevant-links.html
=======
</div>


  <footer>
    <p class="meta">
      
  


  


  <span class="byline author vcard">Authored by <span class="fn">
  
    <a href="https://plus.google.com/113645349664564078087" rel="author">Bioops</a>
  
  </span></span>


      




<time class='entry-date' datetime='2011-02-17T00:00:00+00:00'><span class='date'><span class='date-month'>Feb</span> <span class='date-day'>17</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
      
      

<span class="categories">
  
    <a class='category' href='/category/bioinformatics/'>bioinformatics</a>, <a class='category' href='/category/genomics/'>genomics</a>, <a class='category' href='/category/software/'>software</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://bioops.info/2011/02/synteny-relevant-links/" data-via="" data-counturl="http://bioops.info/2011/02/synteny-relevant-links/" >Tweet</a>
  
  
  <div class="g-plusone" data-size="medium"></div>
  
  
    <div class="fb-like" data-layout="button_count" data-send="false" data-width="300" data-show-faces="false"></div>
  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left" href="/2011/02/software-packages-for-next-generation-sequence-analysis-from-seqanswers-com/" title="Previous Post: Software packages for next generation sequence analysis (from seqanswers.com)">&laquo; Software packages for next generation sequence analysis (from seqanswers.com)</a>
      
      
        <a class="basic-alignment right" href="/2011/02/windows-perl-blast/" title="Next Post: 本地blast">本地blast &raquo;</a>
      
    </p>
  </footer>
</article>

  <section>
    <h1>Comments</h1>
    <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
  </section>


</div>

    </div>
  </div>
  <footer role="contentinfo"><!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>

<p>
  Copyright &copy; 2016 - Bioops -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a> | Themed with <a href="https://github.com/lucaslew/whitespace">Whitespace</a></span>
</p>


</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'bioopsgithub';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://bioops.info/2011/02/synteny-relevant-links/';
        var disqus_url = 'http://bioops.info/2011/02/synteny-relevant-links/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>






<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
>>>>>>> d80cd8fa3e1fb5461144707ba04f7385ec6726a7:2011/02/synteny-relevant-links/index.html
