---
layout: post
title: Software packages for discovering structural variation with next-generation
  sequencing
categories:
- Bioinformatics
- NGS
- Software
tags:
- Bioinformatics
- NGS
- Software
- Structural Variation
published: true
comments: true
---
<p>More softwares and information will be added. (latest updated on 28/02/2011)</p>

<p><strong>Highly recommended you to read this paper. Mapping copy number variation by population-scale genome sequencing (<a href="http://www.nature.com/nature/journal/v470/n7332/full/nature09708.html" target="_blank">Mills et al. 2011</a>)</strong></p>

<p><a href="http://sv.gersteinlab.org/pemer/" target="_blank">PEMer</a> (<a href="http://w02.genomebiology.com/2009/10/2/R23/" target="_blank">Korbel et al. 2009</a>) It comprises an analysis pipeline, compatible with several next-generation sequencing platforms; simulation-based error models, yielding confidence-values for each structural variant; and a back-end database.</p>

<p><a href="http://www.broadinstitute.org/cgi-bin/cancer/publications/pub_paper.cgi?mode=view&amp;paper_id=182" target="_blank">SegSeq</a> (<a href="http://www.nature.com/nmeth/journal/v6/n1/abs/nmeth.1276.html" target="_blank">Chiang et al. 2009</a>)  Detect and localize copy-number alterations from massively parallel sequence data. A simple approach would be to partition the genome into windows of fixed size, estimate the tumor-normal ratios for each window and use standard segmentation algorithms to decompose the genome into regions of equivalent copy number.</p>

<p><a href="http://compbio.cs.sfu.ca/strvar.htm" target="_blank">VariationHunter</a> (<a href="http://genome.cshlp.org/content/19/7/1270.full" target="_blank">Hormozdiari et al. 2009</a>) Combinatorial algorithms for structural variation detection in high-throughput sequenced genomes. It's based on maximum parsimony</p>

<p><a href="http://compbio.cs.toronto.edu/modil/">MoDIL</a> (<a href="http://www.nature.com/nmeth/journal/v6/n7/full/nmeth.f.256.html">Lee et al.2009</a>) MoDIL, or Mixture of Distributions Indel Locator, a novel method for finding medium sized indels from high throughput sequencing datasets. The MoDIL algorithm compares the distribution of insert sizes in the sequenced library to the distribution of the observed mapped distances at a particular genomic location.</p>

<p><a href="http://www.ebi.ac.uk/~kye/pindel/" target="_blank">Pindel</a> (<a href="http://bioinformatics.oxfordjournals.org/content/25/21/2865.full" target="_blank">Ye et al. 2009</a>) A pattern growth approach, to detect breakpoints of large deletions and medium-sized insertions from paired-end short reads.</p>

<p><a href="http://breakdancer.sourceforge.net/" target="_blank">BreakDancer</a> (<a href="http://www.nature.com/nmeth/journal/v6/n9/abs/nmeth.1363.html" target="_blank">Chen et al. 2009</a>) BreakDancerMax predicts five types of structural variants: insertions, deletions, inversions, inter- and intra-chromosomal translocations from next-generation short paired-end sequencing reads using read pairs that are mapped with unexpected separation distances or orientation.</p>

<p><a href="http://genome.wustl.edu/software/cnvhmm" target="_blank">cnvHMM</a> copy number analysis using Hidden Markov algorithm</p>

<p>Geometric Analysis of Structural Variants (<a href="http://cs.brown.edu/people/braphael/software.html" target="_blank">GASV</a>) (<a href="http://bioinformatics.oxfordjournals.org/content/25/12/i222" target="_blank">Sindi et al 2009</a>) A geometric approach for identification, classification and comparison of structural variants. The software is for analysis of structural variation from paired-end sequencing and/or array-CGH data.</p>

<p>Sequence Variant Analyzer (<a href="http://people.chgv.lsrc.duke.edu/~dg48//sva/index.php" target="_blank">SVA</a>) SVA is a computer software project designed to annotate, visualize, and analyze the genetic variants identified through next-generation sequencing studies, including whole-genome sequencing (WGS) and exome sequencing studies.</p>

<p><a href="http://genome.wustl.edu/software/swt" target="_blank">SWT</a> It is a collection of R functions for statistical analysisof genome-wide data, by Qunyuan Zhang (qunyuan@wustl.edu), DSG.</p>

<p><a href="http://varscan.sourceforge.net/" target="_blank">VarScan</a> (<a href="http://bioinformatics.oxfordjournals.org/content/25/17/2283" target="_blank">Koboldt et al. 2009</a>) VarScan is a platform-independent, technology-independent software tool for identifying SNPs and indels in massively parallel sequencing of individual and pooled samples. (not actually structural variation)</p>

<p><a href="http://tiger.dbs.nus.edu.sg/CNV-seq/" target="_blank">CNV-seq</a> (<a href="http://www.biomedcentral.com/1471-2105/10/80" target="_blank">Xie and Tammi. 2009</a>) The method is based on a robust statistical model that describes the complete analysis procedure and allows the computation of essential confidence values for detection of CNV.</p>

<p><a href="http://sv.gersteinlab.org/breakseq/" target="_blank">BreakSeq</a> (<a href="http://www.nature.com/nbt/journal/v28/n1/abs/nbt.1600.html" target="_blank">Lam et al. 2010</a>) a pipeline for annotation, classification and analysis of SVs at single nucleotide resolution.</p>

<p><a href="http://bioinformatics.oxfordjournals.org/content/26/21/2776.full" target="_blank">CopyMap</a> (<a href="http://bioinformatics.oxfordjournals.org/content/26/21/2776.full" target="_blank">Zöllner 2010</a>) The program package CopyMap identifies copy number variation from oligo-hybridization and CGH data. Using a time-dependent hidden Markov model to combine evidence of copy number variants (CNVs) across multiple carriers, CopyMap is substantially more accurate than standard hidden Markov methods in identifying CNVs and calling CNV-carriers. Moreover, CopyMap provides more precise estimates of CNV-boundaries.</p>

<p><a href="http://www-genepi.med.utah.edu/suppl/SLOPE/index.html" target="_blank">SLOPE</a> (<a href="http://bioinformatics.oxfordjournals.org/content/early/2010/09/27/bioinformatics.btq528" target="_blank">Abel et al. 2010</a>) A quick and accurate method for locating non-SNP structural variation from targeted next-generation sequence data.<br />
By focusing on a small (a few kb to a few Mb) target reference sequence, SLOPE can perform fast and ﬂexible split-read alignments and determine ‘chimeric’ sequences with single-base resolution.<br />
SLOPE aims to detect sequence breakpoints from only one side of a split read, and therefore does not rely on the insert size for detection.</p>

<p><a href="http://code.google.com/p/hydra-sv/" target="_blank">HYDRA</a> (<a href="http://genome.cshlp.org/content/20/5/623.full" target="_blank">Quinlan et al. 2010</a>) Hydra detects structural variation (SV) breakpoints by clustering discordant paired-end alignments whose "signatures" corroborate the same putative breakpoint. Hydra can detect breakpoints caused by all classes of structural variation. Moreover, it was designed to detect variation in both unique and duplicated genomic regions; therefore, it will examine paired-end reads having multiple discordant alignments.</p>

<p><a href="http://www.sanger.ac.uk/resources/software/cnd/" target="_blank">CnD</a> (<a href="http://bioinformatics.oxfordjournals.org/content/26/4/565.full" target="_blank">Simpson et al. 2010</a>) A copy number variant caller for inbred strains.<br />
The target organism is assumed to be inbred, and therefore homozygous, so regions of apparent heterozygous SNPs (as called by MAQ) can be used to detect copy number gains. cnD uses both the rate of these paralogous sequence variants, and the raw sequence depth, to call copy number gains and losses using a hidden markov model.</p>

<p><a href="http://sv.gersteinlab.org/age/" target="_blank">AGE</a> (<a href="http://bioinformatics.oxfordjournals.org/content/27/5/595.full" target="_blank">Abyzov and Gerstein 2011</a>) defining breakpoints of genomic structural variants at single-nucleotide resolution, through optimal alignments with gap excision.</p>

<p><a href="http://sv.gersteinlab.org/cnvnator/" target="_blank">CNVnator</a> (<a href="http://genome.cshlp.org/content/early/2011/02/04/gr.114876.110.abstract" target="_blank">Abyzov et al. 2011</a>) An approach to discover, genotype and characterize typical and atypical CNVs from family and population genome sequencing.</p>

<p><strong>
</strong><strong> Other relevant softwares</strong></p>

<p><a href="http://samtools.sourceforge.net/" target="_blank">SAMtools</a> SAM Tools provide various utilities for manipulating alignments in the SAM format, including sorting, merging, indexing and generating alignments in a per-position format.</p>

<p><a href="http://maq.sourceforge.net/" target="_blank">Maq</a> Maq stands for Mapping and Assembly with Quality It builds assembly by mapping short reads to reference sequences.</p>
