---
layout: post
title: Comparison of de novo assembly tools for next-generation sequencing
categories:
- Bioinformatics
- NGS
- Paper
tags:
- Assembly
- Bioinformatics
- Genomics
- NGS
- Paper
- Software
published: true
comments: true
---
<p><strong><a href="http://bioinformatics.oxfordjournals.org/content/early/2011/06/02/bioinformatics.btr319.abstract" target="_blank">Comparative Studies of <em>de novo</em> Assembly Tools for Next-generation Sequencing Technologies</a></strong></p>

<p>Yong Lin, Jian Li, Hui Shen, Lei Zhang, Christopher J Papasian and Hong-Wen Deng</p>

<p><strong>Motivation:</strong> Several new de novo assembly tools have been developed recently to assemble short sequencing reads generated by next-generation sequencing platforms. However, the performance of these tools under various conditions has not been fully investigated, and sufficient information is not currently available for informed decisions to be made regarding the tool that would be most likely to produce the best performance under a specific set of conditions.</p>

<p><strong>Results:</strong> We studied and compared the performance of commonly used de novo assembly tools specifically designed for next-generation sequencing data, including SSAKE, VCAKE, Euler-sr, Edena, Velvet, ABySS and SOAPdenovo. Tools were compared using several performance criteria, including N50 length, sequence cover-age, and assembly accuracy. Various properties of read data, including single-end/paired-end, sequence GC content, depth of coverage and base calling error rates, were investigated for their effects on the performance of different assembly tools. We also compared the computation time and memory usage of these seven tools. Based on the results of our comparison, the relative perform-ance of individual tools are summarized and tentative guidelines for optimal selection of different assembly tools, under different condi-tions, are provided.</p>

<p>拿流行的几个<em>de novo</em> assembly tools for next-generation sequencing做了系统的比较，是一篇很不错的文章。曾经有多少人徘徊于各种assembly软件中，不知道选择哪一种，这篇文章提供了很好的帮助。</p>

<p>作者比较了<a href="http://www.bcgsc.ca/platform/bioinfo/software/ssake" target="_blank">SSAKE</a>, <a href="http://vcake.sourceforge.net/" target="_blank">VCAKE</a>, <a href="http://euler-assembler.ucsd.edu/portal/" target="_blank">Euler-sr</a>, <a href="http://www.genomic.ch/edena.php" target="_blank">Edena</a>, <a href="http://www.ebi.ac.uk/~zerbino/velvet/" target="_blank">Velvet</a>, <a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss" target="_blank">ABySS</a> and <a href="http://soap.genomics.org.cn/soapdenovo.html" target="_blank">SOAPdenovo</a>，（竟然没有<a href="https://www.broadinstitute.org/scientific-community/science/programs/genome-sequencing-and-analysis/computational-rd/computational-" target="_blank">ALLPATHS-LG</a>！）。</p>

<p>先用大到人类（其实最多只用了100Mb序列片段）小到<em>E. coli</em>的不同GC含量的基因组，模拟出single-end和paired-end、以及不同测序深度、不同测序片段长度的短序列片段，然后使用各个软件拼接，缺省设置，在各有2.40GB双核+12GB的8个nodes上运行，然后比较拼接结果：<a href="http://bioops.info/2011/03/n50/" target="_blank">N50</a>、基因组覆盖率（sequence coverage）、拼接错误（assembly error）和base  calling error rate (BCER)等，还有软件运行时的硬件消耗。</p>

<p><strong>最终的结论：（<span style="color: #ff0000;">结论比较复杂，不同条件和要求下，各个软件表现有些差异。总的来看ABySS和SOAPdenovo比较“万金油”。</span>）</strong></p>

<p><a href="http://bioops.info/images/uploads/2011/07/20110721212646.gif"><img class="alignnone size-full wp-image-480" title="20110721212646" src="http://bioops.info/images/uploads/2011/07/20110721212646.gif" alt="" width="1040" height="408" /></a></p>

<p><strong>再看非常重要的硬件消耗（<span style="color: #ff0000;">还是SOAPdenovo和abyss给力啊，看来我眼光不错，我就是用的这两个，还有ALLPATHS-LG！</span>）</strong></p>

<p><a href="http://bioops.info/images/uploads/2011/07/20110721212958.gif"><img class="alignnone size-full wp-image-481" title="20110721212958" src="http://bioops.info/images/uploads/2011/07/20110721212958.gif" alt="" width="436" height="634" /></a></p>

<p><span style="color: #ff0000;"><strong>本人看法：SOAPdenovo能拼接<a href="http://bioops.info/2011/03/scaffold-or-supercontig/" target="_blank">scaffolds</a>，能产生更长的N50，<strong>而ABySS只能拼接到<a href="http://bioops.info/2011/03/contig/" target="_blank">contigs</a>；</strong>但ABySS基于<a href="https://computing.llnl.gov/tutorials/parallel_comp/" target="_blank">MPI并行</a>，对硬件需求低。<strong>ALLPATHS-LG和<strong>SOAPdenovo相似。</strong></strong></strong></span></p>

<p><strong>
</strong></p>
