---
layout: post
title: ! '[paper] Efficient de novo assembly of large genomes using compressed data
  structures'
categories:
- Bioinformatics
- NGS
- Paper
tags:
- Assembly
- Genomics
- NGS
- Paper
published: true
comments: true
---
<p><strong>Efficient de novo assembly of large genomes using compressed data structures</strong></p>

<p>Jared T. Simpson and Richard Durbin</p>

<p>Wellcome Trust Sanger Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SA, United Kingdom</p>

<p><strong>Abstract</strong></p>

<p>De novo genome sequence assembly is important both to generate new sequence assemblies for previously uncharacterized genomes and to identify the genome sequence of individuals in a reference-unbiased way. We present memory efficient data structures and algorithms for assembly using the FM-index derived from the compressed Burrows-Wheeler transform, and a new assembler based on these called <a href="https://github.com/jts/sga" target="_blank">SGA </a>(String Graph Assembler). We describe algorithms to error-correct, assemble, and scaffold large sets of sequence data. SGA uses the overlap-based string graph model of assembly, unlike most de novo assemblers that rely on de Bruijn graphs, and is simply parallelizable. We demonstrate the error correction and assembly performance of SGA on 1.2 billion sequence reads from a human genome, which we are able to assemble using 54 GB of memory. The resulting contigs are highly accurate and contiguous, while covering 95% of the reference genome (excluding contigs &lt;200 bp in length). Because of the low memory requirements and parallelization without requiring inter-process communication, SGA provides the first practical assembler to our knowledge for a mammalian-sized genome on a low-end computing cluster.</p>

<p><a href="http://genome.cshlp.org/content/22/3/549.full?rss=1" target="_blank">fulltext</a></p>
