---
layout: post
title: Next-generation sequencing quality control tools
categories:
- Bioinformatics
- NGS
- Software
tags:
- NGS
- QC
- Software
published: true
comments: true
---
<p><a href="http://hannonlab.cshl.edu/fastx_toolkit/" target="_blank"><strong>FASTX-Toolkit</strong></a>: The FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/ FASTQ files preprocessing, including:
<ul>
	<li>FASTQ-to-FASTA converter
<div>Convert FASTQ files to FASTA files.</div></li>
	<li>FASTQ Information
<div>Chart Quality Statistics and Nucleotide Distribution</div></li>
	<li>FASTQ/A Collapser
<div>Collapsing identical sequences in a FASTQ/A file into a single sequence (while maintaining reads counts)</div></li>
	<li>FASTQ/A Trimmer
<div>Shortening reads in a FASTQ or FASTQ files (removing barcodes or noise).</div></li>
	<li>FASTQ/A Renamer
<div>Renames the sequence identifiers in FASTQ/A file.</div></li>
	<li>FASTQ/A Clipper
<div>Removing sequencing adapters / linkers</div></li>
	<li>FASTQ/A Reverse-Complement
<div>Producing the Reverse-complement of each sequence in a FASTQ/FASTA file.</div></li>
	<li>FASTQ/A Barcode splitter
<div>Splitting a FASTQ/FASTA files containning multiple samples</div></li>
	<li>FASTA Formatter
<div>changes the width of sequences line in a FASTA file</div></li>
	<li>FASTA Nucleotide Changer
<div>Convets FASTA sequences from/to RNA/DNA</div></li>
	<li>FASTQ Quality Filter
<div>Filters sequences based on quality</div></li>
	<li>FASTQ Quality Trimmer
<div>Trims (cuts) sequences based on quality</div></li>
	<li>FASTQ Masker
<div>Masks nucleotides with 'N' (or other character) based on quality</div></li>
</ul>
<strong><a href="https://main.g2.bx.psu.edu/" target="_blank">Galaxy NGS QC and manipulation tools</a> (<a href="http://bioinformatics.oxfordjournals.org/content/26/14/1783.short" target="_blank">citation</a>)</strong>: Galaxy provides a tool suite that functions on all of the commonly known FASTQ format variants and provides a pipeline for manipulating next generation sequencing data taken from a sequencing machine all the way through the quality filtering steps.</p>

<p><strong><a href="http://brainarray.mbni.med.umich.edu/brainarray/ngsqc/" target="_blank">NGSQC</a> (<a href="http://www.biomedcentral.com/1471-2164/11/S4/S7/" target="_blank">citation</a>)</strong>:The NGSQC pipeline provides a set of novel quality control measures for quickly detecting a wide variety of quality issues in deep sequencing data derived from two dimensional surfaces, regardless of the assay technology used. It also enables researchers to determine whether sequencing data related to their most interesting biological discoveries are caused by sequencing quality issues. NGSQC can help to ensure that biological conclusions, in particular those based on relatively rare sequences, are not caused by low quality sequencing.</p>

<p><strong><a href="http://www.nipgr.res.in/ngsqctoolkit.html" target="_blank">NGS QC Toolkit</a> (<a href="http://www.plosone.org/article/info:doi/10.1371/journal.pone.0030619#abstract0" target="_blank">citation</a>)</strong>: A toolkit for the quality control (QC) of next generation sequencing (NGS) data. The toolkit comprises of user-friendly stand alone tools for quality control of the sequence data generated using Illumina and Roche 454 platforms with detailed results in the form of tables and graphs, and filtering of high-quality sequence data. It also includes few other tools, which are helpful in NGS data quality control and analysis.</p>

<p><strong><a href="http://prinseq.sourceforge.net/" target="_blank">PRINSEQ</a> (<a href="http://bioinformatics.oxfordjournals.org/content/27/6/863" target="_blank">citation</a>)</strong>: PRINSEQ can be used to filter, reformat, or trim your genomic and metagenomic sequence data. It generates summary statistics of your sequences in graphical and tabular format. It is easily configurable and provides a user-friendly interface.</p>

<p><strong><a href="http://solexaqa.sourceforge.net/" target="_blank">SolexaQA</a> (<a href="http://www.biomedcentral.com/1471-2105/11/485" target="_blank">citation</a>)</strong>: SolexaQA is a Perl-based software package that calculates quality statistics and creates visual representations of data quality from FASTQ files generated by Illumina second-generation sequencing technology (“Solexa”).</p>
