---
layout: post
title: ! '[paper] De novo assembly and genotyping of variants using colored de Bruijn
  graphs'
categories:
- Bioinformatics
- NGS
- Paper
- Software
tags:
- Assembly
- Cortex
- Paper
- Software
- Variation Discovery
published: true
comments: true
---
<p><strong><a href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.1028.html" target="_blank"><em>De novo</em> assembly and genotyping of variants using colored de Bruijn graphs</a></strong></p>

<p>Detecting genetic variants that are highly divergent from a reference sequence remains a major challenge in genome sequencing. We introduce <em>de novo</em>assembly algorithms using colored de Bruijn graphs for detecting and genotyping simple and complex genetic variants in an individual or population. We provide an efficient software implementation, <a href="http://cortexassembler.sourceforge.net/" target="_blank">Cortex</a>, the first <em>de novo </em>assembler capable of assembling multiple eukaryotic genomes simultaneously. Four applications of Cortex are presented. First, we detect and validate both simple and complex structural variations in a high-coverage human genome. Second, we identify more than 3 Mb of sequence absent from the human reference genome, in pooled low-coverage population sequence data from the 1000 Genomes Project. Third, we show how population information from ten chimpanzees enables accurate variant calls without a reference sequence. Last, we estimate classical human leukocyte antigen (HLA) genotypes at <em>HLA-B</em>, the most variable gene in the human genome.</p>
