---
layout: post
title: ! '[paper] GAGE: A critical evaluation of genome assemblies and assembly algorithms'
categories:
- Bioinformatics
- NGS
- Paper
- Software
tags:
- ABySS
- Assembly
- Genomics
- NGS
- Paper
published: true
comments: true
---
<p><strong>GAGE: A critical evaluation of genome assemblies and assembly algorithms</strong></p>

<p>Abstract</p>

<p>New sequencing technology has dramatically altered the landscape of whole-genome sequencing, allowing scientists to initiate numerous projects to decode the genomes of previously unsequenced organisms. The lowest-cost technology can generate deep coverage of most species, including mammals, in just a few days. The sequence data generated by one of these projects consist of millions or billions of short DNA sequences (reads) that range from 50 to 150 nt in length. These sequences must then be assembled de novo before most genome analyses can begin. Unfortunately, genome assembly remains a very difficult problem, made more difficult by shorter reads and unreliable long-range linking information. In this study, we evaluated several of the leading de novo assembly algorithms on four different short-read data sets, all generated by Illumina sequencers. Our results describe the relative performance of the different assemblers as well as other significant differences in assembly difficulty that appear to be inherent in the genomes themselves. Three overarching conclusions are apparent: first, that data quality, rather than the assembler itself, has a dramatic effect on the quality of an assembled genome; second, that the degree of contiguity of an assembly varies enormously among different assemblers and different genomes; and third, that the correctness of an assembly also varies widely and is not well correlated with statistics on contiguity. To enable others to replicate our results, all of our data and methods are freely available, as are all assemblers used in this study.</p>

<p><a href="http://genome.cshlp.org/content/22/3/557.full?rss=1" target="_blank">fulltext</a></p>
