
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Parse Gff3 File - Bioops</title>
  <meta name="author" content="Bioops">

  
  <meta name="description" content="Here is an example script of parsing gff3 file to get the cds sequences. The annotation of a reference genome usually is stored as a gff3 format file &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bioops.info/2011/03/parse-gff3-file">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="Bioops" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="/javascripts/libs/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet" type="text/css">
<link href="//fonts.googleapis.com/css?family=Fjalla+One" rel="stylesheet" type="text/css">
<!--- MathJax Configuration -->
<script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-21431888-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'stats.g.doubleclick.net/dc.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   class="collapse-sidebar sidebar-footer" >
  <header role="banner"><hgroup>
  <h1><a href="/">Bioops</a></h1>
  
    <h2>Bioinformatics=(ACGAAG->AK)+(#!/bin/sh)+(P(A|B)=P(B|A)*P(A)/P(B))</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscribe" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 13.310204,73.332654 C 5.967347,73.332654 0,79.322448 0,86.621428 c 0,7.338776 5.967347,13.262246 13.310204,13.262246 7.370408,0 13.328572,-5.92245 13.328572,-13.262246 0,-7.29898 -5.958164,-13.288774 -13.328572,-13.288774 z M 0.01530612,33.978572 V 53.143878 C 12.493878,53.143878 24.229592,58.02347 33.068368,66.865306 41.894898,75.685714 46.767346,87.47449 46.767346,100 h 19.25 C 66.017346,63.592858 36.4,33.979592 0.01530612,33.978572 l 0,0 z M 0.03877552,0 V 19.17449 C 44.54796,19.17551 80.77551,55.437756 80.77551,100 H 100 C 100,44.87653 55.15102,0 0.03877552,0 z"></path></svg></a></li>
  
</ul>
  
  
  
  
  
<ul class="subscribe">
  <li><a href="https://github.com/bioops" rel="subscribe-github" title="@bioops on GitHub"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="25" height="25" viewbox="0 0 100 100"><path class="social" d="M 50,0 C 22.385714,0 0,22.385714 0,50 0,77.614286 22.385714,100 50,100 77.614286,100 100,77.614286 100,50 100,22.385714 77.614286,0 50,0 z m 29.692858,79.692858 c -3.859184,3.859182 -8.351022,6.887754 -13.35,9.00306 -1.27041,0.536736 -2.560204,1.009184 -3.867348,1.415306 v -7.493878 c 0,-3.938774 -1.35102,-6.835714 -4.053062,-8.690816 1.692858,-0.163264 3.24694,-0.390816 4.663266,-0.683672 1.416326,-0.292858 2.913266,-0.716328 4.491838,-1.27041 1.57857,-0.55408 2.994896,-1.213264 4.247958,-1.97755 1.253062,-0.765306 2.458164,-1.758164 3.613266,-2.978572 1.155102,-1.220408 2.12449,-2.604082 2.905102,-4.15 0.780612,-1.545918 1.4,-3.40204 1.855102,-5.566326 0.455102,-2.164286 0.683674,-4.54898 0.683674,-7.153062 0,-5.045918 -1.643878,-9.341836 -4.931634,-12.890816 C 77.44796,33.35 77.285714,29.10204 75.463266,24.512244 l -1.22143,-0.145918 c -0.845918,-0.09796 -2.368366,0.260204 -4.565306,1.07449 -2.196938,0.814286 -4.663264,2.14796 -7.396938,4.004082 -3.87449,-1.07449 -7.893878,-1.611224 -12.061224,-1.611224 -4.19898,0 -8.203062,0.536734 -12.012246,1.611224 -1.72449,-1.17245 -3.361224,-2.139796 -4.907142,-2.905102 C 31.753062,25.77449 30.516326,25.254082 29.587756,24.97653 28.660204,24.7 27.79796,24.528572 27,24.463266 c -0.79796,-0.0653 -1.310204,-0.08062 -1.537756,-0.04898 -0.22755,0.03164 -0.390816,0.0653 -0.487754,0.09796 -1.82347,4.62245 -1.985714,8.87143 -0.487756,12.743878 -3.287754,3.54796 -4.931632,7.844898 -4.931632,12.890816 0,2.604082 0.227552,4.988776 0.683674,7.153062 0.456122,2.164286 1.07449,4.020408 1.855102,5.566326 0.780612,1.545918 1.75,2.929592 2.905102,4.15 1.155102,1.220408 2.360204,2.213266 3.613264,2.978572 1.253062,0.766326 2.669388,1.42449 4.24796,1.97755 1.578572,0.554082 3.07551,0.976532 4.491836,1.27041 1.416328,0.292856 2.970408,0.521428 4.663266,0.683672 -2.669388,1.82347 -4.004082,4.720408 -4.004082,8.690816 v 7.639796 C 36.536734,89.818368 35.083674,89.3 33.656122,88.695918 c -4.99898,-2.115306 -9.490816,-5.143878 -13.35,-9.00306 -3.859184,-3.859184 -6.887754,-8.351022 -9.00306,-13.35 C 9.1163263,61.171428 8.0071428,55.67347 8.0071428,50 c 0,-5.67347 1.1091835,-11.171428 3.2969392,-16.342858 2.115306,-4.998978 5.143878,-9.490816 9.00306,-13.35 3.859184,-3.859182 8.351022,-6.887754 13.35,-9.00306 C 38.828572,9.1163266 44.32653,8.0071428 50,8.0071428 c 5.67347,0 11.171428,1.1091838 16.342858,3.2969392 5,2.115306 9.490816,5.143878 13.35,9.00306 3.859182,3.859184 6.887754,8.351022 9.00306,13.35 2.186736,5.17245 3.295918,10.67041 3.295918,16.342858 0,5.672448 -1.109182,11.171428 -3.296938,16.342858 -2.115306,4.998978 -5.143878,9.490816 -9.00204,13.35 l 0,0 z"></path></svg></a></li>
</ul>
  
  
  
  
  
  
  
    
      <form action="http://google.com/search" method="get">
        <fieldset role="search">
          <input type="hidden" name="sitesearch" value="bioops.info" />
    
          <input class="search" type="text" name="q" results="0" placeholder="Search"/>
        </fieldset>
      </form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      
        <h1 class="entry-title">Parse Gff3 File</h1>
      
    
    
      <p class="meta">
        




<time class='entry-date' datetime='2011-03-08T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>8</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
        
         | <a href="#disqus_thread">Comments</a>
        
        
      </p>
    
  </header>


<div class="entry-content"><p>Here is an example script of parsing gff3 file to get the cds sequences.</p>

<p>The annotation of a reference genome usually is stored as a <a href="http://gmod.org/wiki/GFF3" target="_blank">gff3 format</a> file. Most of the time, the genome sequencing group would provide CDS sequences along with the reference genome. However, sometimes they have transcripts instead of CDS, or you want to get 3&#8217;-UTR region, or 500 bp upstream sequences from the coding regions. It&#8217;s a good idea to parse the gff3 files to get information you need.</p>

<p>If you have any questions about the script or parsing gff3 file, leave a comment. Thank you!</p>

<figure class='code'><figcaption><span>parse_gff3_cds.pl</span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
<span class='line-number'>36</span>
<span class='line-number'>37</span>
<span class='line-number'>38</span>
<span class='line-number'>39</span>
<span class='line-number'>40</span>
<span class='line-number'>41</span>
<span class='line-number'>42</span>
<span class='line-number'>43</span>
<span class='line-number'>44</span>
<span class='line-number'>45</span>
<span class='line-number'>46</span>
<span class='line-number'>47</span>
<span class='line-number'>48</span>
<span class='line-number'>49</span>
<span class='line-number'>50</span>
<span class='line-number'>51</span>
<span class='line-number'>52</span>
<span class='line-number'>53</span>
<span class='line-number'>54</span>
<span class='line-number'>55</span>
<span class='line-number'>56</span>
<span class='line-number'>57</span>
<span class='line-number'>58</span>
<span class='line-number'>59</span>
<span class='line-number'>60</span>
<span class='line-number'>61</span>
<span class='line-number'>62</span>
<span class='line-number'>63</span>
<span class='line-number'>64</span>
<span class='line-number'>65</span>
<span class='line-number'>66</span>
</pre></td><td class='code'><pre><code class='perl'><span class='line'><span class="c1">#!/usr/bin/perl</span>
</span><span class='line'><span class="c1"># parse_gff3.pl</span>
</span><span class='line'><span class="c1"># get the cds sequences from gff3 file.</span>
</span><span class='line'><span class="c1"># usage: perl parse_gff3_cds.pl gff3_input_file genome_sequence_directory &gt;output.txt</span>
</span><span class='line'>
</span><span class='line'><span class="k">use</span> <span class="n">strict</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="n">warnings</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::Tools::</span><span class="n">GFF</span><span class="p">;</span>
</span><span class='line'><span class="k">use</span> <span class="nn">Bio::DB::</span><span class="n">Fasta</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$gff3_file</span><span class="o">=</span><span class="nv">$ARGV</span><span class="p">[</span><span class="mi">0</span><span class="p">];</span>  <span class="c1">#gff3 input file</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$genome</span><span class="o">=</span><span class="nv">$ARGV</span><span class="p">[</span><span class="mi">1</span><span class="p">];</span> <span class="c1">#genome sequence directory</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$gffio</span> <span class="o">=</span> <span class="nn">Bio::Tools::</span><span class="n">GFF</span> <span class="o">-&gt;</span> <span class="k">new</span><span class="p">(</span><span class="o">-</span><span class="n">file</span> <span class="o">=&gt;</span><span class="nv">$gff3_file</span> <span class="p">,</span> <span class="o">-</span><span class="n">gff_version</span> <span class="o">=&gt;</span> <span class="mi">3</span><span class="p">);</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$db</span>    <span class="o">=</span> <span class="nn">Bio::DB::</span><span class="n">Fasta</span>  <span class="o">-&gt;</span> <span class="k">new</span><span class="p">(</span><span class="nv">$genome</span><span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="k">my</span> <span class="nv">$i</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$first</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$strand</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'><span class="k">my</span> <span class="nv">$seq</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">;</span>
</span><span class='line'><span class="k">while</span><span class="p">(</span><span class="k">my</span> <span class="nv">$feature</span> <span class="o">=</span> <span class="nv">$gffio</span><span class="o">-&gt;</span><span class="n">next_feature</span><span class="p">())</span> <span class="p">{</span>
</span><span class='line'><span class="c1"># Sometimes, the gff3 file format is a little with the standard format,</span>
</span><span class='line'><span class="c1"># So that the keys of the hashes are different.</span>
</span><span class='line'><span class="c1"># Use the following lines of masked code to see the keys of the hashes. Some Change may be needed.</span>
</span><span class='line'><span class="c1">#  $i++;</span>
</span><span class='line'><span class="c1">#  if ($i==2){</span>
</span><span class='line'><span class="c1">#    my @a=keys %{$feature};</span>
</span><span class='line'><span class="c1">#    print &quot;@a\n&quot;;</span>
</span><span class='line'><span class="c1">#    my @b=keys %{$feature-&gt;{_location}};</span>
</span><span class='line'><span class="c1">#    print &quot;@b\n&quot;;</span>
</span><span class='line'><span class="c1">#    print &quot;$feature-&gt;{_primary_tag}\n&quot;;</span>
</span><span class='line'><span class="c1">#  }</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$feature</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_primary_tag</span><span class="p">}</span><span class="o">=~</span><span class="sr">/mrna/i</span><span class="p">){</span>
</span><span class='line'>    <span class="nv">$first</span><span class="o">++</span><span class="p">;</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$first</span><span class="o">==</span><span class="mi">1</span><span class="p">){</span>
</span><span class='line'>      <span class="nv">$seq</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">;</span>
</span><span class='line'>    <span class="p">}</span><span class="k">else</span><span class="p">{</span>
</span><span class='line'>      <span class="n">print_sequence</span><span class="p">(</span><span class="nv">$seq</span><span class="p">,</span><span class="mi">80</span><span class="p">,</span><span class="nv">$strand</span><span class="p">);</span>
</span><span class='line'>      <span class="nv">$seq</span><span class="o">=</span><span class="s">&#39;&#39;</span><span class="p">;</span>
</span><span class='line'>    <span class="p">}</span>
</span><span class='line'>    <span class="k">print</span> <span class="s">&quot;&gt;$feature-&gt;{_gsf_tag_hash}-&gt;{Name}-&gt;[0]|$feature-&gt;{_gsf_tag_hash}-&gt;{ID}-&gt;[0]\n&quot;</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nv">$feature</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_primary_tag</span><span class="p">}</span><span class="o">=~</span><span class="sr">/cds/i</span><span class="p">){</span>
</span><span class='line'>    <span class="k">my</span> <span class="nv">$seq_temp</span><span class="o">=</span><span class="nv">$db</span><span class="o">-&gt;</span><span class="n">seq</span><span class="p">(</span><span class="nv">$feature</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_gsf_seq_id</span><span class="p">},</span> <span class="nv">$feature</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_location</span><span class="p">}</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_start</span><span class="p">}</span><span class="o">=&gt;</span><span class="nv">$feature</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_location</span><span class="p">}</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_end</span><span class="p">});</span>
</span><span class='line'>    <span class="k">if</span><span class="p">(</span><span class="nv">$feature</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_location</span><span class="p">}</span><span class="o">-&gt;</span><span class="p">{</span><span class="n">_strand</span><span class="p">}</span><span class="o">=~</span><span class="sr">/-/</span><span class="p">){</span>  <span class="c1"># to know the strand</span>
</span><span class='line'>      <span class="nv">$seq</span><span class="o">=</span><span class="nv">$seq_temp</span><span class="o">.</span><span class="nv">$seq</span><span class="p">;</span>
</span><span class='line'>      <span class="nv">$strand</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span>
</span><span class='line'>    <span class="p">}</span><span class="k">else</span><span class="p">{</span>
</span><span class='line'>      <span class="nv">$seq</span><span class="o">.=</span><span class="nv">$seq_temp</span><span class="p">;</span>
</span><span class='line'>      <span class="nv">$strand</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span>
</span><span class='line'>    <span class="p">}</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'><span class="n">print_sequence</span><span class="p">(</span><span class="nv">$seq</span><span class="p">,</span><span class="mi">80</span><span class="p">,</span><span class="nv">$strand</span><span class="p">);</span>
</span><span class='line'>
</span><span class='line'><span class="nv">$gffio</span><span class="o">-&gt;</span><span class="nb">close</span><span class="p">();</span>
</span><span class='line'>
</span><span class='line'><span class="k">sub </span><span class="nf">print_sequence</span> <span class="p">{</span>
</span><span class='line'>  <span class="k">my</span><span class="p">(</span><span class="nv">$sequence</span><span class="p">,</span> <span class="nv">$length</span><span class="p">,</span><span class="nv">$strand</span><span class="p">)</span> <span class="o">=</span> <span class="nv">@_</span><span class="p">;</span>
</span><span class='line'>  <span class="k">if</span><span class="p">(</span><span class="nv">$strand</span><span class="o">==</span><span class="mi">0</span><span class="p">){</span>  <span class="c1"># if the sequence is on the minus strand, get its reverse-complement counterpart</span>
</span><span class='line'>    <span class="nv">$sequence</span><span class="o">=~</span><span class="nb">tr</span><span class="sr">/ACGTacgt/</span><span class="n">TGCAtgca</span><span class="o">/</span><span class="p">;</span>
</span><span class='line'>    <span class="nv">$sequence</span><span class="o">=</span><span class="nb">reverse</span> <span class="nv">$sequence</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'>  <span class="k">for</span> <span class="p">(</span> <span class="k">my</span> <span class="nv">$pos</span> <span class="o">=</span> <span class="mi">0</span> <span class="p">;</span> <span class="nv">$pos</span> <span class="o">&lt;</span> <span class="nb">length</span><span class="p">(</span><span class="nv">$sequence</span><span class="p">)</span> <span class="p">;</span> <span class="nv">$pos</span> <span class="o">+=</span> <span class="nv">$length</span> <span class="p">)</span> <span class="p">{</span>
</span><span class='line'>    <span class="k">print</span> <span class="nb">substr</span><span class="p">(</span><span class="nv">$sequence</span><span class="p">,</span> <span class="nv">$pos</span><span class="p">,</span> <span class="nv">$length</span><span class="p">),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
</span><span class='line'>  <span class="p">}</span>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>

</div>


  <footer>
    <p class="meta">
      
  


  


  <span class="byline author vcard">Authored by <span class="fn">
  
    <a href="https://plus.google.com/113645349664564078087" rel="author">Bioops</a>
  
  </span></span>


      




<time class='entry-date' datetime='2011-03-08T00:00:00+00:00'><span class='date'><span class='date-month'>Mar</span> <span class='date-day'>8</span><span class='date-suffix'>th</span>, <span class='date-year'>2011</span></span> <span class='time'>12:00 am</span></time>
      
      

<span class="categories">
  
    <a class='category' href='/category/bioinformatics/'>bioinformatics</a>, <a class='category' href='/category/genomics/'>genomics</a>, <a class='category' href='/category/perl/'>perl</a>, <a class='category' href='/category/programming/'>programming</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://bioops.info/2011/03/parse-gff3-file/" data-via="" data-counturl="http://bioops.info/2011/03/parse-gff3-file/" >Tweet</a>
  
  
  <div class="g-plusone" data-size="medium"></div>
  
  
    <div class="fb-like" data-layout="button_count" data-send="false" data-width="300" data-show-faces="false"></div>
  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left" href="/2011/03/shotgun-reads/" title="Previous Post: Shotgun Reads">&laquo; Shotgun Reads</a>
      
      
        <a class="basic-alignment right" href="/2011/03/vim-commands/" title="Next Post: useful, essential and most often used Vim commands">useful, essential and most often used Vim commands &raquo;</a>
      
    </p>
  </footer>
</article>

  <section>
    <h1>Comments</h1>
    <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
  </section>


</div>

    </div>
  </div>
  <footer role="contentinfo"><!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>

<p>
  Copyright &copy; 2016 - Bioops -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a> | Themed with <a href="https://github.com/lucaslew/whitespace">Whitespace</a></span>
</p>


</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'bioopsgithub';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://bioops.info/2011/03/parse-gff3-file/';
        var disqus_url = 'http://bioops.info/2011/03/parse-gff3-file/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>






<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
